<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Marc Subirana-Granés" />
  <meta name="author" content="Jill Hoffman" />
  <meta name="author" content="Haoyu Zhang" />
  <meta name="author" content="Christina Akirtava" />
  <meta name="author" content="Sutanu Nandi" />
  <meta name="author" content="Kevin Fotso" />
  <meta name="author" content="Milton Pividori" />
  <meta name="dcterms.date" content="2024-10-10" />
  <meta name="keywords" content="genetic studies, gwas, twas, gene modules, gene co-expression networks, QTL, complex traits" />
  <title>Genetic studies through the lens of gene networks</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="Genetic studies through the lens of gene networks" />
  <meta name="citation_title" content="Genetic studies through the lens of gene networks" />
  <meta property="og:title" content="Genetic studies through the lens of gene networks" />
  <meta property="twitter:title" content="Genetic studies through the lens of gene networks" />
  <meta name="dc.date" content="2024-10-10" />
  <meta name="citation_publication_date" content="2024-10-10" />
  <meta property="article:published_time" content="2024-10-10" />
  <meta name="dc.modified" content="2024-10-10T16:43:30+00:00" />
  <meta property="article:modified_time" content="2024-10-10T16:43:30+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Marc Subirana-Granés" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0000-0003-3934-839X" />
  <meta name="twitter:creator" content="@msubirana20" />
  <meta name="citation_author" content="Jill Hoffman" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0000-0002-2690-2593" />
  <meta name="citation_author" content="Haoyu Zhang" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0009-0005-6025-0217" />
  <meta name="citation_author" content="Christina Akirtava" />
  <meta name="citation_author_institution" content="Department of Biochemistry and Molecular Genetics, RNA Bioscience Initiative, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0000-0001-7587-2303" />
  <meta name="twitter:creator" content="@bio2data" />
  <meta name="citation_author" content="Sutanu Nandi" />
  <meta name="citation_author_institution" content="Department of Pharmacology, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0009-0001-5887-999X" />
  <meta name="citation_author" content="Kevin Fotso" />
  <meta name="citation_author_institution" content="Office of Information Technology, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author" content="Milton Pividori" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado School of Medicine, Aurora, CO 80045, USA" />
  <meta name="citation_author_institution" content="Colorado Center for Personalized Medicine, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0000-0002-3035-4403" />
  <meta name="twitter:creator" content="@miltondp" />
  <link rel="canonical" href="https://pivlab.github.io/annual_review_of_biomedical_data_science/" />
  <meta property="og:url" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/" />
  <meta property="twitter:url" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/" />
  <meta name="citation_fulltext_html_url" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/" />
  <meta name="citation_pdf_url" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://pivlab.github.io/annual_review_of_biomedical_data_science/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://pivlab.github.io/annual_review_of_biomedical_data_science/v/c2524cc88cbe2714df69644fdc4b9d3d5105b835/" />
  <meta name="manubot_html_url_versioned" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/v/c2524cc88cbe2714df69644fdc4b9d3d5105b835/" />
  <meta name="manubot_pdf_url_versioned" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/v/c2524cc88cbe2714df69644fdc4b9d3d5105b835/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Genetic studies through the lens of gene networks</h1>
</header>
<p><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css"></p>
<p><span class="banner lightblue"><i class="fas fa-info-circle fa-lg"></i> <strong>When citing this paper, please use the following:</strong><br><br>
<em>Subirana-Granés M, Hoffman J, Zhang H, Nandi S, Akirtava C, Fotso K, Pividori M. Genetic studies through the lens of gene networks. 2025. Annu. Rev. Biomed. Data Sci. 8. https://doi.org/10.1146/annurev-biodatasci-103123-095355</em></span></p>
<p><small><em>
This manuscript
(<a href="https://pivlab.github.io/annual_review_of_biomedical_data_science/v/c2524cc88cbe2714df69644fdc4b9d3d5105b835/">permalink</a>)
was automatically generated
from <a href="https://github.com/pivlab/annual_review_of_biomedical_data_science/tree/c2524cc88cbe2714df69644fdc4b9d3d5105b835">pivlab/annual_review_of_biomedical_data_science@c2524cc</a>
on October 10, 2024.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Marc Subirana-Granés</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-3934-839X">0000-0003-3934-839X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/msubirana">msubirana</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/msubirana20">msubirana20</a>
<br>
<small>
Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Jill Hoffman</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2690-2593">0000-0002-2690-2593</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/jillhoffman">jillhoffman</a>
<br>
<small>
Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Haoyu Zhang</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0005-6025-0217">0009-0005-6025-0217</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/haoyu-zc">haoyu-zc</a>
<br>
<small>
Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Christina Akirtava</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7587-2303">0000-0001-7587-2303</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/bio2data">bio2data</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/bio2data">bio2data</a>
<br>
<small>
Department of Biochemistry and Molecular Genetics, RNA Bioscience Initiative, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Sutanu Nandi</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0001-5887-999X">0009-0001-5887-999X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/snandi-DS">snandi-DS</a>
<br>
<small>
Department of Pharmacology, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Kevin Fotso</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/kf-cuanschutz">kf-cuanschutz</a>
<br>
<small>
Office of Information Technology, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Milton Pividori</strong>
<sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3035-4403">0000-0002-3035-4403</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/miltondp">miltondp</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/miltondp">miltondp</a>
· <img src="images/mastodon.svg" class="inline_icon" width="16" height="16" alt="Mastodon icon" />
<a href="https://genomic.social/@miltondp">@miltondp@genomic.social</a>
<br>
<small>
Department of Biomedical Informatics, University of Colorado School of Medicine, Aurora, CO 80045, USA; Colorado Center for Personalized Medicine, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
· Funded by The National Human Genome Research Institute (K99/R00 HG011898); The Eunice Kennedy Shriver National Institute of Child Health and Human Development (R01 HD109765)
</small></p></li>
</ul>
<div id="correspondence">
<p>✉ — Correspondence possible via <a href="https://github.com/pivlab/annual_review_of_biomedical_data_science/issues">GitHub Issues</a>
or email to
Milton Pividori &lt;milton.pividori@cuanschutz.edu&gt;.</p>
</div>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>This is a very cool paper about something.
We are doing this and that.
And this are the conclusions.
And this is in <strong>boldface</strong>.</p>
<h2 id="introduction">Introduction</h2>
<p>Understanding the genetics of complex traits remains a formidable challenge.
Genome-Wide Association Studies (GWAS) have been instrumental in advancing our understanding of the genetics behind complex traits.
By scanning the genomes of large populations, GWAS identify associations between genetic variants and traits, providing valuable insights into the genetic architecture of diseases and other phenotypic traits.
This approach has been particularly successful in identifying thousands of variant-trait associations, shedding light on potential genetic factors involved in a wide range of conditions.
However, a significant limitation of GWAS is that most associated variants reside in non-coding regions of the genome, underscoring the critical role of gene regulation in generating diverse phenotypes from identical DNA sequences.</p>
<p>To bridge this gap, various methods have been developed to enhance the interpretation of GWAS findings.
One such approach is MAGMA, which assigns variants to genes based on physical proximity without considering gene expression data.
However, it has been shown that the closest gene to a variant is not necessarily the gene influenced by that variant, revealing the limitations of proximity-based methods.
Transcriptome-Wide Association Studies (TWAS) overcome this limitation by linking genetic variants to gene expression (eQTLs).
TWAS has evolved into a more general framework to test other molecular phenotypes (molQTLs).</p>
<p>Despite their advancements, TWAS and similar approaches typically focus on single variants or individual genes, thereby overlooking the intricate gene–gene interactions that are pivotal according to modern theories such as the omnigenic model of genetic architecture for complex traits.
Capturing these interactions necessitates more sophisticated methodologies.
One effective approach involves leveraging gene modules based on co-expression patterns.
Machine learning methods, such as decomposition techniques and autoencoders, have demonstrated superior performance in identifying these co-expression modules by capturing latent patterns in high-dimensional datasets.
However, the reliance on co-expression alone can lead to false positives, limiting the accuracy of these modules.</p>
<p>To mitigate this, integrating additional data sources is essential.
For instance, PhenoPLIER exemplifies an integrative approach by combining genetic variant-trait associations from GWAS and variant-gene expression-trait associations from TWAS with groups of functionally related genes.
This integration strengthens the reliability of gene modules by cross-validating co-expression patterns with genetic evidence, thus reducing the likelihood of false positives and enhancing the biological relevance of the results.</p>
<p>In this review, we explore how integrating gene modeling machine learning frameworks with genetic and multi-omics data enhances our understanding of the molecular mechanisms underlying complex traits and diseases, effectively approaching genetic studies through the lens of gene networks.</p>
<h2 id="single-variant-single-gene-approaches">Single variant / single gene approaches</h2>
<div id="fig:fig1" class="fignos">
<figure>
<img src="images/fig1.svg" style="width:100.0%" alt="Figure 1: Major approaches based on single variant/single gene strategies to understand the genetic basis of complex traits. GWAS identifies associations between DNA variants and traits. All the other approaches test how different molecular mechanisms might mediate variant-trait associations from GWAS. TWAS uses variants linked to changes in gene expression (i.e., expression quantitative trait loci or eQTLs). PWAS links protein functionality to traits via protein QTLs (pQTLs). HAWAS maps histone modifications to traits using hQTLs, while MWAS explores DNA methylation patterns in relation to traits through mQTLs." />
<figcaption aria-hidden="true"><span>Figure 1:</span> <strong>Major approaches based on single variant/single gene strategies to understand the genetic basis of complex traits</strong>.
GWAS identifies associations between DNA variants and traits.
All the other approaches test how different molecular mechanisms might mediate variant-trait associations from GWAS.
TWAS uses variants linked to changes in gene expression (i.e., expression quantitative trait loci or eQTLs).
PWAS links protein functionality to traits via protein QTLs (pQTLs).
HAWAS maps histone modifications to traits using hQTLs, while MWAS explores DNA methylation patterns in relation to traits through mQTLs.</figcaption>
</figure>
</div>
<h3 id="from-genetic-variants-to-traits-genome-wide-association-study">From genetic variants to traits: genome-wide association study</h3>
<p>In a simple trait, a single gene can be responsible for the disease (monogenic), for example in the case of Sickle Cell Anemia and Huntington’s disease.
However, most traits are not so simplistic and are the result of many mutations across the genome (polygenic).
Even monogenic traits can sometimes be influenced by the polygenic background <span class="citation" data-cites="VqGYQMXO">[<a href="#ref-VqGYQMXO" role="doc-biblioref">1</a>]</span>.
Genome-wide association studies (GWAS) examine the relationship between specific genetic variants and phenotypes by comparing allele frequencies in individuals of similar ancestry with distinct phenotypic traits (<strong>Figure <a href="#fig:fig1">1</a></strong>).
Both causal and non-causal variants are found to be significant and due to the phenomena of linkage disequilibrium, the results of GWAS are often grouped into risk loci.
These loci represent clusters of variants that demonstrate significant associations, although not every variant may be causal <span class="citation" data-cites="1873zCaaX">[<a href="#ref-1873zCaaX" role="doc-biblioref">2</a>]</span>.</p>
<p>Since the first GWAS on age-related macular degeneration (AMD) in 2005, the GWAS catalog rapidly expanded, containing SNP-trait associations across &gt;5,000 human traits <span class="citation" data-cites="VRCjjZ9L">[<a href="#ref-VRCjjZ9L" role="doc-biblioref">3</a>]</span>.
These studies have successfully identified risk loci for a variety of traits such as type 2 diabetes, auto-immune disease, schizophrenia, major depressive disorder, and many more <span class="citation" data-cites="17B4AigZT yqBcaXHD">[<a href="#ref-17B4AigZT" role="doc-biblioref">4</a>,<a href="#ref-yqBcaXHD" role="doc-biblioref">5</a>]</span>.
As GWAS participants increase, more loci are able to be identified.
For example, a study investigating insomnia with a sample size of &gt;1M identified 202 risk loci <span class="citation" data-cites="sBTrhvlJ">[<a href="#ref-sBTrhvlJ" role="doc-biblioref">6</a>]</span>, compared to earlier studies with a sample size of ~110,00, which were only able to identify 3 risk loci <span class="citation" data-cites="crCAymA1">[<a href="#ref-crCAymA1" role="doc-biblioref">7</a>]</span>.
Beyond the identification of risk loci, GWAS have also led to the discovery of novel biological mechanisms in Crohn’s disease <span class="citation" data-cites="s6z5ZuY3 gxwza7fO">[<a href="#ref-s6z5ZuY3" role="doc-biblioref">8</a>,<a href="#ref-gxwza7fO" role="doc-biblioref">9</a>]</span>, rheumatoid arthritis <span class="citation" data-cites="HicyXvA4">[<a href="#ref-HicyXvA4" role="doc-biblioref">10</a>]</span>, and more.</p>
<p>Although GWAS have identified numerous genetic variants associated with complex traits, translating these findings into biological insights remains challenging <span class="citation" data-cites="17B4AigZT">[<a href="#ref-17B4AigZT" role="doc-biblioref">4</a>]</span>.
For example, genes can have epistatic interactions where a secondary loci affects a primary loci.
The peripheral effects of the secondary loci are often too small to be picked up through GWAS alone <span class="citation" data-cites="IrGVpNji">[<a href="#ref-IrGVpNji" role="doc-biblioref">11</a>]</span>.
Furthermore, any significant GWAS variants are located in non-coding regions, making it difficult to identify the specific genes that drive trait associations <span class="citation" data-cites="BqjxxhyF">[<a href="#ref-BqjxxhyF" role="doc-biblioref">12</a>]</span>.
A simple strategy to link GWAS-associated variants with a gene is by physical proximity, which typically selects the closest gene to the most significant SNP at each locus.
Approaches like MAGMA <span class="citation" data-cites="19XiXgYmd">[<a href="#ref-19XiXgYmd" role="doc-biblioref">13</a>]</span> are based on proximity to compute a gene-trait association.
However, SNPs can have distal effects on the expression of the target gene <span class="citation" data-cites="9Pr9idng OBxYzUMh 167QL5tMV WfdUHVDA">[<a href="#ref-9Pr9idng" role="doc-biblioref">14</a>,<a href="#ref-OBxYzUMh" role="doc-biblioref">15</a>,<a href="#ref-167QL5tMV" role="doc-biblioref">16</a>,<a href="#ref-WfdUHVDA" role="doc-biblioref">17</a>]</span>.</p>
<p>Recent approaches combine multi-omics data from various cell types and tissues with GWAS to identify potential mechanisms of SNPs and the associated gene through molecular quantitative trait loci (molQTLs) <span class="citation" data-cites="QIEm5ZWK">[<a href="#ref-QIEm5ZWK" role="doc-biblioref">18</a>]</span> (Figure <a href="#fig:fig1">1</a>).
For example, an expression quantitative trait loci (eQTL) is a genetic region associated with the expression levels of a nearby or distant gene <span class="citation" data-cites="1FHKg3JK5">[<a href="#ref-1FHKg3JK5" role="doc-biblioref">19</a>]</span>.
eQTL studies estimate that the majority of heritability is explained by the combination of multiple weak <em>trans</em>-eQTLs <span class="citation" data-cites="sGgkLsyy">[<a href="#ref-sGgkLsyy" role="doc-biblioref">20</a>]</span>.</p>
<p>In the next section, we briefly review computational approaches that test different molecular phenotypes via QTLs to link variants with genes.
Both GWAS and QTL studies have been penalized by their large multiple testing burden, causing the need to adopt a high level of significance.
This results in GWAS being underpowered to detect all heritability explained by SNPs <span class="citation" data-cites="17B4AigZT">[<a href="#ref-17B4AigZT" role="doc-biblioref">4</a>]</span>.</p>
<h3 id="from-gwas-to-gene-transcriptome-wide-association-studies">From GWAS to gene: transcriptome-wide association studies</h3>
<p>Transcriptome-wide association studies (TWAS) integrate GWAS with gene expression data (<strong>Figure <a href="#fig:fig1">1</a></strong>) from eQTL analysis, to prioritize genes whose expression across different tissues is influenced by GWAS variants <span class="citation" data-cites="ReOPt75u l6ogswV3">[<a href="#ref-ReOPt75u" role="doc-biblioref">21</a>,<a href="#ref-l6ogswV3" role="doc-biblioref">22</a>]</span>.
By leveraging predicted gene expression levels, TWAS provides a mechanistic link between genetic variants and traits, allowing researchers to move beyond associations with individual SNPs to identify putatively causal genes <span class="citation" data-cites="l6ogswV3">[<a href="#ref-l6ogswV3" role="doc-biblioref">22</a>]</span>.
Since TWAS models the genetic regulation of gene expression, the approach allows researchers to impute expression levels in GWAS cohorts where expression data may not be available.
A key advantage of TWAS over GWAS lies in its ability to increase interpretability by providing a gene-trait association: TWAS connects trait-associated SNPs (which are mostly non-coding) to genes, which are biologically functional units.</p>
<p>Several TWAS approaches have been introduced <span class="citation" data-cites="RmhTH35O">[<a href="#ref-RmhTH35O" role="doc-biblioref">23</a>]</span>, including PrediXcan <span class="citation" data-cites="Z8bvDdVq">[<a href="#ref-Z8bvDdVq" role="doc-biblioref">24</a>]</span>, FUSION <span class="citation" data-cites="1D63fEEPb">[<a href="#ref-1D63fEEPb" role="doc-biblioref">25</a>]</span> and TIGAR <span class="citation" data-cites="19Klmizj8">[<a href="#ref-19Klmizj8" role="doc-biblioref">26</a>]</span>.
However, all of them implement a similar framework that consists in three steps: 1) model training, 2) gene expression imputation, and 3) gene-trait association.
For example, during 1), PrediXcan, builds one expression prediction model per gene and tissue using penalized linear regression with ElasticNet to model sparse genetic architectures.
These models contain weights for each SNP used as predictor for a gene expression in a tissue.
Given genotype data in a cohort without measured gene expression, during 2), the SNPs weights from the models can be used to impute tissue-specific gene expression for individuals.
During 3), a gene-tissue-trait association is computed by correlating the tissue-specific imputed gene expression with the trait of interest.
Most methods (such as Summary-PrediXcan <span class="citation" data-cites="vLyTudUB">[<a href="#ref-vLyTudUB" role="doc-biblioref">27</a>]</span> or S-PrediXcan), however, offer a shortcut by computing a gene-tissue-trait association directly from GWAS summary statistics without the need of individual-level data.
This process, however, requires the user to select a tissue of interest, which might not be straightforward <span class="citation" data-cites="l6ogswV3">[<a href="#ref-l6ogswV3" role="doc-biblioref">22</a>]</span>.
To address this limitation, approaches such as MultiXcan <span class="citation" data-cites="1FFzCXo1s">[<a href="#ref-1FFzCXo1s" role="doc-biblioref">28</a>]</span> or UTMOST <span class="citation" data-cites="93R9hBin">[<a href="#ref-93R9hBin" role="doc-biblioref">29</a>]</span> combine information across tissues to generate a gene-trait association.
These multi-tissue approaches are generally more powerful than single-tissue ones, although they do not provide a direction of effect (i.e., whether a higher or lower predicted expression is associated with a higher or lower disease risk).</p>
<h3 id="going-beyond-twas">Going beyond TWAS</h3>
<p>The flexibility of this 3-step framework can also be used to test whether other molecular phenotypes might mediate the association between GWAS variants and a trait of interest.
In addition to integrating eQTLs, the framework has been implemented also with splicing QTLs (sQTLs), protein QTLs (pQTLs), histone acetylation QTLs (haQTLs) and methylation QTLs (mQTLs).</p>
<p>PWAS complements GWAS and TWAS by aggregating genetic variations in protein-coding regions to assess their combined impact on protein function and phenotype (<strong>Figure <a href="#fig:fig1">1</a></strong>) <span class="citation" data-cites="1Dsj6bEd3">[<a href="#ref-1Dsj6bEd3" role="doc-biblioref">30</a>]</span>.
Unlike GWAS, which focuses on individual variants, PWAS evaluates the cumulative effects of multiple coding variants, reducing the scale of multiple testing, detecting complex inheritance patterns, and offering deeper insights into potential disease links.
PWAS employs the FIRM machine learning model to evaluate how variants—such as missense, nonsense, and frameshift mutations—affect protein function by assigning impairment scores <span class="citation" data-cites="1Dsj6bEd3">[<a href="#ref-1Dsj6bEd3" role="doc-biblioref">30</a>]</span>.
The aggregate effects of these variants on each gene are then assessed, followed by statistical tests to identify significant associations between gene scores and phenotypes.
For example, PWAS confirmed the association of <em>MUTYH</em> with colorectal cancer using UK Biobank data, even when no individual variant reached genome-wide significance in GWAS, highlighting its ability to uncover functional associations missed by other methods.
However, PWAS relies on high-quality proteomic data, may miss non-coding variant associations, and is computationally intensive.</p>
<p>Epigenome-wide association studies (EWAS) encompass methodologies such methylome-wide association studies (MWAS) (<strong>Figure <a href="#fig:fig1">1</a></strong>) and histone acetylome-wide association studies (HAWAS) (<strong>Figure <a href="#fig:fig1">1</a></strong>), based on haQTLs and mQLTs respectability, which focus on specific types of epigenetic modifications to reveal their roles in gene regulation and disease etiology <span class="citation" data-cites="TPqvAJda">[<a href="#ref-TPqvAJda" role="doc-biblioref">31</a>]</span>.
MWAS targets DNA methylation patterns, identifying loci where methylation changes are linked to particular traits or diseases.
For instance, a study by Shen et al. demonstrated a causal relationship between DNA methylation and depression, indicating that epigenetic modifications may mediate genetic risk for psychiatric disorders <span class="citation" data-cites="1DNxNkLsT">[<a href="#ref-1DNxNkLsT" role="doc-biblioref">32</a>]</span>.
HAWAS focuses on histone acetylation modifications, which are crucial for regulating chromatin structure and gene expression.
Del Rosario et al.conducted a HAWAS identifying over 2,000 differentially acetylated loci in immune cells from Mycobacterium tuberculosis-infected individuals, linking these changes to gene expression and potassium channel genes like <em>KCNJ15</em> in modulating apoptosis and Mtb clearance.
haQTL analysis further revealed variants associated with immune phenotypes, complementing GWAS findings and enhancing understanding of disease mechanisms <span class="citation" data-cites="1BYHeZlgB">[<a href="#ref-1BYHeZlgB" role="doc-biblioref">33</a>]</span>.
The tissue-specific nature of epigenetic modifications, alongside the capacity to capture cellular plasticity and environmental influences, enhances our insight into the effects of genetic variants across distinct tissues, temporal cellular states, and gene-environment interactions <span class="citation" data-cites="FeaSnwhX KfWCLspO">[<a href="#ref-FeaSnwhX" role="doc-biblioref">34</a>,<a href="#ref-KfWCLspO" role="doc-biblioref">35</a>]</span>.
Despite the advancements facilitated by single variant and single gene methodologies, a common thread persists: the focus remains on one gene at a time.
The underlying expectation is that identifying a gene linked to a trait will directly unveil the biological mechanisms driving disease processes.
While this has been successful in certain monogenic disorders, complex traits often involve intricate interactions among multiple genes and environmental factors.
Consequently, the one-gene-at-a-time paradigm may oversimplify the multifaceted nature of these traits.
In light of these considerations, it becomes imperative to reevaluate our approaches to dissecting the genetic architecture of complex traits.
This necessitates a shift towards methodologies that can capture the polygenic and network-based interactions inherent in complex diseases.</p>
<h2 id="from-single-genes-to-gene-networks-the-omnigenic-model-for-complex-traits">From single genes to gene networks: the omnigenic model for complex traits</h2>
<h3 id="omnigenic-model-as-a-framework-for-deciphering-genetic-architecture-of-traits">Omnigenic model as a framework for deciphering genetic architecture of traits</h3>
<div id="fig:fig2" class="fignos">
<figure>
<img src="images/fig2.svg" style="width:100.0%" alt="Figure 2: Schematic of the omnigenic model. Schematic representation of the biological networks and associated phenotypes in an omnigenic context. Nodes represent individual genes, while edges indicate functional relationships between the nodes, such as transcription factor (TF) binding, splicing events, post-translational modifications (PTMs), gene co-expression, non-coding RNA (ncRNA), and protein-protein interactions (PPIs). The size of the arrows connecting phenotypes indicates the magnitude of their effect, with thicker arrows representing larger phenotypic influences. iological networks are indicated by shaded regions, where genes might play a role in different networks. Genes are categorized as “core” (diamonds) if they have a direct effect on the phenotype, or as “peripheral” (circles) if they have an indirect effect on the phenotype by moduling core genes. Phenotypes are linked to their relevant biological networks, highlighting the interplay between different systems and their contribution to phenotypic outcomes." />
<figcaption aria-hidden="true"><span>Figure 2:</span> <strong>Schematic of the omnigenic model</strong>.
Schematic representation of the biological networks and associated phenotypes in an omnigenic context.
Nodes represent individual genes, while edges indicate functional relationships between the nodes, such as transcription factor (TF) binding, splicing events, post-translational modifications (PTMs), gene co-expression, non-coding RNA (ncRNA), and protein-protein interactions (PPIs).
The size of the arrows connecting phenotypes indicates the magnitude of their effect, with thicker arrows representing larger phenotypic influences.
iological networks are indicated by shaded regions, where genes might play a role in different networks.
Genes are categorized as “core” (diamonds) if they have a direct effect on the phenotype, or as “peripheral” (circles) if they have an indirect effect on the phenotype by moduling core genes.
Phenotypes are linked to their relevant biological networks, highlighting the interplay between different systems and their contribution to phenotypic outcomes.</figcaption>
</figure>
</div>
<p>In the early 20th century, a significant debate arose between Mendelian geneticists and biometricians concerning the inheritance of complex traits.
Mendelians emphasized single gene traits with discrete inheritance patterns, while biometricians argued that such models could not account for the continuous phenotypic variation observed in populations.
This conflict highlighted a fundamental question: how can discrete genetic factors produce continuous phenotypic variation? <span class="citation" data-cites="vpIDZCSa 15ldVppuv">[<a href="#ref-vpIDZCSa" role="doc-biblioref">36</a>,<a href="#ref-15ldVppuv" role="doc-biblioref">37</a>]</span>.
The disagreement was resolved by R.A. Fisher in 1918, who introduced the infinitesimal model, demonstrating that if many genes influence a trait, their combined effects can generate continuous phenotypic distributions <span class="citation" data-cites="Qxk70FFV">[<a href="#ref-Qxk70FFV" role="doc-biblioref">38</a>]</span>.
Despite the success of Fisher’s infinitesimal model, the actual number of genes involved in complex traits and the magnitude of their effects remained uncertain for much of the 20th century.
Translating genotypes into phenotypes proved challenging due to the complexity of the genetic mechanisms driving trait variation.
In this context, GWAS brought a new perspective for identifying specific genetic variants with functional effects on complex traits.
Over a decade of GWAS has led to several unexpected findings:
The strongest GWAS associations exhibit modest effect sizes on disease risk, collectively accounting for only a minor fraction of the predicted heritability (missing heritability) <span class="citation" data-cites="RASeYPIy">[<a href="#ref-RASeYPIy" role="doc-biblioref">39</a>]</span>.
It is now understood that much of this missing heritability arises from a vast number of common variants with small effects that remain undetected in current sample sizes <span class="citation" data-cites="1AOy1zxAv">[<a href="#ref-1AOy1zxAv" role="doc-biblioref">40</a>]</span>.
A surprising uniform distribution of these small-effect significant variants exists throughout the genome, including non-coding regions, with a notable concentration in regulatory elements such as enhancers and promoters <span class="citation" data-cites="mwTa2RUK lVJVFaaZ">[<a href="#ref-mwTa2RUK" role="doc-biblioref">41</a>,<a href="#ref-lVJVFaaZ" role="doc-biblioref">42</a>]</span>.
For instance, it is estimated that 71%–100% of 1-MB windows in the genome contribute to heritability for schizophrenia <span class="citation" data-cites="XvQe1H3A">[<a href="#ref-XvQe1H3A" role="doc-biblioref">43</a>]</span>.
Trait variation is influenced by the broader genome, not only by genes with direct biological links to specific phenotypes.
However, the idea that nearly all genomic variants could exert direct additive effects on a given phenotype is biologically unlikely <span class="citation" data-cites="vpIDZCSa">[<a href="#ref-vpIDZCSa" role="doc-biblioref">36</a>]</span>.
In response to these unexpected findings, Boyle et al. proposed the omnigenic model <span class="citation" data-cites="vpIDZCSa">[<a href="#ref-vpIDZCSa" role="doc-biblioref">36</a>]</span>, a conceptual framework in which the genetic architecture of complex traits could be explained by highly interconnected gene regulatory networks.
The omnigenic model reframes the earlier infinitesimal model by distinguishing between “core” genes, which have a direct biological role in the disease or trait, and “peripheral” genes, which influence core genes and thus indirectly affect the disease or trait through regulatory networks <span class="citation" data-cites="vpIDZCSa">[<a href="#ref-vpIDZCSa" role="doc-biblioref">36</a>]</span>.
The omnigenic model, unlike the traditional infinitesimal model, offers a mechanistic rationale for complex traits architecture based on molecular and cellular biology, understanding the molecular pathways that connect genetic variation to phenotypic traits (<strong>Figure <a href="#fig:fig2">2</a></strong>).</p>
<p>The key proposals of this model are: 1) nearly all genes expressed in cells relevant to the trait have the potential to impact the regulation of core genes 2) that for typical traits, nearly all heritability is attributable to variation near peripheral genes.
Consequently, while core genes function as the primary drivers of disease, it is the cumulative effects of numerous peripheral gene variants that determine polygenic risk <span class="citation" data-cites="LXvTZzEA">[<a href="#ref-LXvTZzEA" role="doc-biblioref">44</a>]</span>.
This framework provides a possible explanation for several previously identified problems, including widespread pleiotropy (the ability of a single gene to affect multiple traits), polygenicity (the involvement of many genes in the manifestation of a single trait), the bias effect size where small-effect variants can collectively account for a large portion of heritability, and the uniform distribution of these effects across the genome.</p>
<h3 id="success-stories-of-the-omnigenic-model">Success stories of the omnigenic model</h3>
<p>Since the introduction of the omnigenic model, numerous studies have adopted it as a framework to deepen our understanding of the genetic architecture of complex traits and the molecular mechanisms that drive them.</p>
<p>In schizophrenia, the application of the omnigenic model has revealed specific core gene sets associated with the disorder.
Notably, the <em>TCF4</em> gene set maintained its significant impact even after excluding SNPs within the <em>TCF4</em> gene itself.
This finding suggests that peripheral genes within the set contribute to the development of schizophrenia.
Additionally, the observed systematic floor effect across polygenic scores aligns with the model’s prediction that most genes expressed in relevant cells contribute to heritability, thereby highlighting the model’s potential in explaining the broad genetic contributions to complex psychiatric conditions <span class="citation" data-cites="fV6THoBG">[<a href="#ref-fV6THoBG" role="doc-biblioref">45</a>]</span>.</p>
<p>Li et al. (2024) identified a significantly connected subgraph formed by cancer-affected coding genes and ncRNAs by focusing on connectivity as a key topological feature.
This approach highlighted the essential role of ncRNAs in linking fragmented cancer-affected genes, consistent with the omnigenic framework’s premise that peripheral genes, including non-coding elements, contribute to complex traits such as cancer.
Importantly, the inclusion of ncRNAs enhanced the identification of cancer-related pathways, indicating that a comprehensive network model encompassing ncRNAs is more effective in characterizing disease relationships than models concentrating solely on coding genes <span class="citation" data-cites="13VkGrWQR">[<a href="#ref-13VkGrWQR" role="doc-biblioref">46</a>]</span>.</p>
<p>Empirical support for the omnigenic theory extends beyond human diseases.
Sundell et al. demonstrated this concept in European black poplars, showing that both core and peripheral genes play a crucial role in predicting phenotypes.
While core genes are indeed important, the information they provide must be complemented by other genes to ensure accurate phenotype predictions.
Furthermore, peripheral genes were found to carry significant biological information contributing to robust predictions <span class="citation" data-cites="EgquVUSg">[<a href="#ref-EgquVUSg" role="doc-biblioref">47</a>]</span>.</p>
<h3 id="limitations-of-the-omnigenic-model-from-a-theoretical-to-a-practical-approach">Limitations of the omnigenic model: from a theoretical to a practical approach</h3>
<p>In essence, the omnigenic model proposes a specific mechanism: that the genetic architecture of complex traits is intrinsically interconnected, involving a multitude of genes that contribute to phenotypic variation both directly (core genes) and indirectly (peripheral genes).
This framework facilitates an understanding of the highly polygenic and often subtle genetic effects on complex traits, shifting the focus from individual genes to gene networks.
While this proves successful in some fields the model exhibits certain limitations and has faced several criticisms.</p>
<p>One major concern is the lack of clear, quantitative methods for identifying peripheral genes or estimating their contribution to heritability.
While the model suggests that most genetic effects arise from peripheral genes, it remains unclear how to statistically define and measure these effects [<span class="citation" data-cites="82fs3SZo">[<a href="#ref-82fs3SZo" role="doc-biblioref">48</a>]</span>]<span class="citation" data-cites="doi:10.20900/jpbs.20170014">[<a href="#ref-doi:10.20900/jpbs.20170014" role="doc-biblioref"><strong>doi:10.20900/jpbs.20170014?</strong></a>]</span>.
Interestingly, research suggests that most heritability in complex traits is controlled by <em>trans</em> effects rather than <em>cis</em> effects, despite their small effect sizes <span class="citation" data-cites="LXvTZzEA">[<a href="#ref-LXvTZzEA" role="doc-biblioref">44</a>]</span>.
This paradox can be explained by natural selection acting most strongly against variants with large effects, limiting the contribution of directly biologically important genes to heritability <span class="citation" data-cites="LXvTZzEA">[<a href="#ref-LXvTZzEA" role="doc-biblioref">44</a>]</span>.
Thus, understanding <em>trans</em> effects, where a variant at one locus influences genes at distant loci is crucial for identifying and quantifying peripheral genes and their contribution to core traits.
However, detecting and quantifying <em>trans</em> effects remains a significant challenge due to their weak individual effects.
The omnigenic model postulates that nearly all genes expressed in cells relevant to a given trait have the potential to influence the regulation of core genes.
However, Yengo et al., in their recent GWAS on height, challenge this view.
They introduced the first “saturated” GWAS, where further increases in sample size are unlikely to yield additional genetic insights unless participant diversity or variant inclusion is expanded.
Their findings revealed that only a subset of genomic regions (~20%) seem to contribute to height determination in individuals of European ancestry.
This suggests that, at least for height, not all genes play a role in the trait as predicted by the omnigenic model <span class="citation" data-cites="HFRaLwGQ">[<a href="#ref-HFRaLwGQ" role="doc-biblioref">49</a>]</span>.
Furthermore, the omnigenic model has been criticized for their binary classification of genes into core and peripheral, which might oversimplify biological systems and potentially underestimate their true complexity <span class="citation" data-cites="82fs3SZo">[<a href="#ref-82fs3SZo" role="doc-biblioref">48</a>]</span>.
It also fails to account for gene-environment interactions, which play a crucial role in shaping complex traits <span class="citation" data-cites="dg9nKuy0">[<a href="#ref-dg9nKuy0" role="doc-biblioref">50</a>]</span>.
Additionally, while the model provides a conceptual framework, it remains unclear how to translate it into a practical statistical model <span class="citation" data-cites="1GlQtJ9Yz">[<a href="#ref-1GlQtJ9Yz" role="doc-biblioref">51</a>]</span>.
As George E. P. Box remarked, “all models are wrong, but some are useful.” While we acknowledge that the omnigenic model simplifies the inherent complexity of biological systems and may not universally apply to all traits, we maintain that it remains a valuable framework for elucidating genetic architectures.
The model effectively bridges quantitative and molecular genetics offering comprehensive mechanistic insights with predictions on quantitative variation.
Moreover, the omnigenic model has been instrumental in shifting the focus from single gene analyses to network-based models.
Recognizing the necessity of translating this model into practical applications, machine learning-derived gene modules, which use this concept to infer gene-gene networks, offer a promising way to ground these theoretical ideas into practical approaches.</p>
<h1 id="from-gene-networks-to-machine-learning-derived-gene-modules-hands-on-strategies-for-inferring-gene-gene-interactions">From gene networks to machine learning derived gene modules: hands-on strategies for inferring gene-gene interactions</h1>
<div id="fig:fig3" class="fignos">
<figure>
<img src="images/fig3.svg" style="width:100.0%" alt="Figure 3: Computational methods for machine learning derived gene module detection.. a). Real biological network, where nodes represent genes and edges denote the relationships between them. It captures all possible patterns that can link genes. Some of these relationships are gene co-expression patterns, which transcriptomics can capture, such as (1), (3) and (4). Other patterns, such as protein-protein interactions (2), are not captured by transcriptomics and are thus not reflected in the co-expression analysis. b). Different approaches for capturing machine learning-derived gene modules. In this example, transcriptome data is used as input, but other data types such as methylomics, proteomics, and metabolomics can also be applied. The first row illustrates the PLIER method, which uses non-negative matrix factorization and prior knowledge to align gene modules with known biological pathways. This approach generates interpretable gene modules but only captures linear relationships. The second row shows a vanilla VAE, which can capture both linear and nonlinear patterns but does not integrate prior knowledge, resulting in less interpretable gene modules. The third row highlights interpretable VAEs, which incorporate prior knowledge into its decoder, achieving a balance between capturing complex relationships and maintaining biological interpretability." />
<figcaption aria-hidden="true"><span>Figure 3:</span> <strong>Computational methods for machine learning derived gene module detection.</strong>.
<strong>a)</strong>.
Real biological network, where nodes represent genes and edges denote the relationships between them.
It captures all possible patterns that can link genes.
Some of these relationships are gene co-expression patterns, which transcriptomics can capture, such as (1), (3) and (4).
Other patterns, such as protein-protein interactions (2), are not captured by transcriptomics and are thus not reflected in the co-expression analysis.
<strong>b).</strong> Different approaches for capturing machine learning-derived gene modules.
In this example, transcriptome data is used as input, but other data types such as methylomics, proteomics, and metabolomics can also be applied.
The first row illustrates the PLIER method, which uses non-negative matrix factorization and prior knowledge to align gene modules with known biological pathways.
This approach generates interpretable gene modules but only captures linear relationships.
The second row shows a vanilla VAE, which can capture both linear and nonlinear patterns but does not integrate prior knowledge, resulting in less interpretable gene modules.
The third row highlights interpretable VAEs, which incorporate prior knowledge into its decoder, achieving a balance between capturing complex relationships and maintaining biological interpretability.</figcaption>
</figure>
</div>
<h2 id="from-single-gene-analysis-to-gene-module-discovery-with-machine-learning-approaches">From single gene analysis to gene module discovery with machine learning approaches</h2>
<p>High-throughput technologies, particularly genome-wide gene expression profiling tools such as RNA sequencing (RNA-seq), have fundamentally transformed the landscape of molecular biology.
Unlike traditional gene-by-gene approaches, RNA-seq offers a comprehensive view of the transcriptomic landscape <span class="citation" data-cites="Zm8nV5AC">[<a href="#ref-Zm8nV5AC" role="doc-biblioref">52</a>]</span>.
These advancements, alongside conceptual frameworks like the omnigenic model, have facilitated a paradigm shift from a single gene perspective to a module-based approach, wherein groups of genes rather than individual genes are essential for elucidating the complexities of biological networks <span class="citation" data-cites="SD886fuE">[<a href="#ref-SD886fuE" role="doc-biblioref">53</a>]</span>.</p>
<p>Gene modules, a key type of biological network, consist of nodes (genes) and edges that reflect coexpression relationships between them (<strong>Figure <a href="#fig:fig3">3</a>a)</strong>).
These modules, formed by genes with coordinated expression patterns under specific biological conditions, not only reveal coexpression but could also provide valuable insights into regulatory mechanisms, gene functions, and the pathways involved in traits and diseases <span class="citation" data-cites="QUI8V4RY">[<a href="#ref-QUI8V4RY" role="doc-biblioref">54</a>]</span>.</p>
<p>Numerous approaches have been proposed for module detection in gene expression data.
Machine learning (ML) techniques are particularly proficient at addressing common challenges in omics data, such as data sparsity and high dimensionality <span class="citation" data-cites="K4KeL4gw">[<a href="#ref-K4KeL4gw" role="doc-biblioref">55</a>]</span>.
ML methods can compress correlations into lower-dimensional representations, facilitating the processing of large datasets to identify intricate coexpression patterns.
Additionally, ML approaches are capable of integrating various types of omics data enhancing the detection of gene modules, providing a more comprehensive understanding of biological networks <span class="citation" data-cites="K4KeL4gw">[<a href="#ref-K4KeL4gw" role="doc-biblioref">55</a>]</span>.</p>
<p>In this context, unsupervised ML approaches have gained significant popularity <span class="citation" data-cites="Ko8gMp8P">[<a href="#ref-Ko8gMp8P" role="doc-biblioref">56</a>]</span>.
These methods facilitate the discovery of patterns and structures within the data without requiring prior knowledge of the system, allowing for the inference of novel findings and hypotheses.
This capability is particularly advantageous for gene module inference, which aims to identify groups of coexpressed and potentially co-regulated genes that contribute to similar biological functions <span class="citation" data-cites="1BVbSrr6M">[<a href="#ref-1BVbSrr6M" role="doc-biblioref">57</a>]</span>.</p>
<p>The most common approach in unsupervised ML is clustering, with methods such as WGCNA, which has been widely used since the first gene expression datasets became available and remains one of the most popular tools today <span class="citation" data-cites="1BVbSrr6M">[<a href="#ref-1BVbSrr6M" role="doc-biblioref">57</a>]</span>.
However, these methodologies typically rely on traditional correlation measures (e.g., the Pearson correlation coefficient), which are limited to capturing linear associations between continuous data.
Newer approaches, such as the Clustermatch Correlation Coefficient (CCC), have been developed to quantify both linear and non-linear correlations in complex gene expression datasets, providing a more flexible means of detecting intricate relationships that are not apparent through linear models alone <span class="citation" data-cites="YgLIM2bZ">[<a href="#ref-YgLIM2bZ" role="doc-biblioref">58</a>]</span>.</p>
<p>In this section, given their successful applications, we will focus on decomposition approaches like Principal Component Analysis (PCA), and deep learning methods for inferring gene modules, highlighting their effectiveness in uncovering complex gene interactions and enhancing our understanding of biological networks (<strong>Figure <a href="#fig:fig3">3</a>b)</strong>).
However, we acknowledge that utilizing transcriptional data alone is insufficient for fully comprehending the intricate system of biological networks; it serves as an initial step toward a more comprehensive understanding.</p>
<h3 id="decomposition-methods">Decomposition methods</h3>
<p>Decomposition methods reduce complex, high-dimensional gene expression data into simpler components, revealing underlying patterns and structures that aid in interpreting large-scale datasets.
These methods outperform other unsupervised machine learning approaches in recovering gene modules by effectively capturing local co-expression effects present in only a subset of biological samples, while also allowing genes to be assigned to multiple modules <span class="citation" data-cites="1BVbSrr6M">[<a href="#ref-1BVbSrr6M" role="doc-biblioref">57</a>]</span>.</p>
<p>Simple models like PCA have traditionally been used to reduce dimensionality in gene expression data, offering a basic understanding of co-expressed genes <span class="citation" data-cites="1703M7bru">[<a href="#ref-1703M7bru" role="doc-biblioref">59</a>]</span>.
However, these models often lack the ability to incorporate prior biological knowledge, which limits their interpretability and biological relevance.
To overcome these limitations, more sophisticated semi-supervised approaches such as the Pathway-Level Information Extractor (PLIER) <span class="citation" data-cites="Ki2ij7zE">[<a href="#ref-Ki2ij7zE" role="doc-biblioref">60</a>]</span>, or GenomicSuperSignature <span class="citation" data-cites="X4fhSCkz">[<a href="#ref-X4fhSCkz" role="doc-biblioref">61</a>]</span> have been developed.
These methodologies combine unsupervised approaches (decomposition methodologies) with supervised approaches to infer gene modules and annotate them with prior biological knowledge.</p>
<p>GenomicSuperSignature employs PCA to identify gene modules applying prior knowledge for biological annotation after module discovery.
This flexibility allows it to adapt to novel datasets and provides computational efficiency, even with large datasets.
However, the lack of prior knowledge integration during training can result in modules that are less biologically interpretable compared to those from PLIER-based methods <span class="citation" data-cites="X4fhSCkz">[<a href="#ref-X4fhSCkz" role="doc-biblioref">61</a>]</span>.</p>
<p>The Pathway-Level Information Extractor (PLIER) (<strong>Figure <a href="#fig:fig3">3</a>b</strong>) is a semi-supervised framework based on non-negative matrix factorization.
During training, it reduces high-dimensional gene expression data into a smaller set of latent variables (LVs) while aligning some of them to known pathway or gene set annotations.
These LVs are designed to capture the greatest variance in the data, where a fraction of the LVs represent known mechanisms (i.e., aligned with prior knowledge/pathways) while others capture potential technical artifacts or novel patterns.
By incorporating prior pathway/gene set information, PLIER generates interpretable latent representations.
Importantly, PLIER retains some gene modules that are not aligned with pathways, allowing researchers to distinguish LVs that represent either known patterns, technical artifacts or potential novel biological insights <span class="citation" data-cites="dg9nKuy0">[<a href="#ref-dg9nKuy0" role="doc-biblioref">50</a>]</span>.</p>
<p>PLIER has been used to integrate large-scale gene expression data using transfer learning across multiple cell types and conditions <span class="citation" data-cites="14rnBunuZ">[<a href="#ref-14rnBunuZ" role="doc-biblioref">62</a>]</span>.
By leveraging large datasets like recount2 <span class="citation" data-cites="GM90STke">[<a href="#ref-GM90STke" role="doc-biblioref">63</a>]</span>, it enhances the accuracy of inferred gene modules by identifying patterns consistent across diverse data.
This application addresses the limitation of small datasets, such as those for rare diseases, by training a PLIER model on a large public data compendium and applying it to smaller datasets, resulting in models that align well with biological factors.
This approach improves the interpretability and reliability of gene expression patterns, particularly in complex experimental contexts.
PLIER can be used in specific cohort contexts, focusing on dissecting traits within smaller, more homogenous datasets.
Nandi et al. (2024) demonstrated its utility on the Human Trisome Project (HTP) <span class="citation" data-cites="v5DtVejF">[<a href="#ref-v5DtVejF" role="doc-biblioref">64</a>/]</span>, identifying gene modules that mediate the effect of karyotype on body mass index in Down syndrome, highlighting key regulators such as <em>GPX1</em> and <em>MCL1</em>.
This allows it to capture condition-specific regulatory mechanisms, but at the expense of broader generalizability, making it a powerful tool for understanding unique biological processes within specialized groups.
PLIER, initially designed for human gene expression data, has been successfully adapted for use in other species, demonstrating its versatility across different biological contexts.
By applying the framework to model organisms, researchers can infer gene modules in species beyond humans, expanding the utility of PLIER for comparative studies and translational research.
One such adaptation is MousiPLIER, which tailors the PLIER framework for use in mouse models, a widely used system in biomedical research <span class="citation" data-cites="1CDQw6VDO">[<a href="#ref-1CDQw6VDO" role="doc-biblioref">65</a>]</span>.</p>
<h3 id="deep-learning-methods-autoencoders">Deep learning methods: autoencoders</h3>
<p>In contrast to linear methods, non-linear models such as deep learning (DL) approaches excel at capturing complex patterns in data.
This strength has been demonstrated in various fields, including predicting gene expression directly from sequence data <span class="citation" data-cites="1HNaN8NTb CdrNvVBn Gy0Zkruj">[<a href="#ref-1HNaN8NTb" role="doc-biblioref">66</a>,<a href="#ref-CdrNvVBn" role="doc-biblioref">67</a>,<a href="#ref-Gy0Zkruj" role="doc-biblioref">68</a>]</span>.</p>
<p>DL models are also highly effective in unsupervised settings such as inferring gene modules.
Specifically, DL methods like autoencoders <span class="citation" data-cites="NLVTJ9Lj">[<a href="#ref-NLVTJ9Lj" role="doc-biblioref">69</a>]</span>, which utilize neural networks to learn compressed representations, can capture both linear and nonlinear patterns between genes.
This makes autoencoders particularly suited for uncovering gene modules and efficiently handling high-dimensional datasets.
For instance, variational autoencoders (VAEs) (<strong>Figure <a href="#fig:fig3">3</a>b)</strong>) enhance this process by encoding data into a continuous latent space, enabling dimensionality reduction and the generation of new data.
However, this increased power to capture more general patterns has a major tradeoff: interpretability.
To overcome this, interpretable VAE models like the Pathway Module VAE (pmVAE) <span class="citation" data-cites="nGVsOd2S">[<a href="#ref-nGVsOd2S" role="doc-biblioref">70</a>]</span>, expiMap <span class="citation" data-cites="JkqcQgM7">[<a href="#ref-JkqcQgM7" role="doc-biblioref">71</a>]</span>, and VEGA <span class="citation" data-cites="170T6ip47">[<a href="#ref-170T6ip47" role="doc-biblioref">72</a>]</span> integrate biological pathway information into the decoder, which improves interpretability, thus making these approaches function as linear dimensionality reduction methods.
(<strong>Figure <a href="#fig:fig3">3</a>b)</strong>).</p>
<p>From now on, we refer to a gene module as a pattern extracted by any unsupervised approach (such as PLIER, GenomicSuperSignature, VAE, clustering, etc.) applied to gene expression data, which could be a latent variable or other types of gene cluster.</p>
<h2 id="a-gene-module-perspective-for-genetic-studies">A gene module perspective for genetic studies</h2>
<div id="fig:fig4" class="fignos">
<figure>
<img src="images/fig4.svg" style="width:100.0%" alt="Figure 4: An integrative, gene module-based approach for genetic studies. Each panel shows a component of the PhenoPLIER framework [73]. a) First, latent variables (LVs) or gene modules are learned from transcriptome data using the PLIER matrix factorization method. PLIER generates matrix \mathbf{Z}, which has gene weights for each module, and matrix \mathbf{B}, which has the samples in the latent space. b) Schematic of gene-trait associations from TWAS and gene-drug scores from LINCS L1000 being projected into the latent space for a joint analysis. c) Schematic of a gene module-based drug reporposing framework, where the projection of TWAS and LINCS L1000 data is used to compute a drug-disease score. d) Schematic of a regression model that tests whether genes that belong to a module (using a column of \mathbf{Z}) tend to be more strongly associated with a trait (using p-values from TWAS). e) (top) Example of a gene module identified as LV246 analyzed in [73]. DGAT2 and ACACA, found in a CRISPR screen to be linked to lipid metabolism and potential core genes, were among the top 15 genes in this module. Using metadata from the top samples for LV246 from \mathbf{B}, the module was found to be expressed mainly in adipose tissue, and liver and astrocyes were part of the top 10 cell types and tissues. Using the regression model depicted in d), this module was significantly associated with high cholesterol, tryglycerides, LDL cholesterol, cholesterol lowering medication, Alzheimer’s disease and dementia in the discovery cohort (UK Biobank), and high cholesterol and memory loss in the replication cohort (eMERGE). (bottom) The p-values and colocalization probability from gene-trait associations from TWAS for a subset of the top 1% genes in LV246 (y-axis) and traits significantly associated with LV246 (x-axis). It can be seen that DGAT2 and ACACA (in boldface), although strongly linked to lipid metabolism, are weakly associated with these lipid-relevant traits." />
<figcaption aria-hidden="true"><span>Figure 4:</span> <strong>An integrative, gene module-based approach for genetic studies.</strong>
Each panel shows a component of the PhenoPLIER framework <span class="citation" data-cites="NM3rHx1i">[<a href="#ref-NM3rHx1i" role="doc-biblioref">73</a>]</span>.
<strong>a)</strong> First, latent variables (LVs) or gene modules are learned from transcriptome data using the PLIER matrix factorization method.
PLIER generates matrix <span class="math inline">\(\mathbf{Z}\)</span>, which has gene weights for each module, and matrix <span class="math inline">\(\mathbf{B}\)</span>, which has the samples in the latent space.
<strong>b)</strong> Schematic of gene-trait associations from TWAS and gene-drug scores from LINCS L1000 being projected into the latent space for a joint analysis.
<strong>c)</strong> Schematic of a gene module-based drug reporposing framework, where the projection of TWAS and LINCS L1000 data is used to compute a drug-disease score.
<strong>d)</strong> Schematic of a regression model that tests whether genes that belong to a module (using a column of <span class="math inline">\(\mathbf{Z}\)</span>) tend to be more strongly associated with a trait (using <span class="math inline">\(p\)</span>-values from TWAS).
<strong>e)</strong> (top) Example of a gene module identified as LV246 analyzed in <span class="citation" data-cites="NM3rHx1i">[<a href="#ref-NM3rHx1i" role="doc-biblioref">73</a>]</span>.
<em>DGAT2</em> and <em>ACACA</em>, found in a CRISPR screen to be linked to lipid metabolism and potential core genes, were among the top 15 genes in this module.
Using metadata from the top samples for LV246 from <span class="math inline">\(\mathbf{B}\)</span>, the module was found to be expressed mainly in adipose tissue, and liver and astrocyes were part of the top 10 cell types and tissues.
Using the regression model depicted in d), this module was significantly associated with high cholesterol, tryglycerides, LDL cholesterol, cholesterol lowering medication, Alzheimer’s disease and dementia in the discovery cohort (UK Biobank), and high cholesterol and memory loss in the replication cohort (eMERGE).
(bottom) The <span class="math inline">\(p\)</span>-values and colocalization probability from gene-trait associations from TWAS for a subset of the top 1% genes in LV246 (<span class="math inline">\(y\)</span>-axis) and traits significantly associated with LV246 (<span class="math inline">\(x\)</span>-axis).
It can be seen that <em>DGAT2</em> and <em>ACACA</em> (in boldface), although strongly linked to lipid metabolism, are weakly associated with these lipid-relevant traits.</figcaption>
</figure>
</div>
<p>All the previously described approaches provide valuable and orthogonal information to better understand the molecular basis of human complex traits, including different molecular mechanisms through variant-trait, variant-gene, gene-trait and gene-gene information.
All of them have limitations and offer a partial view of the complex molecular interplay underlying pathophysiological processes.
In this section, we describe a gene module-based approach aimed at identifying patterns shared across these data sources.
The underlying hypothesis of this approach is that biologically meaningful, disease-relevant gene-gene links will be revealed when different yet incomplete data modalities are integrated.
In particular, we show examples on how the integration of gene modules with gene-trait associations and gene-drug data highlight core genes, disease-relevant molecular processes and drug’s mechanisms of action.</p>
<p>The PhenoPLIER framework <span class="citation" data-cites="NM3rHx1i">[<a href="#ref-NM3rHx1i" role="doc-biblioref">73</a>]</span> is a recent computational framework that implements this gene module-based approach.<strong>
PhenoPLIER can integrate gene modules extracted from any transcriptome data with gene-trait associations from TWAS and drug-induced transcriptional profiles (</strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span><strong>).
The approach integrates different molecular data sources by using a latent space derived from transcriptome data where each latent variable (LV) is a gene module (</strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>a)<strong>).
Then, different data modalities are integrated into this common latent space for a joint analysis, which was shown to detect important genes missed by standard methods, capture relevant trait clusters and predict drug-disease links more accurately than state-of-the-art drug repurposing approaches.
An advantage of PhenoPLIER is interpretability: it uses gene module models from PLIER (</strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>a)**), which provides not only information about which genes belong to a module but also what are the top samples where those genes are expressed.
If present, by looking at the metadata of the RNA-seq datasets, it is possible to infer if those samples represent consistent conditions such as tissues, cell types, or other more complex contexts.</p>
<p>In PhenoPLIER, this latent transcriptional space is described by 987 gene modules derived from recount2 <span class="citation" data-cites="6SPTvFXq 14rnBunuZ">[<a href="#ref-14rnBunuZ" role="doc-biblioref">62</a>,<a href="#ref-6SPTvFXq" role="doc-biblioref">74</a>]</span> (Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>a), a large and highly heterogeneous expression compendia.
Given the heterogeneity of the dataset, these gene modules can capture genes expressed across different contexts such as tissues, cell types (across differentiation stages), different disease states or stimuli.
These contexts can be extracted from sample metadata.
PhenoPLIER incorporates gene-trait associations computed with the PrediXcan family of TWAS methods <span class="citation" data-cites="Z8bvDdVq vLyTudUB 1FFzCXo1s">[<a href="#ref-Z8bvDdVq" role="doc-biblioref">24</a>,<a href="#ref-vLyTudUB" role="doc-biblioref">27</a>,<a href="#ref-1FFzCXo1s" role="doc-biblioref">28</a>]</span> on two different cohorts: the UK Biobank <span class="citation" data-cites="nmJxPpE5 lY5ln3dB">[<a href="#ref-nmJxPpE5" role="doc-biblioref">75</a>,<a href="#ref-lY5ln3dB" role="doc-biblioref">76</a>]</span> as discovery, and the Electronic Medical Records and Genomics (eMERGE) network phase III <span class="citation" data-cites="wfqjCerX gZAOkumx">[<a href="#ref-wfqjCerX" role="doc-biblioref">77</a>,<a href="#ref-gZAOkumx" role="doc-biblioref">78</a>]</span> as replication.
Gene-trait associations are then projected into the latent space (<strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>b)</strong>) where cluster analysis on traits was performed and expected and stable groupings of traits were detected, such as asthma and allergies, heel bone-densitometry measurements, hematological assays on red blood cells, physical measures, keratometry measurements, assays on white blood cells and platelets, skin and hair color traits, autoimmune disorders, and cardiovascular diseases.
The cardiovascular grouping also included other cardiovascular-related traits such as hand-grip strength <span class="citation" data-cites="doi:10.1016/S0140-6736">[<a href="#ref-doi:10.1016/S0140-6736" role="doc-biblioref"><strong>doi:10.1016/S0140-6736?</strong></a>]</span>, and environmental/behavioral factors such as physical activity and diet.
The PhenoPLIER approach also proposed a methodology based on interpretable classifiers to detect which gene modules are driving the different clusters.
For example, gene modules associated with the red blood cells cluster were 1) well-aligned to pathways related to early progenitors of the erythrocytes lineage, 2) predominantly expressed in early differentiation stages of erythropoiesis and 3) strongly associated with different assays on red blood cells from TWAS.
Other gene modules were associated with the keratometry measures grouping and expressed in corneal endothelial cells, and the grouping with autoimmune disorders was driven by gene modules expressed in T cells.
These results show that shared patterns exist between prior knowledge (pathways), gene expression, and gene-trait associations from TWAS.</p>
<p>In PhenoPLIER, information about gene-drug links is incorporated from transcriptional responses to small molecule perturbations profiled in LINCS L1000 <span class="citation" data-cites="F7lIlh2N">[<a href="#ref-F7lIlh2N" role="doc-biblioref">79</a>]</span>.
This information was integrated with gene-trait associations to build a gene module-based drug repurposing framework with the purpose of assessing how substituting LVs for single genes predicted known treatment-disease relationships better (<strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>c))</strong>).
Based on an established drug repurposing strategy that matches reversed transcriptome patterns between genes and drug-induced perturbations <span class="citation" data-cites="doi:10.1126/scitranslmed.300264 mZjkE1xU">[<a href="#ref-mZjkE1xU" role="doc-biblioref">80</a>,<a href="#ref-doi:10.1126/scitranslmed.300264" role="doc-biblioref"><strong>doi:10.1126/scitranslmed.300264?</strong></a>]</span>, in PhenoPLIER the authors adapted an existing drug repurposing framework <span class="citation" data-cites="17oeJ0CXy">[<a href="#ref-17oeJ0CXy" role="doc-biblioref">81</a>]</span> that uses single gene information from TWAS to gene modules.
Then, these two approaches, the single gene-based and gene module-based one, were compared using a manually-curated gold standard set of drug-disease medical indications <span class="citation" data-cites="O21tn8vf 10KA5jTBQ">[<a href="#ref-O21tn8vf" role="doc-biblioref">82</a>,<a href="#ref-10KA5jTBQ" role="doc-biblioref">83</a>]</span> for 322 drugs across 53 diseases to evaluate the prediction performance.
The gene module-based approach outperformed the single gene-based one with an area under the curve of 0.63 vs 0.57, and average precision of 0.86 vs 0.64.
Although the performance difference in this task was not large, the authors noted that the gene module-based approach represents a compressed version of the entire set of single gene-based results, and the higher performance implied that the low-dimensional latent space used (which necessarily misses some information) captured biologically meaningful gene-gene patterns.
Additionally, since gene modules represent interpretable features, the authors found that lipid-related gene modules expressed in adipose tissue and liver were among the top modules contributing to the prediction of high cholesterol and Nicotinic acid (Niacin, which can treat lipid disorders), potentially resembling known mechanisms of action of Niacin such as decreasing the production of low-density lipoproteins (LDL) either by modulating triglyceride synthesis in hepatocytes or by inhibiting adipocyte triglyceride lipolysis <span class="citation" data-cites="LVihFr3g">[<a href="#ref-LVihFr3g" role="doc-biblioref">84</a>]</span>.</p>
<p>The PhenoPLIER framework is also able to compute an association between a gene module and a trait by integrating single gene TWAS results (<strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>d)</strong>).
The association is computed using a regression model that tests whether genes that strongly belong to a module (using a column in matrix Z in (<strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>a)</strong>) are also strongly associated with the trait (using a row in the yellow TWAS matrix in <strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>b)</strong>.
For validation, the study conducted a CRISPR Cas9 screen in the HepG2 (liver) cell line to identify genes associated with lipid regulation, and found a high-confidence lipid-increasing gene set that included genes <em>DGAT2</em> and <em>ACACA</em>, which fit the definition of core genes.
The gene module identified as LV246 (<strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>e)</strong>) was the module most strongly enriched with this lipid-increasing gene set, containing <em>DGAT2</em> and <em>ACACA</em> among the top 15 genes most strongly connected with the module.
LV246 was found to be 1) aligned with lipid metabolism pathways, 2) expressed in adipose tissue, liver, astrocytes, among others, and 3) significantly associated with lipid-related traits in the discovery cohort (UK Biobank; <strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>e)</strong>) and replication cohort (eMERGE), such as high cholesterol, triglycerides, LDL cholesterol and cholesterol lowering medications in the UK Biobank, and high cholesterol (hyperlipidemia, phecode 272.11) in eMERGE.
Although a direct link between lipids and <em>DGAT2</em> and <em>ACACA</em> was found by the CRISPR screening, the authors noted that these genes lack strong TWAS associations with any of these lipid-related traits (<strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>e)</strong>, bottom).
The GWAS Catalog shows that genetic variants in <em>DGAT2</em> were found to be associated with cholesterol levels <span class="citation" data-cites="xJjWGVyN">[<a href="#ref-xJjWGVyN" role="doc-biblioref">85</a>]</span>, but <em>ACACA</em> lacks any association with these phenotypes <span class="citation" data-cites="1GM0PmkWn">[<a href="#ref-1GM0PmkWn" role="doc-biblioref">86</a>]</span>.
A potential explanation provided by the study says that, based on the omnigenic model, <em>DGAT2</em> and <em>ACACA</em> could represent core genes, and most of the other genes in LV246 might be peripheral genes that potentially <em>trans</em> regulate them.
This example suggests that strong GWAS hits could represent either peripheral or core genes, and that some core genes might be missed by GWAS alone.</p>
<p>Additionally, another interesting trait association in module LV246 was with Alzheimer’s disease (AD) and dementia in the UK Biobank, and memory loss (phecode 292.3) in eMERGE, which tends to worsen as AD progresses <span class="citation" data-cites="Vxi0ejah">[<a href="#ref-Vxi0ejah" role="doc-biblioref">87</a>]</span>.
The association with AD is relevant since this module was also found to be expressed to some degree in astrocytes, and has <em>APOE</em> as one of the top genes (<strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>e)</strong>, bottom).
<em>APOE</em>, the gene most strongly associated with late-onset AD <span class="citation" data-cites="hn3T60wf">[<a href="#ref-hn3T60wf" role="doc-biblioref">88</a>]</span>, encodes apolipoprotein E (ApoE), which is involved in lipid transport in the brain <span class="citation" data-cites="10ZgydXh1">[<a href="#ref-10ZgydXh1" role="doc-biblioref">89</a>]</span>, a mechanism that, when dysregulated, may play an important role in AD pathogenesis.
From TWAS alone, it can be seen that <em>APOE</em> is strongly associated and colocalized with the same traits as LV246 (<strong>Figure <span class="citation" data-cites="fig4">[<a href="#ref-fig4" role="doc-biblioref"><strong>fig4?</strong></a>]</span>e)</strong>, bottom).
However, a recent systematic survey across clinical trials that evaluate ApoE-targeted drugs for AD found modest to no efficacy in the treatment for this disease <span class="citation" data-cites="doi:10.1212/WNL.00000000002048">[<a href="#ref-doi:10.1212/WNL.00000000002048" role="doc-biblioref"><strong>doi:10.1212/WNL.00000000002048?</strong></a>]</span>.
Although there is strong evidence that <em>APOE</em> is a causal gene for AD, it still remains to be determined whether it represents a core gene or plays a more peripheral role.
A gene module-based approach might help prioritize core genes that remain elusive when using standard single-gene strategies.</p>
<p>Another approach to compute an association between gene modules and traits is MAGMA gene-set analysis <span class="citation" data-cites="19XiXgYmd">[<a href="#ref-19XiXgYmd" role="doc-biblioref">13</a>]</span>.
Actually, the regression model used by PhenoPLIER is based on MAGMA.
The difference is that PhenoPLIER uses gene-trait association from the PrediXcan family of TWAS methods, while MAGMA does not incorporate eQTL data and it uses a proximity-based approach in linking variants with genes.
Another difference is that MAGMA allows to perform conditional and interaction gene-set analysis to account for correlated gene modules, and the approach was found to be useful in detecting novel pathways in the context of blood pressure <span class="citation" data-cites="Om8ZhS06">[<a href="#ref-Om8ZhS06" role="doc-biblioref">90</a>]</span>.
The use of eQTL data by TWAS, however, might fit better the gene regulatory network assumptions of the omnigenic model than proximity-based approaches like MAGMA.</p>
<p>Approaches based on gene modules have the potential to go beyond prior knowledge and capture patterns that might be unique to different disease datasets.
First, although a gene module can be aligned to a pathway, this does not mean that it is restricted to prior knowledge as in standard pathway analyses.
Being aligned with pathways means that a module resembles a known mechanism, and this helps in interpretability and also in separating from other patterns that might be related to technical noise but the module could capture other genes as well that are not part of the pathway but potentially also involved the function.
Second, gene modules can also be extracted from very large, heterogeneous datasets such as recount2, or more specific ones such as the Human Trisome Project, which includes gene expression from people with Down Syndrome (DS).
Approaches such as PLIER and PhenoPLIER have been fundamental to extract DS-specific gene modules related to obesity, a common co-occurring condition The omni-PLIER framework, combining the PLIER model with the omnigenic approach, uncovers molecular mechanisms linking Down syndrome (DS) and obesity, identifying key latent variables with top genes such as GPX1 and MCL1, which mediate the relationship between trisomy 21 and body mass index (BMI), offering insights into DS-specific and general obesity-related processes. <code class="yellow">[CITE Sutanu’s paper]</code>.</p>
<p>A disadvantage of the PhenoPLIER approach is that gene modules are generated by an algorithm, and as such, they could represent artifacts or be aligned with technical noise.
Interpretable methods such as PLIER and some VAE models that help segregate technical noise from relevant biology are key in solving this problem.
Another limitation is the potential difficulty in determining the contexts in which a gene module is expressed.
In the example of LV256 and lipid-related traits, the module was also found to be expressed, with different degrees, in tumor samples, skin, skeletal muscle and other contexts that might be hard to interpret.
The interpretability advantage is also limited by the quality of the RNA-seq metadata.
Additionally, compared with GWAS and TWAS, where the unit is an objective molecular entity (a SNP or gene in a particular chromosome and position), a gene module not only might or might not exist, but the algorithm could fail to capture all gene modules relevant for a particular study, which could bias any downstream analysis.
Finally, a gene module, equivalent to a gene co-expression network, represents a set of correlated genes’ expression in some conditions, but it does not provide information about gene-gene links (it’s not directed) and could include many false positives <span class="citation" data-cites="jVQ2rMqc">[<a href="#ref-jVQ2rMqc" role="doc-biblioref">91</a>]</span>.
Future approaches should also incorporate other data modalities to refine a gene module such as transcription factor binding data or chromatin accessibility.</p>
<p>Although there are several limitations when we only correlate gene pairs across RNA-seq samples <span class="citation" data-cites="jVQ2rMqc">[<a href="#ref-jVQ2rMqc" role="doc-biblioref">91</a>]</span>, integrating gene networks with other data modalities have the potential to identify disease-relevant molecular mechanisms <span class="citation" data-cites="dg9nKuy0">[<a href="#ref-dg9nKuy0" role="doc-biblioref">50</a>]</span>.
Here we described an approach that integrates gene modules with other data sources such as gene-trait and gene-drug information.
The underlying hypothesis of this approach is that biologically meaningful gene-gene links will be captured when different data modalities that at least partially capture these links are integrated.
In our examples, these data modalities are gene modules from transcriptome data, single gene-trait associations and single gene-drug links via drug-induced transcriptional profiles.
We also showed that if the unsupervised approach used to extract a gene module is also interpretable (i.e., provides information about the specific contexts such as cell types or tissues that explain why those genes were grouped together), the data integration approach also sheds light on potential context-specific transcriptional mechanisms.</p>
<h2 class="page_break_before" id="future-perspectives">Future perspectives</h2>
<h2 class="page_break_before" id="conclusions">Conclusions</h2>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-VqGYQMXO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>Polygenic background modifies penetrance of monogenic variants for tier 1 genomic conditions</strong> <div class="csl-block">Akl C Fahed, Minxian Wang, Julian R Homburger, Aniruddh P Patel, Alexander G Bick, Cynthia L Neben, Carmen Lai, Deanna Brockman, Anthony Philippakis, Patrick T Ellinor, … Amit V Khera</div> <em>Nature Communications</em> (2020-08-20) <a href="https://doi.org/gg9hqv">https://doi.org/gg9hqv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-020-17374-3">10.1038/s41467-020-17374-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32820175">32820175</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7441381">PMC7441381</a></div></div>
</div>
<div id="ref-1873zCaaX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>Open problems in human trait genetics</strong> <div class="csl-block">Nadav Brandes, Omer Weissbrod, Michal Linial</div> <em>Genome Biology</em> (2022-06-20) <a href="https://doi.org/gqc962">https://doi.org/gqc962</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-022-02697-9">10.1186/s13059-022-02697-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35725481">35725481</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9208223">PMC9208223</a></div></div>
</div>
<div id="ref-VRCjjZ9L" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>The NHGRI-EBI GWAS Catalog: knowledgebase and deposition resource</strong> <div class="csl-block">Elliot Sollis, Abayomi Mosaku, Ala Abid, Annalisa Buniello, Maria Cerezo, Laurent Gil, Tudor Groza, Osman Güneş, Peggy Hall, James Hayhurst, … Laura W Harris</div> <em>Nucleic Acids Research</em> (2022-11-09) <a href="https://doi.org/gq73r5">https://doi.org/gq73r5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkac1010">10.1093/nar/gkac1010</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36350656">36350656</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9825413">PMC9825413</a></div></div>
</div>
<div id="ref-17B4AigZT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>Benefits and limitations of genome-wide association studies</strong> <div class="csl-block">Vivian Tam, Nikunj Patel, Michelle Turcotte, Yohan Bossé, Guillaume Paré, David Meyre</div> <em>Nature Reviews Genetics</em> (2019-05-08) <a href="https://doi.org/ggcxxb">https://doi.org/ggcxxb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41576-019-0127-1">10.1038/s41576-019-0127-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31068683">31068683</a></div></div>
</div>
<div id="ref-yqBcaXHD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>10 Years of GWAS Discovery: Biology, Function, and Translation</strong> <div class="csl-block">Peter M Visscher, Naomi R Wray, Qian Zhang, Pamela Sklar, Mark I McCarthy, Matthew A Brown, Jian Yang</div> <em>The American Journal of Human Genetics</em> (2017-07) <a href="https://doi.org/gcsmnm">https://doi.org/gcsmnm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ajhg.2017.06.005">10.1016/j.ajhg.2017.06.005</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28686856">28686856</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5501872">PMC5501872</a></div></div>
</div>
<div id="ref-sBTrhvlJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>Genome-wide analysis of insomnia in 1,331,010 individuals identifies new risk loci and functional pathways</strong><div class="csl-block">, Philip R Jansen, Kyoko Watanabe, Sven Stringer, Nathan Skene, Julien Bryois, Anke R Hammerschlag, Christiaan A de Leeuw, Jeroen S Benjamins, Ana B Muñoz-Manchado, … Danielle Posthuma</div> <em>Nature Genetics</em> (2019-02-25) <a href="https://doi.org/gfv4p2">https://doi.org/gfv4p2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-018-0333-3">10.1038/s41588-018-0333-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30804565">30804565</a></div></div>
</div>
<div id="ref-crCAymA1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>Genome-wide association analysis of insomnia complaints identifies risk genes and genetic overlap with psychiatric and metabolic traits</strong> <div class="csl-block">Anke R Hammerschlag, Sven Stringer, Christiaan A de Leeuw, Suzanne Sniekers, Erdogan Taskesen, Kyoko Watanabe, Tessa F Blanken, Kim Dekker, Bart HW te Lindert, Rick Wassing, … Danielle Posthuma</div> <em>Nature Genetics</em> (2017-06-12) <a href="https://doi.org/gbhzs5">https://doi.org/gbhzs5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.3888">10.1038/ng.3888</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28604731">28604731</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5600256">PMC5600256</a></div></div>
</div>
<div id="ref-s6z5ZuY3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>A genome-wide association scan of nonsynonymous SNPs identifies a susceptibility variant for Crohn disease in ATG16L1</strong> <div class="csl-block">Jochen Hampe, Andre Franke, Philip Rosenstiel, Andreas Till, Markus Teuber, Klaus Huse, Mario Albrecht, Gabriele Mayr, Francisco M De La Vega, Jason Briggs, … Stefan Schreiber</div> <em>Nature Genetics</em> (2006-12-31) <a href="https://doi.org/b9fr2w">https://doi.org/b9fr2w</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng1954">10.1038/ng1954</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17200669">17200669</a></div></div>
</div>
<div id="ref-gxwza7fO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>A Crohn’s disease variant in Atg16l1 enhances its degradation by caspase 3</strong> <div class="csl-block">Aditya Murthy, Yun Li, Ivan Peng, Mike Reichelt, Anand Kumar Katakam, Rajkumar Noubade, Merone Roose-Girma, Jason DeVoss, Lauri Diehl, Robert R Graham, Menno van Lookeren Campagne</div> <em>Nature</em> (2014-02) <a href="https://doi.org/f5r7wh">https://doi.org/f5r7wh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature13044">10.1038/nature13044</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24553140">24553140</a></div></div>
</div>
<div id="ref-HicyXvA4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>Multi-ancestry genome-wide association analyses identify novel genetic mechanisms in rheumatoid arthritis</strong> <div class="csl-block">Kazuyoshi Ishigaki, Saori Sakaue, Chikashi Terao, Yang Luo, Kyuto Sonehara, Kensuke Yamaguchi, Tiffany Amariuta, Chun Lai Too, Vincent A Laufer, Ian C Scott, … Soumya Raychaudhuri</div> <em>Nature Genetics</em> (2022-11) <a href="https://doi.org/gq8psq">https://doi.org/gq8psq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-022-01213-w">10.1038/s41588-022-01213-w</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36333501">36333501</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10165422">PMC10165422</a></div></div>
</div>
<div id="ref-IrGVpNji" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Molecular mechanisms of epistasis within and between genes</strong> <div class="csl-block">Ben Lehner</div> <em>Trends in Genetics</em> (2011-08) <a href="https://doi.org/fk3d5b">https://doi.org/fk3d5b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.tig.2011.05.007">10.1016/j.tig.2011.05.007</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21684621">21684621</a></div></div>
</div>
<div id="ref-BqjxxhyF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Demystifying non-coding GWAS variants: an overview of computational tools and methods</strong> <div class="csl-block">Marijn Schipper, Danielle Posthuma</div> <em>Human Molecular Genetics</em> (2022-08-16) <a href="https://doi.org/grqdp7">https://doi.org/grqdp7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/hmg/ddac198">10.1093/hmg/ddac198</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35972862">35972862</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9585674">PMC9585674</a></div></div>
</div>
<div id="ref-19XiXgYmd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>MAGMA: Generalized Gene-Set Analysis of GWAS Data</strong> <div class="csl-block">Christiaan A de Leeuw, Joris M Mooij, Tom Heskes, Danielle Posthuma</div> <em>PLOS Computational Biology</em> (2015-04-17) <a href="https://doi.org/gf92gp">https://doi.org/gf92gp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1004219">10.1371/journal.pcbi.1004219</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25885710">25885710</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4401657">PMC4401657</a></div></div>
</div>
<div id="ref-9Pr9idng" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>The GTEx Consortium atlas of genetic regulatory effects across human tissues</strong><div class="csl-block">, François Aguet, Shankara Anand, Kristin G Ardlie, Stacey Gabriel, Gad A Getz, Aaron Graubert, Kane Hadley, Robert E Handsaker, Katherine H Huang, … Simona Volpi</div> <em>Science</em> (2020-09-11) <a href="https://doi.org/ghbnhr">https://doi.org/ghbnhr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.aaz1776">10.1126/science.aaz1776</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32913098">32913098</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7737656">PMC7737656</a></div></div>
</div>
<div id="ref-OBxYzUMh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>Systematic Localization of Common Disease-Associated Variation in Regulatory DNA</strong> <div class="csl-block">Matthew T Maurano, Richard Humbert, Eric Rynes, Robert E Thurman, Eric Haugen, Hao Wang, Alex P Reynolds, Richard Sandstrom, Hongzhu Qu, Jennifer Brody, … John A Stamatoyannopoulos</div> <em>Science</em> (2012-09-07) <a href="https://doi.org/f36td3">https://doi.org/f36td3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.1222794">10.1126/science.1222794</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22955828">22955828</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3771521">PMC3771521</a></div></div>
</div>
<div id="ref-167QL5tMV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>Obesity-associated variants within FTO form long-range functional connections with IRX3</strong> <div class="csl-block">Scott Smemo, Juan J Tena, Kyoung-Han Kim, Eric R Gamazon, Noboru J Sakabe, Carlos Gómez-Marín, Ivy Aneas, Flavia L Credidio, Débora R Sobreira, Nora F Wasserman, … Marcelo A Nóbrega</div> <em>Nature</em> (2014-03-12) <a href="https://doi.org/f58b2g">https://doi.org/f58b2g</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature13138">10.1038/nature13138</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24646999">24646999</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4113484">PMC4113484</a></div></div>
</div>
<div id="ref-WfdUHVDA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>Disruption of a long-range cis-acting regulator for
&lt;i&gt;Shh&lt;/i&gt;
causes preaxial polydactyly</strong> <div class="csl-block">Laura A Lettice, Taizo Horikoshi, Simon JH Heaney, Marijke J van Baren, Herma C van der Linde, Guido J Breedveld, Marijke Joosse, Nurten Akarsu, Ben A Oostra, Naoto Endo, … Sumihare Noji</div> <em>Proceedings of the National Academy of Sciences</em> (2002-05-14) <a href="https://doi.org/bjrcnb">https://doi.org/bjrcnb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.112212199">10.1073/pnas.112212199</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/12032320">12032320</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC124279">PMC124279</a></div></div>
</div>
<div id="ref-QIEm5ZWK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>Strengthening Causal Inference for Complex Disease Using Molecular Quantitative Trait Loci</strong> <div class="csl-block">Sonja Neumeyer, Gibran Hemani, Eleftheria Zeggini</div> <em>Trends in Molecular Medicine</em> (2020-02) <a href="https://doi.org/ggmq94">https://doi.org/ggmq94</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.molmed.2019.10.004">10.1016/j.molmed.2019.10.004</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31718940">31718940</a></div></div>
</div>
<div id="ref-1FHKg3JK5" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>eQTL studies: from bulk tissues to single cells</strong> <div class="csl-block">Jingfei Zhang, Hongyu Zhao</div> <em>Journal of Genetics and Genomics</em> (2023-12) <a href="https://doi.org/g7n6d9">https://doi.org/g7n6d9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.jgg.2023.05.003">10.1016/j.jgg.2023.05.003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37207929">37207929</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10656365">PMC10656365</a></div></div>
</div>
<div id="ref-sGgkLsyy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>Large-scale cis- and trans-eQTL analyses identify thousands of genetic loci and polygenic scores that regulate blood gene expression</strong> <div class="csl-block">Urmo Võsa, Annique Claringbould, Harm-Jan Westra, Marc Jan Bonder, Patrick Deelen, Biao Zeng, Holger Kirsten, Ashis Saha, Roman Kreuzhuber, Seyhan Yazar, … Lude Franke</div> <em>Nature Genetics</em> (2021-09) <a href="https://doi.org/gmpj66">https://doi.org/gmpj66</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-021-00913-z">10.1038/s41588-021-00913-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34475573">34475573</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8432599">PMC8432599</a></div></div>
</div>
<div id="ref-ReOPt75u" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>From GWAS to Gene: Transcriptome-Wide Association Studies and Other Methods to Functionally Understand GWAS Discoveries</strong> <div class="csl-block">Binglan Li, Marylyn D Ritchie</div> <em>Frontiers in Genetics</em> (2021-09-30) <a href="https://doi.org/g7n6kb">https://doi.org/g7n6kb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fgene.2021.713230">10.3389/fgene.2021.713230</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34659337">34659337</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8515949">PMC8515949</a></div></div>
</div>
<div id="ref-l6ogswV3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>Opportunities and challenges for transcriptome-wide association studies</strong> <div class="csl-block">Michael Wainberg, Nasa Sinnott-Armstrong, Nicholas Mancuso, Alvaro N Barbeira, David A Knowles, David Golan, Raili Ermel, Arno Ruusalepp, Thomas Quertermous, Ke Hao, … Anshul Kundaje</div> <em>Nature Genetics</em> (2019-03-29) <a href="https://doi.org/gf3hmr">https://doi.org/gf3hmr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-019-0385-z">10.1038/s41588-019-0385-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30926968">30926968</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6777347">PMC6777347</a></div></div>
</div>
<div id="ref-RmhTH35O" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>Transcriptome-wide association studies: recent advances in methods, applications and available databases</strong> <div class="csl-block">Jialin Mai, Mingming Lu, Qianwen Gao, Jingyao Zeng, Jingfa Xiao</div> <em>Communications Biology</em> (2023-09-01) <a href="https://doi.org/gs9f8r">https://doi.org/gs9f8r</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s42003-023-05279-y">10.1038/s42003-023-05279-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37658226">37658226</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10474133">PMC10474133</a></div></div>
</div>
<div id="ref-Z8bvDdVq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>A gene-based association method for mapping traits using reference transcriptome data</strong> <div class="csl-block">Eric R Gamazon, Heather E Wheeler, Kaanan P Shah, Sahar V Mozaffari, Keston Aquino-Michaels, Robert J Carroll, Anne E Eyler, Joshua C Denny, Dan L Nicolae, … Hae Kyung Im</div> <em>Nature Genetics</em> (2015-08-10) <a href="https://doi.org/f7p9zv">https://doi.org/f7p9zv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.3367">10.1038/ng.3367</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26258848">26258848</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4552594">PMC4552594</a></div></div>
</div>
<div id="ref-1D63fEEPb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>Integrative approaches for large-scale transcriptome-wide association studies</strong> <div class="csl-block">Alexander Gusev, Arthur Ko, Huwenbo Shi, Gaurav Bhatia, Wonil Chung, Brenda WJH Penninx, Rick Jansen, Eco JC de Geus, Dorret I Boomsma, Fred A Wright, … Bogdan Pasaniuc</div> <em>Nature Genetics</em> (2016-02-08) <a href="https://doi.org/f3vf4p">https://doi.org/f3vf4p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.3506">10.1038/ng.3506</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26854917">26854917</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4767558">PMC4767558</a></div></div>
</div>
<div id="ref-19Klmizj8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>TIGAR: An Improved Bayesian Tool for Transcriptomic Data Imputation Enhances Gene Mapping of Complex Traits</strong> <div class="csl-block">Sini Nagpal, Xiaoran Meng, Michael P Epstein, Lam C Tsoi, Matthew Patrick, Greg Gibson, Philip L De Jager, David A Bennett, Aliza P Wingo, Thomas S Wingo, Jingjing Yang</div> <em>The American Journal of Human Genetics</em> (2019-08) <a href="https://doi.org/g7n6dv">https://doi.org/g7n6dv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ajhg.2019.05.018">10.1016/j.ajhg.2019.05.018</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31230719">31230719</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6698804">PMC6698804</a></div></div>
</div>
<div id="ref-vLyTudUB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>Exploring the phenotypic consequences of tissue specific gene expression variation inferred from GWAS summary statistics</strong> <div class="csl-block">Alvaro N Barbeira, Scott P Dickinson, Rodrigo Bonazzola, Jiamao Zheng, Heather E Wheeler, Jason M Torres, Eric S Torstenson, Kaanan P Shah, Tzintzuni Garcia, Todd L Edwards, … Hae Kyung Im</div> <em>Nature Communications</em> (2018-05-08) <a href="https://doi.org/gdjvp5">https://doi.org/gdjvp5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-018-03621-1">10.1038/s41467-018-03621-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29739930">29739930</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5940825">PMC5940825</a></div></div>
</div>
<div id="ref-1FFzCXo1s" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>Integrating predicted transcriptome from multiple tissues improves association detection</strong> <div class="csl-block">Alvaro N Barbeira, Milton Pividori, Jiamao Zheng, Heather E Wheeler, Dan L Nicolae, Hae Kyung Im</div> <em>PLOS Genetics</em> (2019-01-22) <a href="https://doi.org/ghs8vx">https://doi.org/ghs8vx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pgen.1007889">10.1371/journal.pgen.1007889</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30668570">30668570</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6358100">PMC6358100</a></div></div>
</div>
<div id="ref-93R9hBin" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>A statistical framework for cross-tissue transcriptome-wide association analysis</strong><div class="csl-block">, Yiming Hu, Mo Li, Qiongshi Lu, Haoyi Weng, Jiawei Wang, Seyedeh M Zekavat, Zhaolong Yu, Boyang Li, Jianlei Gu, … Hongyu Zhao</div> <em>Nature Genetics</em> (2019-02-25) <a href="https://doi.org/gjjt5f">https://doi.org/gjjt5f</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-019-0345-7">10.1038/s41588-019-0345-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30804563">30804563</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6788740">PMC6788740</a></div></div>
</div>
<div id="ref-1Dsj6bEd3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>PWAS: proteome-wide association study—linking genes and phenotypes by functional variation in proteins</strong> <div class="csl-block">Nadav Brandes, Nathan Linial, Michal Linial</div> <em>Genome Biology</em> (2020-07-14) <a href="https://doi.org/gnvtr9">https://doi.org/gnvtr9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-020-02089-x">10.1186/s13059-020-02089-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32665031">32665031</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7386203">PMC7386203</a></div></div>
</div>
<div id="ref-TPqvAJda" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>Epigenome-wide association studies: current knowledge, strategies and recommendations</strong> <div class="csl-block">Maria Pia Campagna, Alexandre Xavier, Jeannette Lechner-Scott, Vicky Maltby, Rodney J Scott, Helmut Butzkueven, Vilija G Jokubaitis, Rodney A Lea</div> <em>Clinical Epigenetics</em> (2021-12) <a href="https://doi.org/gn3kds">https://doi.org/gn3kds</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13148-021-01200-8">10.1186/s13148-021-01200-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34863305">34863305</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8645110">PMC8645110</a></div></div>
</div>
<div id="ref-1DNxNkLsT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>DNA methylome-wide association study of genetic risk for depression implicates antigen processing and immune responses</strong> <div class="csl-block">Xueyi Shen, Doretta Caramaschi, Mark J Adams, Rosie M Walker, Josine L Min, Alex Kwong, Gibran Hemani, Miruna C Barbu, Heather C Whalley, … Andrew M McIntosh</div> <em>Genome Medicine</em> (2022-03-31) <a href="https://doi.org/g7n6jn">https://doi.org/g7n6jn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13073-022-01039-5">10.1186/s13073-022-01039-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35354486">35354486</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8969265">PMC8969265</a></div></div>
</div>
<div id="ref-1BYHeZlgB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>Histone acetylome-wide associations in immune cells from individuals with active Mycobacterium tuberculosis infection</strong> <div class="csl-block">Ricardo CH del Rosario, Jeremie Poschmann, Carey Lim, Catherine Y Cheng, Pavanish Kumar, Catherine Riou, Seow Theng Ong, Sherif Gerges, Hajira Shreen Hajan, Dilip Kumar, … Shyam Prabhakar</div> <em>Nature Microbiology</em> (2022-01-31) <a href="https://doi.org/gpmp47">https://doi.org/gpmp47</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41564-021-01049-w">10.1038/s41564-021-01049-w</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35102304">35102304</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9439955">PMC9439955</a></div></div>
</div>
<div id="ref-FeaSnwhX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>Histone Variants: Guardians of Genome Integrity</strong> <div class="csl-block">Juliette Ferrand, Beatrice Rondinelli, Sophie E Polo</div> <em>Cells</em> (2020-11-05) <a href="https://doi.org/g6vwfw">https://doi.org/g6vwfw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3390/cells9112424">10.3390/cells9112424</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33167489">33167489</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7694513">PMC7694513</a></div></div>
</div>
<div id="ref-KfWCLspO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>Epigenetics: The DNA Methylation Profile of Tissue-Dependent and Differentially Methylated Regions in Cells</strong> <div class="csl-block">J Ohgane, S Yagi, K Shiota</div> <em>Placenta</em> (2008-03) <a href="https://doi.org/dfph4p">https://doi.org/dfph4p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.placenta.2007.09.011">10.1016/j.placenta.2007.09.011</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18031808">18031808</a></div></div>
</div>
<div id="ref-vpIDZCSa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>An Expanded View of Complex Traits: From Polygenic to Omnigenic</strong> <div class="csl-block">Evan A Boyle, Yang I Li, Jonathan K Pritchard</div> <em>Cell</em> (2017-06) <a href="https://doi.org/gcpgdz">https://doi.org/gcpgdz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2017.05.038">10.1016/j.cell.2017.05.038</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28622505">28622505</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5536862">PMC5536862</a></div></div>
</div>
<div id="ref-15ldVppuv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>Polygenic inheritance, GWAS, polygenic risk scores, and the search for functional variants</strong> <div class="csl-block">Daniel JM Crouch, Walter F Bodmer</div> <em>Proceedings of the National Academy of Sciences</em> (2020-08-04) <a href="https://doi.org/gqh7nq">https://doi.org/gqh7nq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.2005634117">10.1073/pnas.2005634117</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32753378">32753378</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7431089">PMC7431089</a></div></div>
</div>
<div id="ref-Qxk70FFV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><strong>XV.—The Correlation between Relatives on the Supposition of Mendelian Inheritance.</strong> <div class="csl-block">RA Fisher</div> <em>Transactions of the Royal Society of Edinburgh</em> (1919) <a href="https://doi.org/gf2bfm">https://doi.org/gf2bfm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1017/s0080456800012163">10.1017/s0080456800012163</a></div></div>
</div>
<div id="ref-RASeYPIy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><strong>Finding the missing heritability of complex diseases</strong> <div class="csl-block">Teri A Manolio, Francis S Collins, Nancy J Cox, David B Goldstein, Lucia A Hindorff, David J Hunter, Mark I McCarthy, Erin M Ramos, Lon R Cardon, Aravinda Chakravarti, … Peter M Visscher</div> <em>Nature</em> (2009-10) <a href="https://doi.org/dt3">https://doi.org/dt3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature08494">10.1038/nature08494</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19812666">19812666</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2831613">PMC2831613</a></div></div>
</div>
<div id="ref-1AOy1zxAv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>Solving the missing heritability problem</strong> <div class="csl-block">Alexander I Young</div> <em>PLOS Genetics</em> (2019-06-24) <a href="https://doi.org/ggnpnc">https://doi.org/ggnpnc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pgen.1008222">10.1371/journal.pgen.1008222</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31233496">31233496</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6611648">PMC6611648</a></div></div>
</div>
<div id="ref-mwTa2RUK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline"><strong>Common polygenic variation contributes to risk of schizophrenia and bipolar disorder</strong> <em>Nature</em> (2009-07-01) <a href="https://doi.org/cb5f2j">https://doi.org/cb5f2j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature08185">10.1038/nature08185</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19571811">19571811</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3912837">PMC3912837</a></div></div>
</div>
<div id="ref-lVJVFaaZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline"><strong>Toward Identification of Functional Sequences and Variants in Noncoding DNA</strong> <div class="csl-block">Remo Monti, Uwe Ohler</div> <em>Annual Review of Biomedical Data Science</em> (2023-08-10) <a href="https://doi.org/g7p7v4">https://doi.org/g7p7v4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1146/annurev-biodatasci-122120-110102">10.1146/annurev-biodatasci-122120-110102</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37262323">37262323</a></div></div>
</div>
<div id="ref-XvQe1H3A" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline"><strong>Contrasting genetic architectures of schizophrenia and other complex diseases using fast variance-components analysis</strong><div class="csl-block">, Po-Ru Loh, Gaurav Bhatia, Alexander Gusev, Hilary K Finucane, Brendan K Bulik-Sullivan, Samuela J Pollack, Teresa R de Candia, Sang Hong Lee, Naomi R Wray, … Alkes L Price</div> <em>Nature Genetics</em> (2015-11-02) <a href="https://doi.org/f3t9dq">https://doi.org/f3t9dq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.3431">10.1038/ng.3431</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26523775">26523775</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4666835">PMC4666835</a></div></div>
</div>
<div id="ref-LXvTZzEA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline"><strong>Trans Effects on Gene Expression Can Drive Omnigenic Inheritance</strong> <div class="csl-block">Xuanyao Liu, Yang I Li, Jonathan K Pritchard</div> <em>Cell</em> (2019-05) <a href="https://doi.org/gfz8bj">https://doi.org/gfz8bj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2019.04.014">10.1016/j.cell.2019.04.014</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31051098">31051098</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6553491">PMC6553491</a></div></div>
</div>
<div id="ref-fV6THoBG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline"><strong>The role of polygenic risk score gene-set analysis in the context of the omnigenic model of schizophrenia</strong><div class="csl-block">, Alexandros Rammos, Lara ANeira Gonzalez, Daniel R Weinberger, Kevin J Mitchell, Kristin K Nicodemus</div> <em>Neuropsychopharmacology</em> (2019-05-11) <a href="https://doi.org/g7p7t3">https://doi.org/g7p7t3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41386-019-0410-z">10.1038/s41386-019-0410-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31078131">31078131</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6785707">PMC6785707</a></div></div>
</div>
<div id="ref-13VkGrWQR" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline"><strong>Non-Coding RNAs Extended Omnigenic Module of Cancers</strong> <div class="csl-block">Jie Li, Bingbo Wang, Xiujuan Ma</div> <em>Entropy</em> (2024-07-27) <a href="https://doi.org/g7p7wq">https://doi.org/g7p7wq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3390/e26080640">10.3390/e26080640</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/39202109">39202109</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11353529">PMC11353529</a></div></div>
</div>
<div id="ref-EgquVUSg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline"><strong>Gene expression predictions and networks in natural populations supports the omnigenic theory</strong> <div class="csl-block">Aurélien Chateigner, Marie-Claude Lesage-Descauses, Odile Rogier, Véronique Jorge, Jean-Charles Leplé, Véronique Brunaud, Christine Paysant-Le Roux, Ludivine Soubigou-Taconnat, Marie-Laure Martin-Magniette, Leopoldo Sanchez, Vincent Segura</div> <em>BMC Genomics</em> (2020-06-22) <a href="https://doi.org/ghqvzz">https://doi.org/ghqvzz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12864-020-06809-2">10.1186/s12864-020-06809-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32571208">32571208</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7310122">PMC7310122</a></div></div>
</div>
<div id="ref-82fs3SZo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline"><strong>Common Disease Is More Complex Than Implied by the Core Gene Omnigenic Model</strong> <div class="csl-block">Naomi R Wray, Cisca Wijmenga, Patrick F Sullivan, Jian Yang, Peter M Visscher</div> <em>Cell</em> (2018-06) <a href="https://doi.org/gdqcd4">https://doi.org/gdqcd4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2018.05.051">10.1016/j.cell.2018.05.051</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29906445">29906445</a></div></div>
</div>
<div id="ref-HFRaLwGQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline"><strong>A saturated map of common genetic variants associated with human height</strong> <div class="csl-block">Loïc Yengo, Sailaja Vedantam, Eirini Marouli, Julia Sidorenko, Eric Bartell, Saori Sakaue, Marielisa Graff, Anders U Eliasen, Yunxuan Jiang, Sridharan Raghavan, … Joel N Hirschhorn</div> <em>Nature</em> (2022-10-12) <a href="https://doi.org/gqz8jh">https://doi.org/gqz8jh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41586-022-05275-y">10.1038/s41586-022-05275-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36224396">36224396</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9605867">PMC9605867</a></div></div>
</div>
<div id="ref-dg9nKuy0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline"><strong>The great hairball gambit</strong> <div class="csl-block">Jonathan Flint, Trey Ideker</div> <em>PLOS Genetics</em> (2019-11-26) <a href="https://doi.org/g7p7wc">https://doi.org/g7p7wc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pgen.1008519">10.1371/journal.pgen.1008519</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31770365">31770365</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6910700">PMC6910700</a></div></div>
</div>
<div id="ref-1GlQtJ9Yz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline"><strong>Quantitative omnigenic model discovers interpretable genome-wide associations</strong> <div class="csl-block">Natália Ružičková, Michal Hledík, Gašper Tkačik</div> <em>Cold Spring Harbor Laboratory</em> (2024-02-01) <a href="https://doi.org/g7x6m2">https://doi.org/g7x6m2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2024.02.01.578486">10.1101/2024.02.01.578486</a></div></div>
</div>
<div id="ref-Zm8nV5AC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline"><strong>High-throughput transcriptomics</strong> <div class="csl-block">Nunzio D’Agostino, Wenli Li, Dapeng Wang</div> <em>Scientific Reports</em> (2022-11-29) <a href="https://doi.org/g7p7vf">https://doi.org/g7p7vf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41598-022-23985-1">10.1038/s41598-022-23985-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36446824">36446824</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9708670">PMC9708670</a></div></div>
</div>
<div id="ref-SD886fuE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline"><strong>Gene module level analysis: identification to networks and dynamics</strong> <div class="csl-block">Xuewei Wang, Ertugrul Dalkic, Ming Wu, Christina Chan</div> <em>Current Opinion in Biotechnology</em> (2008-10) <a href="https://doi.org/fprfkv">https://doi.org/fprfkv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.copbio.2008.07.011">10.1016/j.copbio.2008.07.011</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18725293">18725293</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2615490">PMC2615490</a></div></div>
</div>
<div id="ref-QUI8V4RY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline"><strong>Emergence of co-expression in gene regulatory networks</strong> <div class="csl-block">Wencheng Yin, Luis Mendoza, Jimena Monzon-Sandoval, Araxi O Urrutia, Humberto Gutierrez</div> <em>PLOS ONE</em> (2021-04-01) <a href="https://doi.org/g7p7wf">https://doi.org/g7p7wf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pone.0247671">10.1371/journal.pone.0247671</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33793561">33793561</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8016302">PMC8016302</a></div></div>
</div>
<div id="ref-K4KeL4gw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline"><strong>Gene regulatory network reconstruction: harnessing the power of single-cell multi-omic data</strong> <div class="csl-block">Daniel Kim, Andy Tran, Hani Jieun Kim, Yingxin Lin, Jean Yee Hwa Yang, Pengyi Yang</div> <em>npj Systems Biology and Applications</em> (2023-10-19) <a href="https://doi.org/gt65pb">https://doi.org/gt65pb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41540-023-00312-6">10.1038/s41540-023-00312-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37857632">37857632</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10587078">PMC10587078</a></div></div>
</div>
<div id="ref-Ko8gMp8P" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline"><strong>The future is unsupervised</strong> <div class="csl-block">Casey S Greene</div> <em>Science Translational Medicine</em> (2016-07-06) <a href="https://doi.org/gg7qr3">https://doi.org/gg7qr3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/scitranslmed.aag3101">10.1126/scitranslmed.aag3101</a></div></div>
</div>
<div id="ref-1BVbSrr6M" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline"><strong>A comprehensive evaluation of module detection methods for gene expression data</strong> <div class="csl-block">Wouter Saelens, Robrecht Cannoodt, Yvan Saeys</div> <em>Nature Communications</em> (2018-03-15) <a href="https://doi.org/gc9x36">https://doi.org/gc9x36</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-018-03424-4">10.1038/s41467-018-03424-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29545622">29545622</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5854612">PMC5854612</a></div></div>
</div>
<div id="ref-YgLIM2bZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline"><strong>An efficient, not-only-linear correlation coefficient based on clustering</strong> <div class="csl-block">Milton Pividori, Marylyn D Ritchie, Diego H Milone, Casey S Greene</div> <em>Cell Systems</em> (2024-09) <a href="https://doi.org/nhs4">https://doi.org/nhs4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cels.2024.08.005">10.1016/j.cels.2024.08.005</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/39243756">39243756</a></div></div>
</div>
<div id="ref-1703M7bru" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline"><strong>Module representatives for refining gene co-expression modules</strong> <div class="csl-block">Nathan Mankovich, Helene Andrews-Polymenis, David Threadgill, Michael Kirby</div> <em>Physical Biology</em> (2023-05-04) <a href="https://doi.org/g7p7vk">https://doi.org/g7p7vk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1088/1478-3975/acce8d">10.1088/1478-3975/acce8d</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37075776">37075776</a></div></div>
</div>
<div id="ref-Ki2ij7zE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline"><strong>Pathway-level information extractor (PLIER) for gene expression data</strong> <div class="csl-block">Weiguang Mao, Elena Zaslavsky, Boris M Hartmann, Stuart C Sealfon, Maria Chikina</div> <em>Nature Methods</em> (2019-06-27) <a href="https://doi.org/gf75g6">https://doi.org/gf75g6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41592-019-0456-1">10.1038/s41592-019-0456-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31249421">31249421</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7262669">PMC7262669</a></div></div>
</div>
<div id="ref-X4fhSCkz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline"><strong>GenomicSuperSignature facilitates interpretation of RNA-seq experiments through robust, efficient comparison to public databases</strong> <div class="csl-block">Sehyun Oh, Ludwig Geistlinger, Marcel Ramos, Daniel Blankenberg, Marius van den Beek, Jaclyn N Taroni, Vincent J Carey, Casey S Greene, Levi Waldron, Sean Davis</div> <em>Nature Communications</em> (2022-06-27) <a href="https://doi.org/gqd7hm">https://doi.org/gqd7hm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-022-31411-3">10.1038/s41467-022-31411-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35760813">35760813</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9237024">PMC9237024</a></div></div>
</div>
<div id="ref-14rnBunuZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline"><strong>MultiPLIER: A Transfer Learning Framework for Transcriptomics Reveals Systemic Features of Rare Disease</strong> <div class="csl-block">Jaclyn N Taroni, Peter C Grayson, Qiwen Hu, Sean Eddy, Matthias Kretzler, Peter A Merkel, Casey S Greene</div> <em>Cell Systems</em> (2019-05) <a href="https://doi.org/gf75g5">https://doi.org/gf75g5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cels.2019.04.003">10.1016/j.cels.2019.04.003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31121115">31121115</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6538307">PMC6538307</a></div></div>
</div>
<div id="ref-GM90STke" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline"><strong>Reproducible RNA-seq analysis using recount2</strong> <div class="csl-block">Leonardo Collado-Torres, Abhinav Nellore, Kai Kammers, Shannon E Ellis, Margaret A Taub, Kasper D Hansen, Andrew E Jaffe, Ben Langmead, Jeffrey T Leek</div> <em>Nature Biotechnology</em> (2017-04) <a href="https://doi.org/gf75hp">https://doi.org/gf75hp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nbt.3838">https://doi.org/10.1038/nbt.3838</a></div></div>
</div>
<div id="ref-v5DtVejF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline"><strong>Crnic Institute Human Trisome Project™</strong> <div class="csl-block">Crnic Institute Human Trisome Project™</div> <a href="https://www.trisome.org">https://www.trisome.org</a></div>
</div>
<div id="ref-1CDQw6VDO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline"><strong>MousiPLIER: A Mouse Pathway-Level Information Extractor Model</strong> <div class="csl-block">Shuo Zhang, Benjamin J Heil, Weiguang Mao, Maria Chikina, Casey S Greene, Elizabeth A Heller</div> <em>eneuro</em> (2024-05-24) <a href="https://doi.org/g7p7wj">https://doi.org/g7p7wj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1523/eneuro.0313-23.2024">10.1523/eneuro.0313-23.2024</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/38789274">38789274</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11154669">PMC11154669</a></div></div>
</div>
<div id="ref-1HNaN8NTb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline"><strong>Deep learning sequence-based ab initio prediction of variant effects on expression and disease risk</strong> <div class="csl-block">Jian Zhou, Chandra L Theesfeld, Kevin Yao, Kathleen M Chen, Aaron K Wong, Olga G Troyanskaya</div> <em>Nature Genetics</em> (2018-07-16) <a href="https://doi.org/gdvmqw">https://doi.org/gdvmqw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-018-0160-6">10.1038/s41588-018-0160-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30013180">30013180</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6094955">PMC6094955</a></div></div>
</div>
<div id="ref-CdrNvVBn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline"><strong>Predicting mRNA Abundance Directly from Genomic Sequence Using Deep Convolutional Neural Networks</strong> <div class="csl-block">Vikram Agarwal, Jay Shendure</div> <em>Cell Reports</em> (2020-05) <a href="https://doi.org/ggw7fr">https://doi.org/ggw7fr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.celrep.2020.107663">10.1016/j.celrep.2020.107663</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32433972">32433972</a></div></div>
</div>
<div id="ref-Gy0Zkruj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline"><strong>Atlas of primary cell-type-specific sequence models of gene expression and variant effects</strong> <div class="csl-block">Ksenia Sokolova, Chandra L Theesfeld, Aaron K Wong, Zijun Zhang, Kara Dolinski, Olga G Troyanskaya</div> <em>Cell Reports Methods</em> (2023-09) <a href="https://doi.org/gt6n4t">https://doi.org/gt6n4t</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.crmeth.2023.100580">10.1016/j.crmeth.2023.100580</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37703883">37703883</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10545936">PMC10545936</a></div></div>
</div>
<div id="ref-NLVTJ9Lj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline"><strong>Auto-Encoding Variational Bayes</strong> <div class="csl-block">Diederik P Kingma, Max Welling</div> <em>arXiv</em> (2022-12-13) <a href="https://arxiv.org/abs/1312.6114">https://arxiv.org/abs/1312.6114</a></div>
</div>
<div id="ref-nGVsOd2S" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline"><strong>pmVAE: Learning Interpretable Single-Cell Representations with Pathway Modules</strong> <div class="csl-block">Gilles Gut, Stefan G Stark, Gunnar Rätsch, Natalie R Davidson</div> <em>Cold Spring Harbor Laboratory</em> (2021-01-30) <a href="https://doi.org/g7p7vq">https://doi.org/g7p7vq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2021.01.28.428664">10.1101/2021.01.28.428664</a></div></div>
</div>
<div id="ref-JkqcQgM7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">71. </div><div class="csl-right-inline"><strong>Biologically informed deep learning to query gene programs in single-cell atlases</strong> <div class="csl-block">Mohammad Lotfollahi, Sergei Rybakov, Karin Hrovatin, Soroor Hediyeh-zadeh, Carlos Talavera-López, Alexander V Misharin, Fabian J Theis</div> <em>Nature Cell Biology</em> (2023-02-02) <a href="https://doi.org/gtb96q">https://doi.org/gtb96q</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41556-022-01072-x">10.1038/s41556-022-01072-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36732632">36732632</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9928587">PMC9928587</a></div></div>
</div>
<div id="ref-170T6ip47" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">72. </div><div class="csl-right-inline"><strong>VEGA is an interpretable generative model for inferring biological network activity in single-cell transcriptomics</strong> <div class="csl-block">Lucas Seninge, Ioannis Anastopoulos, Hongxu Ding, Joshua Stuart</div> <em>Nature Communications</em> (2021-09-28) <a href="https://doi.org/gm77mc">https://doi.org/gm77mc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-021-26017-0">10.1038/s41467-021-26017-0</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34584103">34584103</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8478947">PMC8478947</a></div></div>
</div>
<div id="ref-NM3rHx1i" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">73. </div><div class="csl-right-inline"><strong>Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms</strong> <div class="csl-block">Milton Pividori, Sumei Lu, Binglan Li, Chun Su, Matthew E Johnson, Wei-Qi Wei, Qiping Feng, Bahram Namjou, Krzysztof Kiryluk, Iftikhar J Kullo, … Casey S Greene</div> <em>Nature Communications</em> (2023-09-09) <a href="https://doi.org/gspsxr">https://doi.org/gspsxr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-023-41057-4">10.1038/s41467-023-41057-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37689782">37689782</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10492839">PMC10492839</a></div></div>
</div>
<div id="ref-6SPTvFXq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline"><strong>Reproducible RNA-seq analysis using recount2</strong> <div class="csl-block">Leonardo Collado-Torres, Abhinav Nellore, Kai Kammers, Shannon E Ellis, Margaret A Taub, Kasper D Hansen, Andrew E Jaffe, Ben Langmead, Jeffrey T Leek</div> <em>Nature Biotechnology</em> (2017-04) <a href="https://doi.org/gf75hp">https://doi.org/gf75hp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nbt.3838">10.1038/nbt.3838</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28398307">28398307</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6742427">PMC6742427</a></div></div>
</div>
<div id="ref-nmJxPpE5" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">75. </div><div class="csl-right-inline"><strong>The UK Biobank resource with deep phenotyping and genomic data</strong> <div class="csl-block">Clare Bycroft, Colin Freeman, Desislava Petkova, Gavin Band, Lloyd T Elliott, Kevin Sharp, Allan Motyer, Damjan Vukcevic, Olivier Delaneau, Jared O’Connell, … Jonathan Marchini</div> <em>Nature</em> (2018-10) <a href="https://doi.org/gfb7h2">https://doi.org/gfb7h2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41586-018-0579-z">10.1038/s41586-018-0579-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30305743">30305743</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6786975">PMC6786975</a></div></div>
</div>
<div id="ref-lY5ln3dB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">76. </div><div class="csl-right-inline"><strong>PhenomeXcan: Mapping the genome to the phenome through the transcriptome</strong> <div class="csl-block">Milton Pividori, Padma S Rajagopal, Alvaro Barbeira, Yanyu Liang, Owen Melia, Lisa Bastarache, YoSon Park, GTEx Consortium, Xiaoquan Wen, Hae K Im</div> <em>Science Advances</em> (2020-09-11) <a href="https://doi.org/ghbvbf">https://doi.org/ghbvbf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/sciadv.aba2083">10.1126/sciadv.aba2083</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32917697">32917697</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11206444">PMC11206444</a></div></div>
</div>
<div id="ref-wfqjCerX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">77. </div><div class="csl-right-inline"><strong>The Electronic Medical Records and Genomics (eMERGE) Network: past, present, and future</strong> <div class="csl-block">Omri Gottesman, Helena Kuivaniemi, Gerard Tromp, WAndrew Faucett, Rongling Li, Teri A Manolio, Saskia C Sanderson, Joseph Kannry, Randi Zinberg, Melissa A Basford, … Marc S Williams</div> <em>Genetics in Medicine</em> (2013-10) <a href="https://doi.org/f5dwbt">https://doi.org/f5dwbt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/gim.2013.72">10.1038/gim.2013.72</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23743551">23743551</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3795928">PMC3795928</a></div></div>
</div>
<div id="ref-gZAOkumx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">78. </div><div class="csl-right-inline"><strong>Multi-ancestry gene-trait connection landscape using electronic health record (EHR) linked biobank data</strong> <div class="csl-block">Binglan Li, Yogasudha Veturi, Anastasia Lucas, Yuki Bradford, Shefali S Verma, Anurag Verma, Joseph Park, Wei-Qi Wei, Qiping Feng, Bahram Namjou, … Marylyn D Ritchie</div> <em>Cold Spring Harbor Laboratory</em> (2021-10-26) <a href="https://doi.org/gnbdnb">https://doi.org/gnbdnb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2021.10.21.21265225">10.1101/2021.10.21.21265225</a></div></div>
</div>
<div id="ref-F7lIlh2N" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">79. </div><div class="csl-right-inline"><strong>A Next Generation Connectivity Map: L1000 Platform and the First 1,000,000 Profiles</strong> <div class="csl-block">Aravind Subramanian, Rajiv Narayan, Steven M Corsello, David D Peck, Ted E Natoli, Xiaodong Lu, Joshua Gould, John F Davis, Andrew A Tubelli, Jacob K Asiedu, … Todd R Golub</div> <em>Cell</em> (2017-11) <a href="https://doi.org/cgwt">https://doi.org/cgwt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2017.10.049">10.1016/j.cell.2017.10.049</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29195078">29195078</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5990023">PMC5990023</a></div></div>
</div>
<div id="ref-mZjkE1xU" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">80. </div><div class="csl-right-inline"><strong>Discovery and Preclinical Validation of Drug Indications Using Compendia of Public Gene Expression Data</strong> <div class="csl-block">Marina Sirota, Joel T Dudley, Jeewon Kim, Annie P Chiang, Alex A Morgan, Alejandro Sweet-Cordero, Julien Sage, Atul J Butte</div> <em>Science Translational Medicine</em> (2011-08-17) <a href="https://doi.org/c3fwxv">https://doi.org/c3fwxv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/scitranslmed.3001318">10.1126/scitranslmed.3001318</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21849665">21849665</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3502016">PMC3502016</a></div></div>
</div>
<div id="ref-17oeJ0CXy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">81. </div><div class="csl-right-inline"><strong>Analysis of genome-wide association data highlights candidates for drug repositioning in psychiatry</strong> <div class="csl-block">Hon-Cheong So, Carlos Kwan-Long Chau, Wan-To Chiu, Kin-Sang Ho, Cho-Pong Lo, Stephanie Ho-Yue Yim, Pak-Chung Sham</div> <em>Nature Neuroscience</em> (2017-08-14) <a href="https://doi.org/gbrssh">https://doi.org/gbrssh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nn.4618">10.1038/nn.4618</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28805813">28805813</a></div></div>
</div>
<div id="ref-O21tn8vf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">82. </div><div class="csl-right-inline"><strong>Systematic integration of biomedical knowledge prioritizes drugs for repurposing</strong> <div class="csl-block">Daniel Scott Himmelstein, Antoine Lizee, Christine Hessler, Leo Brueggeman, Sabrina L Chen, Dexter Hadley, Ari Green, Pouya Khankhanian, Sergio E Baranzini</div> <em>eLife</em> (2017-09-22) <a href="https://doi.org/cdfk">https://doi.org/cdfk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7554/elife.26726">10.7554/elife.26726</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28936969">28936969</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5640425">PMC5640425</a></div></div>
</div>
<div id="ref-10KA5jTBQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">83. </div><div class="csl-right-inline"><strong>dhimmel/indications v1.0. PharmacotherapyDB: the open catalog of drug therapies for disease</strong> <div class="csl-block">Daniel S Himmelstein, Pouya Khankhanian, Christine S Hessler, Ari J Green, Sergio E Baranzini</div> <em>Zenodo</em> (2016-03-15) <a href="https://doi.org/f3mqwb">https://doi.org/f3mqwb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.5281/zenodo.47664">10.5281/zenodo.47664</a></div></div>
</div>
<div id="ref-LVihFr3g" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">84. </div><div class="csl-right-inline"><strong>Mechanism of Action of Niacin</strong> <div class="csl-block">Vaijinath S Kamanna, Moti L Kashyap</div> <em>The American Journal of Cardiology</em> (2008-04) <a href="https://doi.org/c8zwdt">https://doi.org/c8zwdt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.amjcard.2008.02.029">10.1016/j.amjcard.2008.02.029</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18375237">18375237</a></div></div>
</div>
<div id="ref-xJjWGVyN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">85. </div><div class="csl-right-inline"><strong>GWAS Catalog</strong> <a href="https://www.ebi.ac.uk/gwas/genes/DGAT2">https://www.ebi.ac.uk/gwas/genes/DGAT2</a></div>
</div>
<div id="ref-1GM0PmkWn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">86. </div><div class="csl-right-inline"><strong>GWAS Catalog</strong> <a href="https://www.ebi.ac.uk/gwas/genes/ACACA">https://www.ebi.ac.uk/gwas/genes/ACACA</a></div>
</div>
<div id="ref-Vxi0ejah" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">87. </div><div class="csl-right-inline"><strong>Memory loss in Alzheimer's disease</strong> <div class="csl-block">Holger Jahn</div> <em>Dialogues in Clinical Neuroscience</em> (2013-12-31) <a href="https://doi.org/gsfxqn">https://doi.org/gsfxqn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.31887/dcns.2013.15.4/hjahn">10.31887/dcns.2013.15.4/hjahn</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24459411">24459411</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3898682">PMC3898682</a></div></div>
</div>
<div id="ref-hn3T60wf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">88. </div><div class="csl-right-inline"><strong>&lt;i&gt;APOE4&lt;/i&gt;
disrupts intracellular lipid homeostasis in human iPSC-derived glia</strong> <div class="csl-block">Grzegorz Sienski, Priyanka Narayan, Julia Maeve Bonner, Nora Kory, Sebastian Boland, Aleksandra A Arczewska, William T Ralvenius, Leyla Akay, Elana Lockshin, Liang He, … Susan Lindquist</div> <em>Science Translational Medicine</em> (2021-03-03) <a href="https://doi.org/gjs64x">https://doi.org/gjs64x</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/scitranslmed.aaz4564">10.1126/scitranslmed.aaz4564</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33658354">33658354</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8218593">PMC8218593</a></div></div>
</div>
<div id="ref-10ZgydXh1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">89. </div><div class="csl-right-inline"><strong>ApoE in Alzheimer’s disease: pathophysiology and therapeutic strategies</strong> <div class="csl-block">Ana-Caroline Raulin, Sydney V Doss, Zachary A Trottier, Tadafumi C Ikezu, Guojun Bu, Chia-Chen Liu</div> <em>Molecular Neurodegeneration</em> (2022-11-08) <a href="https://doi.org/j3j8">https://doi.org/j3j8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13024-022-00574-4">10.1186/s13024-022-00574-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36348357">36348357</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9644639">PMC9644639</a></div></div>
</div>
<div id="ref-Om8ZhS06" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">90. </div><div class="csl-right-inline"><strong>Conditional and interaction gene-set analysis reveals novel functional pathways for blood pressure</strong> <div class="csl-block">Christiaan A de Leeuw, Sven Stringer, Ilona A Dekkers, Tom Heskes, Danielle Posthuma</div> <em>Nature Communications</em> (2018-09-14) <a href="https://doi.org/gd6d85">https://doi.org/gd6d85</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-018-06022-6">10.1038/s41467-018-06022-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30218068">30218068</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6138636">PMC6138636</a></div></div>
</div>
<div id="ref-jVQ2rMqc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">91. </div><div class="csl-right-inline"><strong>Gene regulatory network inference in the era of single-cell multi-omics</strong> <div class="csl-block">Pau Badia-i-Mompel, Lorna Wessels, Sophia Müller-Dott, Rémi Trimbour, Ricardo O Ramirez Flores, Ricard Argelaguet, Julio Saez-Rodriguez</div> <em>Nature Reviews Genetics</em> (2023-06-26) <a href="https://doi.org/gsdq3p">https://doi.org/gsdq3p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41576-023-00618-5">10.1038/s41576-023-00618-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37365273">37365273</a></div></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
