## Introduction 
Understanding the genetics of complex traits remains a formidable challenge.
Genome-Wide Association Studies (GWAS) have been instrumental in advancing our understanding of the genetics behind complex traits.
By scanning the genomes of large populations, GWAS identify associations between genetic variants and traits, providing valuable insights into the genetic architecture of diseases and other phenotypic traits.
This approach has been particularly successful in identifying thousands of variant-trait associations, shedding light on potential genetic factors involved in a wide range of conditions.
However, a significant limitation of GWAS is that most associated variants reside in non-coding regions of the genome, underscoring the critical role of gene regulation in generating diverse phenotypes from identical DNA sequences.


To bridge this gap, various methods have been developed to enhance the interpretation of GWAS findings.
One such approach is MAGMA, which assigns variants to genes based on physical proximity without considering gene expression data.
However, it has been shown that the closest gene to a variant is not necessarily the gene influenced by that variant, revealing the limitations of proximity-based methods.
Transcriptome-Wide Association Studies (TWAS) overcome this limitation by linking genetic variants to gene expression (eQTLs).
TWAS has evolved into a more general framework to test other molecular phenotypes (molQTLs).

Despite their advancements, TWAS and similar approaches typically focus on single variants or individual genes, thereby overlooking the intricate gene–gene interactions that are pivotal according to modern theories such as the omnigenic model of genetic architecture for complex traits.
Capturing these interactions necessitates more sophisticated methodologies.
One effective approach involves leveraging gene modules based on co-expression patterns.
Machine learning methods, such as decomposition techniques and autoencoders, have demonstrated superior performance in identifying these co-expression modules by capturing latent patterns in high-dimensional datasets.
However, the reliance on co-expression alone can lead to false positives, limiting the accuracy of these modules.

To mitigate this, integrating additional data sources is essential.
For instance, PhenoPLIER exemplifies an integrative approach by combining genetic variant-trait associations from GWAS and variant-gene expression-trait associations from TWAS with groups of functionally related genes.
This integration strengthens the reliability of gene modules by cross-validating co-expression patterns with genetic evidence, thus reducing the likelihood of false positives and enhancing the biological relevance of the results.

WE NEED TO ADD FUTURE PRESEPCTIVE PART: Depen if we incorporate the idea of GRN we can mention about the integration of other data types that can reconstruct GRN such as chromatin accessibility, histone marks…Alternatively, quantitative trait loci (QTL) analyses like *trans*-eQTLs and sQTLs explore distal genetic relationships and splicing variations, respectively.


In this review, we explore how integrating gene modeling machine learning frameworks with genetic and multi-omics data enhances our understanding of the molecular mechanisms underlying complex traits and diseases, effectively approaching genetic studies through the lens of gene networks.
