<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Marc Subirana-Granés" />
  <meta name="author" content="Jill Hoffman" />
  <meta name="author" content="Haoyu Zhang" />
  <meta name="author" content="Sutanu Nandi" />
  <meta name="author" content="Christina Akirtava" />
  <meta name="author" content="Kevin Fotso" />
  <meta name="author" content="Milton Pividori" />
  <meta name="dcterms.date" content="2024-10-10" />
  <meta name="keywords" content="genetic studies, gwas, twas, gene modules, gene co-expression networks, QTL, complex traits" />
  <title>Genetic studies through the lens of gene networks</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="Genetic studies through the lens of gene networks" />
  <meta name="citation_title" content="Genetic studies through the lens of gene networks" />
  <meta property="og:title" content="Genetic studies through the lens of gene networks" />
  <meta property="twitter:title" content="Genetic studies through the lens of gene networks" />
  <meta name="dc.date" content="2024-10-10" />
  <meta name="citation_publication_date" content="2024-10-10" />
  <meta property="article:published_time" content="2024-10-10" />
  <meta name="dc.modified" content="2024-10-10T16:25:25+00:00" />
  <meta property="article:modified_time" content="2024-10-10T16:25:25+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Marc Subirana-Granés" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0000-0003-3934-839X" />
  <meta name="twitter:creator" content="@msubirana20" />
  <meta name="citation_author" content="Jill Hoffman" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0000-0002-2690-2593" />
  <meta name="citation_author" content="Haoyu Zhang" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0009-0005-6025-0217" />
  <meta name="citation_author" content="Sutanu Nandi" />
  <meta name="citation_author_institution" content="Department of Pharmacology, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0009-0001-5887-999X" />
  <meta name="citation_author" content="Christina Akirtava" />
  <meta name="citation_author_institution" content="Department of Biochemistry and Molecular Genetics, RNA Bioscience Initiative, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0000-0001-7587-2303" />
  <meta name="twitter:creator" content="@bio2data" />
  <meta name="citation_author" content="Kevin Fotso" />
  <meta name="citation_author_institution" content="Office of Information Technology, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="XXX" />
  <meta name="citation_author" content="Milton Pividori" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado School of Medicine, Aurora, CO 80045, USA" />
  <meta name="citation_author_institution" content="Colorado Center for Personalized Medicine, University of Colorado Anschutz Medical Campus, Aurora, CO, USA" />
  <meta name="citation_author_orcid" content="0000-0002-3035-4403" />
  <meta name="twitter:creator" content="@miltondp" />
  <link rel="canonical" href="https://pivlab.github.io/annual_review_of_biomedical_data_science/" />
  <meta property="og:url" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/" />
  <meta property="twitter:url" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/" />
  <meta name="citation_fulltext_html_url" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/" />
  <meta name="citation_pdf_url" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://pivlab.github.io/annual_review_of_biomedical_data_science/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://pivlab.github.io/annual_review_of_biomedical_data_science/v/d3fd10990be3fa041e93f303cded32856ae04864/" />
  <meta name="manubot_html_url_versioned" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/v/d3fd10990be3fa041e93f303cded32856ae04864/" />
  <meta name="manubot_pdf_url_versioned" content="https://pivlab.github.io/annual_review_of_biomedical_data_science/v/d3fd10990be3fa041e93f303cded32856ae04864/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Genetic studies through the lens of gene networks</h1>
</header>
<p><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css"></p>
<p><span class="banner lightblue"><i class="fas fa-info-circle fa-lg"></i> <strong>When citing this paper, please use the following:</strong><br><br>
<em>Subirana-Granés M, Hoffman J, Zhang H, Nandi S, Akirtava C, Fotso K, Pividori M. Genetic studies through the lens of gene networks. 2025. Annu. Rev. Biomed. Data Sci. 8. https://doi.org/10.1146/annurev-biodatasci-103123-095355</em></span></p>
<p><small><em>
This manuscript
(<a href="https://pivlab.github.io/annual_review_of_biomedical_data_science/v/d3fd10990be3fa041e93f303cded32856ae04864/">permalink</a>)
was automatically generated
from <a href="https://github.com/pivlab/annual_review_of_biomedical_data_science/tree/d3fd10990be3fa041e93f303cded32856ae04864">pivlab/annual_review_of_biomedical_data_science@d3fd109</a>
on October 10, 2024.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Marc Subirana-Granés</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-3934-839X">0000-0003-3934-839X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/msubirana">msubirana</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/msubirana20">msubirana20</a>
<br>
<small>
Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Jill Hoffman</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2690-2593">0000-0002-2690-2593</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/jillhoffman">jillhoffman</a>
<br>
<small>
Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Haoyu Zhang</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0005-6025-0217">0009-0005-6025-0217</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/haoyu-zc">haoyu-zc</a>
<br>
<small>
Department of Biomedical Informatics, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Sutanu Nandi</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0001-5887-999X">0009-0001-5887-999X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/snandi-DS">snandi-DS</a>
<br>
<small>
Department of Pharmacology, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Christina Akirtava</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7587-2303">0000-0001-7587-2303</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/bio2data">bio2data</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/bio2data">bio2data</a>
<br>
<small>
Department of Biochemistry and Molecular Genetics, RNA Bioscience Initiative, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Kevin Fotso</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/XXX">XXX</a>
<br>
<small>
Office of Information Technology, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
</small></p></li>
<li><p><strong>Milton Pividori</strong>
<sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3035-4403">0000-0002-3035-4403</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/miltondp">miltondp</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/miltondp">miltondp</a>
· <img src="images/mastodon.svg" class="inline_icon" width="16" height="16" alt="Mastodon icon" />
<a href="https://genomic.social/@miltondp">@miltondp@genomic.social</a>
<br>
<small>
Department of Biomedical Informatics, University of Colorado School of Medicine, Aurora, CO 80045, USA; Colorado Center for Personalized Medicine, University of Colorado Anschutz Medical Campus, Aurora, CO, USA
· Funded by The National Human Genome Research Institute (K99/R00 HG011898); The Eunice Kennedy Shriver National Institute of Child Health and Human Development (R01 HD109765)
</small></p></li>
</ul>
<div id="correspondence">
<p>✉ — Correspondence possible via <a href="https://github.com/pivlab/annual_review_of_biomedical_data_science/issues">GitHub Issues</a>
or email to
Milton Pividori &lt;milton.pividori@cuanschutz.edu&gt;.</p>
</div>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>This is a very cool paper about something.
We are doing this and that.
And this are the conclusions.
And this is in <strong>boldface</strong>.</p>
<h2 id="introduction">Introduction</h2>
<p>Understanding the genetics of complex traits remains a formidable challenge, primarily addressed through Genome-Wide Association Studies (GWAS) that identify variant-trait associations without providing a mechanistic understanding of how these variants influence phenotypes.
A significant limitation of GWAS is that most associated variants reside in non-coding regions of the genome, underscoring the critical role of gene regulation in generating diverse phenotypes from identical DNA sequences.
To bridge this gap, Transcriptome-Wide Association Studies (TWAS) have been developed to link genetic variants to gene expression (eQTLs) or alternative splicing (sQTLs) and subsequently to specific traits or phenotypes.
However, TWAS and similar approaches typically focus on single variants or individual genes, thereby overlooking the intricate gene-gene interactions that are pivotal according to modern theories such as the omnigenic model of genetic architecture for complex traits.
Capturing these interactions necessitates more sophisticated methodologies.
Traditional approaches involve pathways or gene sets, which rely on predefined groups of functionally related genes but are constrained by existing knowledge.
Alternatively, QTL analyses like trans-eQTLs and sQTLs explore distal genetic relationships and splicing variations, respectively.
Another promising avenue is the construction of gene regulatory networks based on co-expression patterns, introducing the concept of gene modules.
These gene modules are often identified using unsupervised learning techniques applied to diverse and heterogeneous datasets, enabling the modeling of both linear and nonlinear relationships within the data.
Such methods transcend basic differential expression analyses by capturing higher-order connections between genes.
Tools like MAGMA and the more recent PhenoPLIER exemplify this integrative approach by combining variant-trait associations from GWAS or variant-gene expression-trait associations from TWAS with groups of functionally related genes, thereby enhancing the reliability of gene modules by mitigating false positives through additional layers of evidence.
Furthermore, these gene modules can be refined using quantitative omnigenic models that incorporate transcription factors (TFs) to better elucidate gene-gene interactions and potentially include aspects of alternative splicing.
Ongoing research continues to explore and develop novel methods that more effectively link expression patterns with genetic variation, aiming to provide a more comprehensive mechanistic understanding of the genetic basis of complex traits.</p>
<h2 id="single-variant-single-gene-approaches">Single-variant / single-gene approaches</h2>
<div id="fig:fig1" class="fignos">
<figure>
<img src="images/fig1.svg" style="width:100.0%" alt="Figure 1: Major approaches based on single-variant/single-gene strategies to understand the genetic basis of complex traits. From left to right GWAS, TWAS, PWAS, HAWAS and MWAS." />
<figcaption aria-hidden="true"><span>Figure 1:</span> <strong>Major approaches based on single-variant/single-gene strategies to understand the genetic basis of complex traits</strong>.
From left to right GWAS, TWAS, PWAS, HAWAS and MWAS.</figcaption>
</figure>
</div>
<p>Understanding the genetic underpinnings of complex traits and diseases has been a pivotal pursuit in the field of human genetics.
Early endeavors predominantly utilized genome-wide association studies (GWAS), which have been instrumental in identifying a foundational map of genomic regions linked to traits, offering valuable insights into the heritable components of complex diseases.</p>
<p>Building upon GWAS findings, efforts have been made to link these associated variants to specific genes, operating under the premise that variation in a single gene can elucidate disease-relevant biology.
Techniques such as Transcriptome-Wide Association Studies (TWAS) have emerged, integrating GWAS data with gene expression profiles to pinpoint genes whose expression levels correlate with phenotypic variation.
These approaches have advanced our understanding by moving from variant-level associations to functional gene-level insights.</p>
<h3 id="from-genetic-variants-to-traits-genome-wide-association-study-gwas">From genetic variants to traits: genome-wide association study (GWAS)</h3>
<p>In a simple trait, a single gene can be responsible for the disease, for example in the case of Sickle Cell Anemia and Huntington’s disease.
However, most traits are not so simplistic and are the result of many mutations across the genome.
Even these monogenic traits are impacted by the polygenic background <span class="citation" data-cites="VqGYQMXO">[<a href="#ref-VqGYQMXO" role="doc-biblioref">1</a>]</span>.
Genome-wide association studies (GWAS) examine the relationship between genetic variants and specific phenotypes by comparing allele frequencies in individuals of similar ancestry with distinct phenotypic traits (<strong>Figure <a href="#fig:fig1">1</a></strong>).
The statistical associations are typically assessed using linear regression for continuous traits and logistic regression for binary traits.
It is crucial to control for covariates such as age, sex, population structure, and ancestry to minimize confounding effects <span class="citation" data-cites="ofSdSqvD 1873zCaaX">[<a href="#ref-ofSdSqvD" role="doc-biblioref">2</a>,<a href="#ref-1873zCaaX" role="doc-biblioref">3</a>]</span>.
As many variants are found to be significant and due to the phenomena of linkage disequilibrium between causal and non-causal variants, the results of GWAS are often grouped into risk loci.
These loci represent clusters of variants that demonstrate significant associations, although not every variant may be causal.
Since the first GWAS or age-related macular degeneration (AMD) in 2005, the GWAS catalog rapidly expanded, containing ~400,000 curated SNP-trait associations from &gt;45,000 published GWAS across &gt;5,000 human traits <span class="citation" data-cites="VRCjjZ9L">[<a href="#ref-VRCjjZ9L" role="doc-biblioref">4</a>]</span>.
These studies have successfully identified risk loci for a variety of traits and diseases such as type 2 diabetes, auto-immune disease, schizophrenia, major depressive disorder, and more <span class="citation" data-cites="17B4AigZT yqBcaXHD">[<a href="#ref-17B4AigZT" role="doc-biblioref">5</a>,<a href="#ref-yqBcaXHD" role="doc-biblioref">6</a>]</span>.
As GWAS participants increase, more loci are able to be identified.
For example, a study investigating insomnia with a sample size of &gt;1M identified 202 risk loci <span class="citation" data-cites="sBTrhvlJ">[<a href="#ref-sBTrhvlJ" role="doc-biblioref">7</a>]</span>, compared to earlier studies with a sample size of ~110,00, which were only able to identify 3 risk loci <span class="citation" data-cites="crCAymA1">[<a href="#ref-crCAymA1" role="doc-biblioref">8</a>]</span>.
The increased sample sizes allows for identification of more common variants as well as rare and low-frequency variants <span class="citation" data-cites="17B4AigZT">[<a href="#ref-17B4AigZT" role="doc-biblioref">5</a>]</span>.
Beyond the identification of risk loci, GWAS has also led to the discovery of novel biological mechanisms.
In Crohn’s disease, a novel SNP was identified through GWAS in the gene <em>ATG16L1</em> (rs2241880) <span class="citation" data-cites="s6z5ZuY3">[<a href="#ref-s6z5ZuY3" role="doc-biblioref">9</a>]</span>.
This SNP causes a missense mutation that enhances caspase-3-mediated cleavage of <em>ATG16L1</em> and diminished autophagy.
As a result, intracellular bacterial clearance is impaired causing a chronic inflammatory state <span class="citation" data-cites="gxwza7fO">[<a href="#ref-gxwza7fO" role="doc-biblioref">10</a>]</span>.
A potential clinical application of GWAS involves quantifying a patient’s risk for a specific disease based on relevant SNPs represented by a polygenic risk score (PRS) <span class="citation" data-cites="ElanTkaD">[<a href="#ref-ElanTkaD" role="doc-biblioref">11</a>]</span>.
PRS’s can eventually be combined with current risk prediction models that include clinical, biochemical, and lifestyle features.
However, due to the majority of GWAS participants being from European ancestry, PRS has shown to not be generalizable to other groups.</p>
<p>Although GWAS have identified numerous genetic variants associated with complex traits, translating these findings into biological insights remains challenging <span class="citation" data-cites="17B4AigZT">[<a href="#ref-17B4AigZT" role="doc-biblioref">5</a>]</span>.
Many significant GWAS variants are located in non-coding regions, making it difficult to identify the specific genes that drive trait associations <span class="citation" data-cites="BqjxxhyF">[<a href="#ref-BqjxxhyF" role="doc-biblioref">12</a>]</span>.
A simple strategy to link GWAS-associated variants with a gene is by physical proximity, which typically selects the closest gene to the most significant SNP at each locus.
However, SNPs can have distal effects on the expression of the target gene <span class="citation" data-cites="9Pr9idng">[<a href="#ref-9Pr9idng" role="doc-biblioref">13</a>]</span>.
Genes can also have epistatic interactions, a secondary loci affecting a primary loci.
The peripheral effect of the secondary loci are often too small to be picked up through GWAS alone <span class="citation" data-cites="IrGVpNji">[<a href="#ref-IrGVpNji" role="doc-biblioref">14</a>]</span></p>
<p>Recent approaches combine multi-omics data from various cell types and tissues with GWAS to identify potential mechanisms of SNPs and the associated gene through molecular quantitative trait loci (moQTLs) <span class="citation" data-cites="QIEm5ZWK">[<a href="#ref-QIEm5ZWK" role="doc-biblioref">15</a>]</span> (<strong>Figure <a href="#fig:fig1">1</a></strong>).
An expression quantitative trait loci (eQTL) is a genetic region where variants are associated with the expression levels of a nearby or distant gene <span class="citation" data-cites="1FHKg3JK5">[<a href="#ref-1FHKg3JK5" role="doc-biblioref">16</a>]</span>.
eQTL studies estimate that most causal genes linked to GWAS loci are not the proximal ones <span class="citation" data-cites="14ZMJkeR2">[<a href="#ref-14ZMJkeR2" role="doc-biblioref">17</a>]</span>.
They have also allowed for insight on epistatic interactions.
A large eQTL study was able to detect epistatic interactions, showing that many primary loci affecting gene expression were also modified by secondary eQTLs <span class="citation" data-cites="IrGVpNji">[<a href="#ref-IrGVpNji" role="doc-biblioref">14</a>]</span>.
Splicing quantitative trait loci (sQTLs) are another major causal mechanism in GWAS loci, but are complex to interpret due to the production of unknown isoforms <span class="citation" data-cites="p13D2SfG">[<a href="#ref-p13D2SfG" role="doc-biblioref">18</a>]</span>.
To overcome this, LeafCutter <span class="citation" data-cites="WqkQN8X">[<a href="#ref-WqkQN8X" role="doc-biblioref">19</a>]</span> maps variable splicing events, alleviating the need for transcript annotations.
how this moqtls other mechanisms</p>
<p>Both GWAS and QTL studies have been penalized by their large multiple testing burden, causing the need to adopt a high level of significance.
This results in GWAS being underpowered to detect all heritability explained by SNPs <span class="citation" data-cites="17B4AigZT">[<a href="#ref-17B4AigZT" role="doc-biblioref">5</a>]</span>.
A strategy to combat this limitation is reducing the number of tests performed by focusing on gene co-expression modules or other prioritized genes [ttps://doi.org/10.1038/s41576-019-0127-1].</p>
<p>There are many strategies for leveraging GWAS data to identify gene sets related to a trait such as MAGMA <span class="citation" data-cites="19XiXgYmd">[<a href="#ref-19XiXgYmd" role="doc-biblioref">20</a>]</span> and FLAGS <span class="citation" data-cites="6CLIWmu2">[<a href="#ref-6CLIWmu2" role="doc-biblioref">21</a>]</span>.
FLAGS uses predefined gene sets from prior knowledge while MAGMA is able to infer its own gene sets by first correlating genes with phenotypes and other genes.
Both methods then statistically associate these modules with specific phenotypes.
These tools are able to include trans-interactions, which TWAS misses, however by looking at the effect of SNPs on a gene set, specific SNP gene connections are lost <span class="citation" data-cites="bjkAXJ07">[<a href="#ref-bjkAXJ07" role="doc-biblioref">22</a>]</span>.</p>
<h3 id="from-gwas-to-gene-transcriptome-wide-association-studies-twas">From GWAS to gene: transcriptome-wide association studies (TWAS)</h3>
<p>Transcriptome-wide association studies (TWAS) address this gap by integrating GWAS data with gene expression data (<strong>Figure <a href="#fig:fig1">1</a></strong>), typically derived from eQTL studies, to prioritize genes whose expression across different tissues is influenced by GWAS variants <span class="citation" data-cites="ReOPt75u l6ogswV3">[<a href="#ref-ReOPt75u" role="doc-biblioref">23</a>,<a href="#ref-l6ogswV3" role="doc-biblioref">24</a>]</span>.
By leveraging predicted gene expression levels, TWAS provides a mechanistic link between genetic variants and traits, allowing researchers to move beyond associations with individual SNPs to identify putatively causal genes <span class="citation" data-cites="l6ogswV3">[<a href="#ref-l6ogswV3" role="doc-biblioref">24</a>]</span>.
Since TWAS models the genetic regulation of gene expression, the approach allows researchers to impute expression levels in GWAS cohorts where expression data may not be available.
A key advantage of TWAS over GWAS lies in its ability to increase interpretability by providing a gene-trait association: TWAS connects trait-associated SNPs (which are mostly non-coding) to genes, which are biologically functional units.</p>
<p>Several TWAS approaches have been introduced <span class="citation" data-cites="RmhTH35O">[<a href="#ref-RmhTH35O" role="doc-biblioref">25</a>]</span>, including PrediXcan <span class="citation" data-cites="Z8bvDdVq">[<a href="#ref-Z8bvDdVq" role="doc-biblioref">26</a>]</span>, FUSION <span class="citation" data-cites="1D63fEEPb">[<a href="#ref-1D63fEEPb" role="doc-biblioref">27</a>]</span> and TIGAR <span class="citation" data-cites="19Klmizj8">[<a href="#ref-19Klmizj8" role="doc-biblioref">28</a>]</span>.
However, all of them implement a similar framework that consists in three steps: 1) model training, 2) gene expression imputation, and 3) gene-trait association.
For example, during 1), PrediXcan, builds one expression prediction model per gene and tissue using penalized linear regression with ElasticNet to model sparse genetic architectures.
These models contain weights for each SNP used as predictor for a gene expression in a tissue.
Given genotype data in a cohort without measured gene expression, during 2), the SNPs weights from the models can be used to impute tissue-specific gene expression for individuals.
During 3), a gene-tissue-trait association is computed by correlating the tissue-specific imputed gene expression with the trait of interest.
Most methods (such as Summary-PrediXcan <span class="citation" data-cites="vLyTudUB">[<a href="#ref-vLyTudUB" role="doc-biblioref">29</a>]</span> or S-PrediXcan), however, offer a shortcut by computing a gene-tissue-trait association directly from GWAS summary statistics without the need of individual-level data.
This process, however, requires the user to select a tissue of interest, which might not be straightforward <span class="citation" data-cites="l6ogswV3">[<a href="#ref-l6ogswV3" role="doc-biblioref">24</a>]</span>.
To address this limitation, approaches such as MultiXcan <span class="citation" data-cites="1FFzCXo1s">[<a href="#ref-1FFzCXo1s" role="doc-biblioref">30</a>]</span> or UTMOST <span class="citation" data-cites="93R9hBin">[<a href="#ref-93R9hBin" role="doc-biblioref">31</a>]</span> combine information across tissues to generate a gene-trait association.
These multi-tissue approaches are generally more powerful than single-tissue ones, although they do not provide a direction of effect (i.e., whether a higher or lower predicted expression is associated with a higher or lower disease risk).</p>
<h3 id="going-beyond-twas">Going beyond TWAS</h3>
<p>The flexibility of this 3-step framework can also be used to test whether other molecular phenotypes might mediate the association between GWAS variants and a trait of interest.
In addition to integrating eQTLs, the framework has been implemented also with splicing QTLs (sQTLs), protein QTLs (pQTLs), histone acetylation QTLs (haQTLs) and methylation QTLs (mQTLs).</p>
<h3 id="mapping-protein-function-to-disease-protein-wide-association-studies-pwas">Mapping protein function to disease: protein-wide association studies (PWAS)</h3>
<p>PWAS complements GWAS and TWAS by aggregating genetic variations in protein-coding regions to assess their combined impact on protein function and phenotypes (<strong>Figure <a href="#fig:fig1">1</a></strong>) <span class="citation" data-cites="1Dsj6bEd3">[<a href="#ref-1Dsj6bEd3" role="doc-biblioref">32</a>]</span>.
While GWAS focuses on individual variant associations, PWAS emphasizes the broader functional consequences of coding variants, providing a more comprehensive view of potential links to disease.
By aggregating multiple variants of a gene, PWAS reduces the multiple testing burden and reveals complex inheritance patterns, such as dominant and recessive traits.
PWAS shares a methodology with GWAS, using genotype, phenotype, and covariate data, but extends this approach by evaluating the cumulative impact of multiple variants on protein function.</p>
<p>To evaluate how genetic variants—such as missense, nonsense, and frameshift mutations—affect protein function, PWAS uses the FIRM machine learning model to assign impairment scores <span class="citation" data-cites="1Dsj6bEd3">[<a href="#ref-1Dsj6bEd3" role="doc-biblioref">32</a>]</span>.
Genes are then evaluated for dominant and recessive inheritance patterns, followed by statistical tests to determine significant associations between gene-level scores and phenotypes.
For example, using PWAS with the UK Biobank confirmed the association of <em>MUTYH</em> with colorectal cancer, even though no individual variant reached genome-wide significance in GWAS <span class="citation" data-cites="1Dsj6bEd3">[<a href="#ref-1Dsj6bEd3" role="doc-biblioref">32</a>]</span>.
This demonstrates PWAS’s ability to uncover functional associations missed by other methods.
Similarly, PWAS identified associations in genes like <em>HLA-DQA2</em> and <em>PSMB9</em> in asthma, highlighting its utility in exploring the genetic components of complex diseases.
<span class="citation" data-cites="CthLehGE">[<a href="#ref-CthLehGE" role="doc-biblioref">33</a>]</span>.
However, PWAS relies heavily on high-quality proteomic data and may overlook associations driven by non-coding variants.
It is also computationally intensive, and future efforts may focus on incorporating rare variants and more diverse data.</p>
<h3 id="enhancing-genetic-studies-with-epigenetic-data-epigenome-wide-association-studies-ewas">Enhancing genetic studies with epigenetic data: Epigenome-wide association studies (EWAS)</h3>
<p>Epigenome-wide association studies (EWAS) offer another valuable approach to understanding complex traits, specifically by exploring single-variant and single-gene relationships within the epigenetic context.
Unlike GWAS, which primarily investigate genetic variants, EWAS focuses on epigenetic markers such as DNA methylation and histone modifications to uncover their roles in gene regulation and disease etiology <span class="citation" data-cites="TPqvAJda">[<a href="#ref-TPqvAJda" role="doc-biblioref">34</a>]</span>.
Within the EWAS framework, methylome-wide association studies (MWAS) (<strong>Figure <a href="#fig:fig1">1</a></strong>) and histone acetylome-wide association studies (HAWAS) (<strong>Figure <a href="#fig:fig1">1</a></strong>), based on haQTLs and mQLTs respectability, represent specialized methodologies targeting specific types of epigenetic modifications .
MWAS concentrates on DNA methylation patterns, enabling the identification of methylation changes at individual loci associated with particular traits or diseases.
For instance, a MWAS conducted by Shen et al.
provided evidence for a causal relationship between DNAm and depression, suggesting that epigenetic modifications may mediate the genetic risk for depression, exemplifying the effectiveness of EWAS in uncovering the biological mechanisms that link genetic predisposition to psychiatric disorders <span class="citation" data-cites="1DNxNkLsT">[<a href="#ref-1DNxNkLsT" role="doc-biblioref">35</a>]</span>.
HAWAS focuses on histone acetylation modifications, which are crucial for regulating chromatin structure and gene expression.
Sun et al.
conducted a HAWAS on autism spectrum disorder (ASD) and identified thousands of differential acetylation loci in the brains of individuals with ASD, indicating that histone acetylation plays a significant role in ASD pathogenesis <span class="citation" data-cites="J7SRKSJl">[<a href="#ref-J7SRKSJl" role="doc-biblioref">36</a>]</span>.</p>
<p>The tissue-specific nature of epigenetic modifications, alongside the capacity to capture cellular plasticity and environmental influences, enhances our insight into the effects of genetic variants across distinct tissues, temporal cellular states, and gene-environment interactions.
This understanding contributes to a more detailed knowledge of the molecular mechanisms and regulatory dynamics underlying complex diseases, thereby supplementing the findings from GWAS <span class="citation" data-cites="FeaSnwhX KfWCLspO">[<a href="#ref-FeaSnwhX" role="doc-biblioref">37</a>,<a href="#ref-KfWCLspO" role="doc-biblioref">38</a>]</span>.</p>
<h3 id="single-gene-approaches-are-not-enough">Single-gene approaches are not enough</h3>
<p>Despite the advancements facilitated by single-variant and single-gene methodologies, a common thread persists: the focus remains on one gene at a time.
The underlying expectation is that identifying a gene linked to a trait will directly unveil the biological mechanisms driving disease processes.
While this has been successful in certain monogenic disorders, complex traits often involve intricate interactions among multiple genes and environmental factors.
Consequently, the one-gene-at-a-time paradigm may oversimplify the multifaceted nature of these traits.
In light of these considerations, it becomes imperative to reevaluate our approaches to dissecting the genetic architecture of complex traits.
This necessitates a shift towards methodologies that can capture the polygenic and network-based interactions inherent in complex diseases.</p>
<h2 id="from-single-genes-to-gene-networks-the-omnigenic-model-for-complex-traits">From single genes to gene networks: the omnigenic model for complex traits</h2>
<div id="fig:fig2" class="fignos">
<figure>
<img src="images/fig2.svg" style="width:100.0%" alt="Figure 2: Schematic of the omnigenic model. Schematic representation of the regulatory networks and associated phenotypes in an omnigenic context. Nodes represent individual genes, while edges indicate functional relationships between the nodes, such as transcription factor (TF) binding, splicing events, post-translational modifications (PTMs), gene co-expression, non-coding RNA (ncRNA), and protein-protein interactions (PPIs). The size of the arrows connecting phenotypes indicates the magnitude of their effect, with thicker arrows representing larger phenotypic influences. Core biological networks are indicated by shaded regions, with peripheral and core components shown as diamonds and circles, respectively. Phenotypes are linked to their relevant biological networks, highlighting the interplay between different systems and their contribution to phenotypic outcomes." />
<figcaption aria-hidden="true"><span>Figure 2:</span> <strong>Schematic of the omnigenic model</strong>.
Schematic representation of the regulatory networks and associated phenotypes in an omnigenic context.
Nodes represent individual genes, while edges indicate functional relationships between the nodes, such as transcription factor (TF) binding, splicing events, post-translational modifications (PTMs), gene co-expression, non-coding RNA (ncRNA), and protein-protein interactions (PPIs).
The size of the arrows connecting phenotypes indicates the magnitude of their effect, with thicker arrows representing larger phenotypic influences.
Core biological networks are indicated by shaded regions, with peripheral and core components shown as diamonds and circles, respectively.
Phenotypes are linked to their relevant biological networks, highlighting the interplay between different systems and their contribution to phenotypic outcomes.</figcaption>
</figure>
</div>
<h3 id="omnigenic-model-as-a-framework-for-deciphering-genetic-architecture-of-traits">Omnigenic model as a framework for deciphering genetic architecture of traits</h3>
<p>In the early 20th century, Mendelian geneticists, focused on traits determined by single genes where the segregation of alleles follows predictable patterns, debated with biometricians who studied continuously varying traits and argued that Mendelian inheritance could not explain the phenotypic distributions observed in populations.
This conflict highlighted a fundamental question: how can discrete genetic factors produce continuous phenotypic variation? <span class="citation" data-cites="vpIDZCSa 15ldVppuv">[<a href="#ref-vpIDZCSa" role="doc-biblioref">39</a>,<a href="#ref-15ldVppuv" role="doc-biblioref">40</a>]</span>.
The debate was resolved by R.
A.
Fisher in 1918 through the introduction of the infinitesimal model <span class="citation" data-cites="Qxk70FFV">[<a href="#ref-Qxk70FFV" role="doc-biblioref">41</a>]</span>.
Fisher demonstrated that if numerous genes influence a trait, the random assortment of alleles at these genes would produce a continuous distribution of phenotypes in a population.
This model reconciled Mendelian inheritance with the observed correlations between relatives for quantitative traits and provided a statistical framework for understanding complex trait inheritance.
Despite the success of Fisher’s infinitesimal model, the actual number of genes involved in complex traits and the magnitude of their effects remained uncertain for much of the 20th century.
Translating genotypes into phenotypes proved challenging due to the complexity of the genetic mechanisms driving trait variation.
The advent of genome-wide association studies (GWAS) brought new hope for identifying specific genetic variants with functional effects on complex traits.
GWAS allowed researchers to scan the entire genome for associations between genetic polymorphisms and traits of interest.</p>
<p>Over a decade of Genome-Wide Association Studies (GWAS) has led to several unexpected findings:
The strongest GWAS associations exhibit modest effect sizes on disease risk, collectively accounting for only a minor fraction of the predicted heritability (often termed “missing heritability”) <span class="citation" data-cites="RASeYPIy">[<a href="#ref-RASeYPIy" role="doc-biblioref">42</a>]</span>.
It is now understood that much of this missing heritability arises from a vast number of common variants with small effects that remain undetected in current sample sizes <span class="citation" data-cites="1AOy1zxAv">[<a href="#ref-1AOy1zxAv" role="doc-biblioref">43</a>]</span>.
A surprising uniform distribution of these small-effect significant variants exists throughout the genome, including noncoding regions, with a notable concentration in regulatory elements such as enhancers and promoters <span class="citation" data-cites="mwTa2RUK lVJVFaaZ">[<a href="#ref-mwTa2RUK" role="doc-biblioref">44</a>,<a href="#ref-lVJVFaaZ" role="doc-biblioref">45</a>]</span>.
For instance, is estimated that 71%–100% of 1-MB windows in the genome contribute to heritability for schizophrenia <span class="citation" data-cites="XvQe1H3A">[<a href="#ref-XvQe1H3A" role="doc-biblioref">46</a>]</span>.
A broader influence of the entire genome on phenotypic outcomes, not only genes with direct biological links to specific phenotypes, drives trait variation.
The notion that nearly all genomic variants could have direct additive effects on a specific phenotype is biologically improbable.
<span class="citation" data-cites="vpIDZCSa">[<a href="#ref-vpIDZCSa" role="doc-biblioref">39</a>]</span>
To reconcile these findings, Boyle et al.
proposed the omnigenic model <span class="citation" data-cites="vpIDZCSa">[<a href="#ref-vpIDZCSa" role="doc-biblioref">39</a>]</span>, a conceptual framework in which the genetic architecture of complex traits could be explained by highly interconnected gene regulatory networks.
The omnigenic model reframes the earlier infinitesimal model by distinguishing between “core genes”, which have a direct biological role in the disease or trait, and peripheral genes, which influence core genes and thus indirectly affect the disease or trait through regulatory networks <span class="citation" data-cites="vpIDZCSa">[<a href="#ref-vpIDZCSa" role="doc-biblioref">39</a>]</span>.
The omnigenic model, unlike the traditional infinitesimal model, offers a mechanistic rationale for complex traits architecture based on molecular and cellular biology, understanding the molecular pathways that connect genetic variation to phenotypic traits (<strong>Figure <a href="#fig:fig2">2</a></strong>).</p>
<p>The key proposals of this model are: 1) nearly all genes expressed in cells relevant to the trait have the potential to impact the regulation of core genes 2) that for typical traits, nearly all heritability is attributable to variation near peripheral genes.
Consequently, while core genes function as the primary drivers of disease, it is the cumulative effects of numerous peripheral gene variants that determine polygenic risk <span class="citation" data-cites="LXvTZzEA">[<a href="#ref-LXvTZzEA" role="doc-biblioref">47</a>]</span>.
This framework provides a possible explanation for several previously identified problems, including widespread pleiotropy (the ability of a single gene to affect multiple traits), polygenicity (the involvement of many genes in the manifestation of a single trait), the bias effect size where small-effect variants can collectively account for a large portion of heritability, and the uniform distribution across the genome.</p>
<h3 id="trans-effects-in-the-omnigenic-model-a-step-toward-decoding-complex-traits"><em>Trans</em> effects in the omnigenic model: a step toward decoding complex traits?</h3>
<p>The omnigenic model, though providing a comprehensive framework for understanding the diffuse heritability observed in GWAS, has faced several criticisms.
One key concern is its lack of clear, quantitative methods for identifying peripheral genes or estimating their contribution to heritability.
The model suggests that most genetic effects arise from these peripheral genes, yet it remains unclear how to define and measure these effects statistically.
<span class="citation" data-cites="82fs3SZo">[<a href="#ref-82fs3SZo" role="doc-biblioref">48</a>]</span>.
Trans effects occur when a genetic variant at one locus affects the expression of genes at distant loci, often on different chromosomes.
This contrasts with cis effects, where variants influence genes in their immediate vicinity.
In this context, understanding <em>trans</em> effects is crucial for grasping the indirect influence of peripheral genes on core traits <span class="citation" data-cites="javtAG7r">[<a href="#ref-javtAG7r" role="doc-biblioref">49</a>]</span>.</p>
<p>However, the study of trans effects has inherent challenges.
One of the primary limitations is their typically small effect sizes, which makes their detection difficult compared to cis effects.
Despite this, research indicates that most of the heritability in complex traits is controlled through trans effects rather than cis effects <span class="citation" data-cites="LXvTZzEA">[<a href="#ref-LXvTZzEA" role="doc-biblioref">47</a>]</span>.
This apparent paradox can be explained by the action of natural selection: it tends to act most strongly against variants with large effects, thereby limiting the potential contribution of genes with the most direct biological importance to heritability <span class="citation" data-cites="LXvTZzEA">[<a href="#ref-LXvTZzEA" role="doc-biblioref">47</a>]</span>.
Consequently, genes that are biologically critical for a trait might contribute less to heritability because their large-effect variants are more constrained by natural selection <span class="citation" data-cites="1BVFHVkmf">[<a href="#ref-1BVFHVkmf" role="doc-biblioref">50</a>]</span>.</p>
<p>Given that a significant portion of trait heritability appears to be driven by a vast number of peripheral variants functioning as weak trans effects for core genes, peripheral gene regulatory variants can become notable contributors if they influence many core genes in a coordinated direction.
When these trans-regulators act on multiple core genes correlatedly, their cumulative effects can substantially impact the trait.
By accounting for this additive effect, we can gain enough power to detect these trans effects, thereby understanding their contribution.</p>
<h3 id="success-stories-of-the-omnigenic-model">Success stories of the omnigenic model</h3>
<p>Since the introduction of the omnigenic model, numerous studies have adopted it as a framework to deepen our understanding of the genetic architecture of complex traits and the molecular mechanisms that drive them.</p>
<p>In schizophrenia, the application of the omnigenic model has revealed specific core gene sets associated with the disorder.
Notably, the <em>TCF4</em> gene set maintained its significant impact even after excluding SNPs within the <em>TCF4</em> gene itself.
This finding suggests that peripheral genes within the set contribute to the development of schizophrenia.
Additionally, the observed systematic floor effect across polygenic scores aligns with the model’s prediction that most genes expressed in relevant cells contribute to heritability, thereby highlighting the model’s potential in uncovering the broad genetic contributions to complex psychiatric conditions <span class="citation" data-cites="fV6THoBG">[<a href="#ref-fV6THoBG" role="doc-biblioref">51</a>]</span>.</p>
<p>Empirical support for the omnigenic theory extends beyond human diseases.
Sundell et al.
demonstrated this concept in European black poplars, showing that both core and peripheral genes play a crucial role in predicting phenotypes.
While core genes are indeed important, the information they provide must be complemented by other genes to ensure accurate phenotype predictions.
Furthermore, peripheral genes were found to carry significant biological information contributing to robust predictions <span class="citation" data-cites="EgquVUSg">[<a href="#ref-EgquVUSg" role="doc-biblioref">52</a>]</span>.</p>
<p>Li et al. (2024) identified a significantly connected subgraph formed by cancer-affected coding genes and ncRNAs by focusing on connectivity as a key topological feature.
This approach highlighted the essential role of ncRNAs in linking fragmented cancer-affected genes, consistent with the omnigenic framework’s premise that peripheral genes, including non-coding elements, contribute to complex traits such as cancer.
Importantly, the inclusion of ncRNAs enhanced the identification of cancer-related pathways, indicating that a comprehensive network model encompassing ncRNAs is more effective in characterizing disease relationships than models concentrating solely on coding genes.
<span class="citation" data-cites="13VkGrWQR">[<a href="#ref-13VkGrWQR" role="doc-biblioref">53</a>]</span>.</p>
<h3 id="limitations-of-the-omnigenic-model-from-a-theoretical-to-a-practical-approach">Limitations of the omnigenic model: from a theoretical to a practical approach</h3>
<p>In essence, the omnigenic model proposes a specific mechanism: that the genetic architecture of complex traits is intrinsically interconnected, involving a multitude of genes that contribute to phenotypic variation both directly (core genes) and indirectly (peripheral genes).
This framework facilitates an understanding of the highly polygenic and often subtle genetic effects on complex traits, shifting the focus from individual genes to gene networks.
However, the authors of omnigenic model proposed a set of equations to explain these interactions that contain many parameters that need to be fit, making it hard to implement from a practical point of view.
Moreover, the model exhibits certain limitations: (1) One limitation lies in the model’s reliance on network pleiotropy to explain the dominance of peripheral gene effects.
While it suggests that peripheral genes influence core genes through shared cellular networks, the model does propose a practical way to fully capture the complexity of these networks.
(2) The model also faces limitations in comprehending trans effects.
As previously discussed, the signals of trans effects are sometimes insufficiently detectable.
(3) It does not account for gene-environment interactions, assuming additivity as the basis for heritability estimates.
Consequently, the multifactorial effects of variants influencing the expression profiles of core genes remain unclear and are largely overlooked, not due to a lack of evidence but rather due to insufficient data integration <span class="citation" data-cites="1MuVFAgM dg9nKuy0">[<a href="#ref-1MuVFAgM" role="doc-biblioref">54</a>,<a href="#ref-dg9nKuy0" role="doc-biblioref">55</a>]</span>.
(4) There are challenges regarding the interpretability of the model and the translation of these theoretical concepts into a practical framework.
Human height: a model common complex trait
https://pubmed.ncbi.nlm.nih.gov/37343163/
it was already predicted on 2014
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4250049/
They discuss this paper: https://www.nature.com/articles/s41586-022-05275-y
And the implication for the omnigenic model.
https://www.nature.com/articles/s41586-022-05275-y#Sec7
cristisim from 2019 omingenic paper revision: In this study, we have defined core genes in line with the definition from Liu et al., 2019.
In that light, a transcription factor that regulates other core genes would be a peripheral gene, or perhaps a master regulator if it simultaneously regulates multiple core genes.
However, as defined by Liu et al., we consider signaling receptors, such as the androgen receptor for testosterone, to be core even though they are transcription factors, since they directly receive inputs from outside the cell.
Transcription factors that act within a cellular regulatory network , rather than directly receiving inputs from outside the cell, would be considered peripheral.
Nevertheless, we admit that this is a simplified, conceptual model, such that not all genes will fit neatly into these definitions.
We have made these points more clearly in the Introduction.
As George E.
P.
Box stated, “All models are wrong, but some are useful” We acknowledge that the omnigenic model simplifies biological complexity and may not be universally applicable to all complex traits.
Nevertheless, it remains a valuable framework for interpreting genetic signals.
This model proposes an explicit mechanism of gene-gene interactions and offers a pragmatic approach to understanding gene networks and their interactions, thereby facilitating insights into the underlying mechanisms of genetic influences on traits.</p>
<p>what we care the most is the network not the number of genes
framework starting point, some points are good other not<br />
We propose advancing the omnigenic model by developing computational frameworks based on this concept to infer gene-gene networks, termed machine learning-derived gene modules, to ground these theoretical ideas into practical applications.
Additionally, integrating multiple layers of omics data (genetic, epigenetic, and transcriptomic) can (1) enhance our understanding of complex biological networks and (2) provide sufficient information to improve the accuracy and comprehension of trans effect detection.
Since trans effects are mediated by mechanisms such as transcription factors, non-coding RNAs, epigenetic modifications, and chromatin remodeling complexes that regulate gene expression across the genome, incorporating these data types is essential.</p>
<h2 id="from-gene-networks-to-machine-learning-derived-gene-modules-hands-on-strategies-for-inferring-gene-gene-interactions">From gene networks to machine learning derived gene modules: hands-on strategies for inferring gene-gene interactions</h2>
<div id="fig:fig3" class="fignos">
<figure>
<img src="images/fig3.svg" style="width:100.0%" alt="Figure 3: Computational methdos for machine learning derived gene module detection.. a). Real biological network, where nodes represent genes and edges denote the relationships between them. It captures all possible patterns that can link genes. Some of these relationships are gene co-expression patterns, which transcriptomics can capture, such as linear relationships between Gene A and Gene B. Other patterns, such as protein-protein interactions between Gene C and Gene D, are not captured by transcriptomics and are thus not reflected in the co-expression analysis. b). Different approaches for capturing machine learning-derived gene modules. In this example, transcriptome data is used as input, but other data types such as methylomics, proteomics, and metabolomics can also be applied. The first row illustrates the PLIER method, which uses non-negative matrix factorization to align gene expression data with known biological pathways by incorporating prior knowledge into its gene modules. This approach generates interpretable gene modules but only captures linear relationships. The second row shows a Vanilla VAE, which can capture both linear and nonlinear patterns but does not integrate prior knowledge, resulting in less interpretable gene modules. The third row highlights the interpretable VAE, which incorporates prior knowledge into its decoder, achieving a balance between capturing complex relationships and maintaining biological interpretability." />
<figcaption aria-hidden="true"><span>Figure 3:</span> <strong>Computational methdos for machine learning derived gene module detection.</strong>.
<strong>a)</strong>.
Real biological network, where nodes represent genes and edges denote the relationships between them.
It captures all possible patterns that can link genes.
Some of these relationships are gene co-expression patterns, which transcriptomics can capture, such as linear relationships between Gene A and Gene B.
Other patterns, such as protein-protein interactions between Gene C and Gene D, are not captured by transcriptomics and are thus not reflected in the co-expression analysis.
<strong>b).</strong> Different approaches for capturing machine learning-derived gene modules.
In this example, transcriptome data is used as input, but other data types such as methylomics, proteomics, and metabolomics can also be applied.
The first row illustrates the PLIER method, which uses non-negative matrix factorization to align gene expression data with known biological pathways by incorporating prior knowledge into its gene modules.
This approach generates interpretable gene modules but only captures linear relationships.
The second row shows a Vanilla VAE, which can capture both linear and nonlinear patterns but does not integrate prior knowledge, resulting in less interpretable gene modules.
The third row highlights the interpretable VAE, which incorporates prior knowledge into its decoder, achieving a balance between capturing complex relationships and maintaining biological interpretability.</figcaption>
</figure>
</div>
<h2 id="from-single-gene-analysis-to-gene-module-discovery-with-machine-learning-approaches">From single gene analysis to gene module discovery with machine learning approaches</h2>
<p>High-throughput technologies, particularly genome-wide gene expression profiling tools such as RNA sequencing (RNA-seq) and microarrays, have fundamentally transformed the landscape of molecular biology.
Unlike traditional gene-by-gene approaches, RNA-seq and microarrays offer a comprehensive view of the transcriptomic landscape <span class="citation" data-cites="Zm8nV5AC">[<a href="#ref-Zm8nV5AC" role="doc-biblioref">56</a>]</span>.
These advancements, alongside conceptual frameworks like the omnigenic model, have facilitated a paradigm shift from a single-gene perspective to a module-based approach, wherein groups of genes rather than individual genes are essential for elucidating the complexities of biological networks <span class="citation" data-cites="SD886fuE">[<a href="#ref-SD886fuE" role="doc-biblioref">57</a>]</span>.</p>
<p>Gene modules, represented as coexpression networks, consist of nodes (genes) and edges that reflect coexpression relationships between them and are a key type of biological network (<strong>Figure <a href="#fig:fig3">3</a>a)</strong>).
These modules, formed by functionally related genes with coordinated expression patterns under specific biological conditions, not only reveal coexpression but also provide valuable insights into regulatory mechanisms, gene functions, and the pathways involved in disease <span class="citation" data-cites="QUI8V4RY">[<a href="#ref-QUI8V4RY" role="doc-biblioref">58</a>]</span>.</p>
<p>Numerous approaches and algorithms have been proposed for module detection in gene expression data.
ML techniques are particularly proficient at addressing common challenges in omics data, such as data sparsity and high dimensionality.
ML methods can compress correlations into lower-dimensional representations, facilitating the processing of large datasets to identify intricate coexpression patterns.
Additionally, ML approaches are capable of integrating various types of omics data enhancing the detection of gene modules, providing a more comprehensive understanding of biological networks <span class="citation" data-cites="K4KeL4gw">[<a href="#ref-K4KeL4gw" role="doc-biblioref">59</a>]</span>.</p>
<p>In this context, unsupervised ML approaches have gained significant popularity <span class="citation" data-cites="Ko8gMp8P">[<a href="#ref-Ko8gMp8P" role="doc-biblioref">60</a>]</span>.
These methods facilitate the discovery of patterns and structures within the data without requiring prior knowledge of the system, allowing for the inference of novel findings and hypotheses.
This capability is particularly advantageous for gene module inference, which aims to identify groups of coexpressed and potentially co-regulated genes that contribute to similar biological functions <span class="citation" data-cites="1BVbSrr6M">[<a href="#ref-1BVbSrr6M" role="doc-biblioref">61</a>]</span>.</p>
<p>In this section, we will focus on decomposition and deep learning methods for inferring gene modules, highlighting their effectiveness in uncovering complex gene interactions and enhancing our understanding of biological networks (<strong>Figure <a href="#fig:fig3">3</a>b)</strong>).
However, we acknowledge that utilizing transcriptional data alone is insufficient for fully comprehending the intricate system of biological networks; it serves as an initial step toward a more comprehensive understanding.</p>
<h3 id="decomposition-methods">Decomposition methods</h3>
<p>Decomposition methods reduce complex, high-dimensional gene expression data into simpler components, revealing underlying patterns and structures that aid in interpreting large-scale datasets.
These methods provide flexible approaches that handle local co-expression and overlap, making them especially powerful for large-scale and complex gene expression datasets since can capture context-specific co-expression and overlapping modules common in biological networks <span class="citation" data-cites="1BVbSrr6M">[<a href="#ref-1BVbSrr6M" role="doc-biblioref">61</a>]</span>.</p>
<p>Simple models like Principal Component Analysis (PCA) have traditionally been used to reduce dimensionality in gene expression data, offering a basic understanding of co-expressed genes <span class="citation" data-cites="1703M7bru">[<a href="#ref-1703M7bru" role="doc-biblioref">62</a>]</span>.
However, these models often lack the ability to incorporate prior biological knowledge, which limits their interpretability and biological relevance.
To overcome these limitations, more sophisticated approaches such as the Pathway-Level Information Extractor (PLIER) <span class="citation" data-cites="Ki2ij7zE">[<a href="#ref-Ki2ij7zE" role="doc-biblioref">63</a>]</span>, or GenomicSuperSignature <span class="citation" data-cites="X4fhSCkz">[<a href="#ref-X4fhSCkz" role="doc-biblioref">64</a>]</span> have been developed.
These methodologies combine unsupervised approaches (decomposition methodologies) with supervised approaches to annotate the gene modules with prior biological knowledge.</p>
<p>GenomicSuperSignature employs principal component analysis (PCA) to identify gene modules applying prior knowledge for biological annotation after module discovery.
This flexibility allows it to adapt to novel datasets and provides computational efficiency, even with large datasets.
However, the lack of prior knowledge integration during training can result in modules that are less biologically interpretable compared to those from PLIER-based methods <span class="citation" data-cites="X4fhSCkz">[<a href="#ref-X4fhSCkz" role="doc-biblioref">64</a>]</span>.</p>
<p>In contrast, PLIER is based on a non-negative matrix factorization method, and aligns gene expression data with known biological pathways by incorporating prior knowledge into its gene modules.
This method offers high interpretability by ensuring that a proportion of the gene modules align with specific biological pathways, enabling direct links between the inferred gene modules and known biological functions (<strong>Figure <a href="#fig:fig3">3</a>b)</strong>).
Notably, PLIER also reserves a subset of gene modules that do not align with pathways, allowing researchers to separate novel biology from technical artifacts.
While PLIER is scalable and effective for large RNA-seq datasets, its reliance on existing pathway information can sometimes limit its ability to explore unknown or poorly understood biological systems.
As PLIER uses prior knowledge to overcome the limitations of co-expression methods like Pearson or Spearman correlation, which can miss key causal gene expression changes due to confounding factors like cell type variability, allowing it to focus on biologically meaningful signals amidst complex gene-expression patterns.<span class="citation" data-cites="dg9nKuy0">[<a href="#ref-dg9nKuy0" role="doc-biblioref">55</a>]</span> .
Building upon PLIER, the MultiPLIER framework was developed to integrate large scale gene expression data and utilize transfer learning, incorporating information from multiple cell types and conditions <span class="citation" data-cites="14rnBunuZ">[<a href="#ref-14rnBunuZ" role="doc-biblioref">65</a>]</span>.
An example of this is its application to large datasets like recount2, a resource containing harmonized RNA-seq data from various experiments.
MultiPLIER leverages these vast datasets to improve the accuracy of inferred gene modules by identifying patterns consistent across diverse data types and conditions.
Most gene expression datasets, especially those related to rare diseases, are often too small to fully benefit from unsupervised machine learning.
To overcome this limitation, MultiPLIER employs transfer learning by training a PLIER model on a large public data compendium and then applying it to small datasets.
This approach generates models that align well with known biological factors and are more comprehensive than models trained on individual datasets.
When applied to rare disease datasets, these models effectively capture biological processes related to disease severity.
It’s worth noting that MultiPLIER allows for the simultaneous integration of multiple datasets, improving the interpretability and reliability of gene expression patterns across different experimental contexts.
In contrast, PLIER is better suited for analyzing a single dataset in a biologically informed manner.
PLIER can also be use in a specific cohort context, focuses on dissecting specific traits within more homogenous datasets., applied to smaller, more condition-specific datasets, as opposed to the MultiPLIER/recount2 model, which is based on more general, heterogeneous datasets.
Its utility have been showed by Nandi et al with the omniPLIER model, where they identify gene modules which mediating the effect of karyotype on body mass index in Down syndrome, revealing key regulators like <em>GPX1</em> and <em>MCL1</em> (Nandi et al., 2024).
This allows it to capture condition-specific regulatory mechanisms, but at the expense of broader generalizability, making it a powerful tool for understanding unique biological processes within specialized groups.</p>
<p>PLIER, initially designed for human gene expression data, has been successfully adapted for use in other species, demonstrating its versatility across different biological contexts.
By extending the framework to model organisms, researchers can infer gene modules in species beyond humans, expanding the utility of PLIER for comparative studies and translational research.
One such adaptation is MousiPLIER, which tailors the PLIER framework for use in mouse models, a widely used system in biomedical research.
<span class="citation" data-cites="1CDQw6VDO">[<a href="#ref-1CDQw6VDO" role="doc-biblioref">66</a>]</span></p>
<h3 id="deep-learning-methods-autoencoders">Deep learning methods: autoencoders</h3>
<p>However, decomposition methods can only infer linear patterns.
In contrast to these linear methods, non-linear models, such as deep learning (DL) approaches, can capture more complex patterns in the data.</p>
<p>A large set of tools based on DL approaches showed success in predicting gene expression directly from sequence <span class="citation" data-cites="1HNaN8NTb CdrNvVBn Gy0Zkruj">[<a href="#ref-1HNaN8NTb" role="doc-biblioref">67</a>,<a href="#ref-CdrNvVBn" role="doc-biblioref">68</a>,<a href="#ref-Gy0Zkruj" role="doc-biblioref">69</a>]</span>
Deep learning methods like autoencoders are particularly effective for inferring gene modules because they can capture non-linear patterns in gene expression data, unlike traditional linear decomposition methods.
Moreover, they can handle very high-dimensional data.
By using neural networks to learn compressed representations, autoencoders identify complex dependencies between genes, making them well-suited for uncovering gene modules.
Variational Autoencoders (VAEs) (<strong>Figure <a href="#fig:fig3">3</a>b)</strong>) enhance this process by encoding data into a continuous latent space, enabling dimensionality reduction and the generation of new data.
However, a major limitation of VAEs is their lack of biological interpretability, as they do not incorporate prior knowledge <span class="citation" data-cites="M8q5fpdY">[<a href="#ref-M8q5fpdY" role="doc-biblioref">70</a>]</span>.
To overcome this, models like the Pathway Module VAE (pmVAE) integrate biological pathway information into the autoencoder framework, improving interpretability <span class="citation" data-cites="nGVsOd2S">[<a href="#ref-nGVsOd2S" role="doc-biblioref">71</a>]</span> (<strong>Figure <a href="#fig:fig3">3</a>a)</strong>).
Another example is expiMap <span class="citation" data-cites="JkqcQgM7">[<a href="#ref-JkqcQgM7" role="doc-biblioref">72</a>]</span>, a biologically informed model that refines known gene programs while discovering novel ones.
This approach has proven successful in analyzing cross-dataset variations and applications such as studying COVID-19 patient responses, demonstrating the importance of combining VAEs with prior biological knowledge to enhance both inference and interpretability.
Additionally, beyond autoencoder-based methods, other correlation-based approaches, such as, the cluster-match correlation coefficient (CCC) have been developed to quantify both linear and non-linear correlations between genes in complex gene expression datasets.
Unlike traditional correlation measures (e.g.
Pearson correlation coefficient), which are limited to assessing linear associations between continuous data, CCC focuses on comparing how different gene expression profiles group or cluster data points, making it particularly effective in biological contexts where non-linearity often exists.
By examining the alignment of clusters across different gene expression categories, CCC provides a more flexible approach to detect complex relationships that are not apparent through linear models alone.
This is critical in gene expression data, where intricate regulatory networks and non-linear interactions between genes frequently occur, making CCC a powerful tool for uncovering deeper insights into gene functions and associations.
<span class="citation" data-cites="YgLIM2bZ">[<a href="#ref-YgLIM2bZ" role="doc-biblioref">73</a>]</span>.
While powerful in capturing intricate gene relationships and integrating multi-omics data, these non-linear models often face challenges in biological interpretability.</p>
<h2 class="page_break_before" id="a-gene-module-perspective-for-genetic-studies">A gene module perspective for genetic studies</h2>
<div id="fig:fig4" class="fignos">
<figure>
<img src="images/fig4.svg" style="width:100.0%" alt="Figure 4: An integrative, gene module-based approach for genetic studies. Each panel shows a component of the PhenoPLIER framework [74]. a) First, latent variables (LVs) or gene modules are learned from transcriptome data using the PLIER matrix factorization method. PLIER generates matrix \mathbf{Z}, which has gene weights for each module, and matrix \mathbf{B}, which has the samples in the latent space. b) Schematic of gene-trait associations from TWAS and gene-drug scores from LINCS L1000 being projected into the latent space for a joint analysis. c) Schematic of a gene module-based drug reporposing framework, where the projection of TWAS and LINCS L1000 data is used to compute a drug-disease score. d) Schematic of a regression model that tests whether genes that belong to a module (using a column of \mathbf{Z}) tend to be more strongly associated with a trait (using p-values from TWAS). e) (top) Example of a gene module identified as LV246 analyzed in [74]. DGAT2 and ACACA, found in a CRISPR screen to be linked to lipid metabolism and potential core genes, were among the top 15 genes in this module. Using metadata from the top samples for LV246 from \mathbf{B}, the module was found to be expressed mainly in adipose tissue, and liver and astrocyes were part of the top 10 cell types and tissues. Using the regression model depicted in d), this module was significantly associated with high cholesterol, tryglycerides, LDL cholesterol, cholesterol lowering medication, Alzheimer’s disease and dementia in the discovery cohort (UK Biobank), and high cholesterol and memory loss in the replication cohort (eMERGE). (bottom) The p-values and colocalization probability from gene-trait associations from TWAS for a subset of the top 1% genes in LV246 (y-axis) and traits significantly associated with LV246 (x-axis). It can be seen that DGAT2 and ACACA (in boldface), although strongly linked to lipid metabolism, are weakly associated with these lipid-relevant traits." />
<figcaption aria-hidden="true"><span>Figure 4:</span> <strong>An integrative, gene module-based approach for genetic studies.</strong>
Each panel shows a component of the PhenoPLIER framework <span class="citation" data-cites="NM3rHx1i">[<a href="#ref-NM3rHx1i" role="doc-biblioref">74</a>]</span>.
<strong>a)</strong> First, latent variables (LVs) or gene modules are learned from transcriptome data using the PLIER matrix factorization method.
PLIER generates matrix <span class="math inline">\(\mathbf{Z}\)</span>, which has gene weights for each module, and matrix <span class="math inline">\(\mathbf{B}\)</span>, which has the samples in the latent space.
<strong>b)</strong> Schematic of gene-trait associations from TWAS and gene-drug scores from LINCS L1000 being projected into the latent space for a joint analysis.
<strong>c)</strong> Schematic of a gene module-based drug reporposing framework, where the projection of TWAS and LINCS L1000 data is used to compute a drug-disease score.
<strong>d)</strong> Schematic of a regression model that tests whether genes that belong to a module (using a column of <span class="math inline">\(\mathbf{Z}\)</span>) tend to be more strongly associated with a trait (using <span class="math inline">\(p\)</span>-values from TWAS).
<strong>e)</strong> (top) Example of a gene module identified as LV246 analyzed in <span class="citation" data-cites="NM3rHx1i">[<a href="#ref-NM3rHx1i" role="doc-biblioref">74</a>]</span>.
<em>DGAT2</em> and <em>ACACA</em>, found in a CRISPR screen to be linked to lipid metabolism and potential core genes, were among the top 15 genes in this module.
Using metadata from the top samples for LV246 from <span class="math inline">\(\mathbf{B}\)</span>, the module was found to be expressed mainly in adipose tissue, and liver and astrocyes were part of the top 10 cell types and tissues.
Using the regression model depicted in d), this module was significantly associated with high cholesterol, tryglycerides, LDL cholesterol, cholesterol lowering medication, Alzheimer’s disease and dementia in the discovery cohort (UK Biobank), and high cholesterol and memory loss in the replication cohort (eMERGE).
(bottom) The <span class="math inline">\(p\)</span>-values and colocalization probability from gene-trait associations from TWAS for a subset of the top 1% genes in LV246 (<span class="math inline">\(y\)</span>-axis) and traits significantly associated with LV246 (<span class="math inline">\(x\)</span>-axis).
It can be seen that <em>DGAT2</em> and <em>ACACA</em> (in boldface), although strongly linked to lipid metabolism, are weakly associated with these lipid-relevant traits.</figcaption>
</figure>
</div>
<h2 class="page_break_before" id="future-perspectives">Future perspectives</h2>
<h2 class="page_break_before" id="conclusions">Conclusions</h2>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-VqGYQMXO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>Polygenic background modifies penetrance of monogenic variants for tier 1 genomic conditions</strong> <div class="csl-block">Akl C Fahed, Minxian Wang, Julian R Homburger, Aniruddh P Patel, Alexander G Bick, Cynthia L Neben, Carmen Lai, Deanna Brockman, Anthony Philippakis, Patrick T Ellinor, … Amit V Khera</div> <em>Nature Communications</em> (2020-08-20) <a href="https://doi.org/gg9hqv">https://doi.org/gg9hqv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-020-17374-3">10.1038/s41467-020-17374-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32820175">32820175</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7441381">PMC7441381</a></div></div>
</div>
<div id="ref-ofSdSqvD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>Genome-wide association studies</strong> <div class="csl-block">Emil Uffelmann, Qin Qin Huang, Nchangwi Syntia Munung, Jantina de Vries, Yukinori Okada, Alicia R Martin, Hilary C Martin, Tuuli Lappalainen, Danielle Posthuma</div> <em>Nature Reviews Methods Primers</em> (2021-08-26) <a href="https://doi.org/gmk7p2">https://doi.org/gmk7p2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s43586-021-00056-9">10.1038/s43586-021-00056-9</a></div></div>
</div>
<div id="ref-1873zCaaX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>Open problems in human trait genetics</strong> <div class="csl-block">Nadav Brandes, Omer Weissbrod, Michal Linial</div> <em>Genome Biology</em> (2022-06-20) <a href="https://doi.org/gqc962">https://doi.org/gqc962</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-022-02697-9">10.1186/s13059-022-02697-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35725481">35725481</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9208223">PMC9208223</a></div></div>
</div>
<div id="ref-VRCjjZ9L" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>The NHGRI-EBI GWAS Catalog: knowledgebase and deposition resource</strong> <div class="csl-block">Elliot Sollis, Abayomi Mosaku, Ala Abid, Annalisa Buniello, Maria Cerezo, Laurent Gil, Tudor Groza, Osman Güneş, Peggy Hall, James Hayhurst, … Laura W Harris</div> <em>Nucleic Acids Research</em> (2022-11-09) <a href="https://doi.org/gq73r5">https://doi.org/gq73r5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkac1010">10.1093/nar/gkac1010</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36350656">36350656</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9825413">PMC9825413</a></div></div>
</div>
<div id="ref-17B4AigZT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>Benefits and limitations of genome-wide association studies</strong> <div class="csl-block">Vivian Tam, Nikunj Patel, Michelle Turcotte, Yohan Bossé, Guillaume Paré, David Meyre</div> <em>Nature Reviews Genetics</em> (2019-05-08) <a href="https://doi.org/ggcxxb">https://doi.org/ggcxxb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41576-019-0127-1">10.1038/s41576-019-0127-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31068683">31068683</a></div></div>
</div>
<div id="ref-yqBcaXHD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>10 Years of GWAS Discovery: Biology, Function, and Translation</strong> <div class="csl-block">Peter M Visscher, Naomi R Wray, Qian Zhang, Pamela Sklar, Mark I McCarthy, Matthew A Brown, Jian Yang</div> <em>The American Journal of Human Genetics</em> (2017-07) <a href="https://doi.org/gcsmnm">https://doi.org/gcsmnm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ajhg.2017.06.005">10.1016/j.ajhg.2017.06.005</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28686856">28686856</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5501872">PMC5501872</a></div></div>
</div>
<div id="ref-sBTrhvlJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>Genome-wide analysis of insomnia in 1,331,010 individuals identifies new risk loci and functional pathways</strong><div class="csl-block">, Philip R Jansen, Kyoko Watanabe, Sven Stringer, Nathan Skene, Julien Bryois, Anke R Hammerschlag, Christiaan A de Leeuw, Jeroen S Benjamins, Ana B Muñoz-Manchado, … Danielle Posthuma</div> <em>Nature Genetics</em> (2019-02-25) <a href="https://doi.org/gfv4p2">https://doi.org/gfv4p2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-018-0333-3">10.1038/s41588-018-0333-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30804565">30804565</a></div></div>
</div>
<div id="ref-crCAymA1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>Genome-wide association analysis of insomnia complaints identifies risk genes and genetic overlap with psychiatric and metabolic traits</strong> <div class="csl-block">Anke R Hammerschlag, Sven Stringer, Christiaan A de Leeuw, Suzanne Sniekers, Erdogan Taskesen, Kyoko Watanabe, Tessa F Blanken, Kim Dekker, Bart HW te Lindert, Rick Wassing, … Danielle Posthuma</div> <em>Nature Genetics</em> (2017-06-12) <a href="https://doi.org/gbhzs5">https://doi.org/gbhzs5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.3888">10.1038/ng.3888</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28604731">28604731</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5600256">PMC5600256</a></div></div>
</div>
<div id="ref-s6z5ZuY3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>A genome-wide association scan of nonsynonymous SNPs identifies a susceptibility variant for Crohn disease in ATG16L1</strong> <div class="csl-block">Jochen Hampe, Andre Franke, Philip Rosenstiel, Andreas Till, Markus Teuber, Klaus Huse, Mario Albrecht, Gabriele Mayr, Francisco M De La Vega, Jason Briggs, … Stefan Schreiber</div> <em>Nature Genetics</em> (2006-12-31) <a href="https://doi.org/b9fr2w">https://doi.org/b9fr2w</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng1954">10.1038/ng1954</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17200669">17200669</a></div></div>
</div>
<div id="ref-gxwza7fO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>A Crohn’s disease variant in Atg16l1 enhances its degradation by caspase 3</strong> <div class="csl-block">Aditya Murthy, Yun Li, Ivan Peng, Mike Reichelt, Anand Kumar Katakam, Rajkumar Noubade, Merone Roose-Girma, Jason DeVoss, Lauri Diehl, Robert R Graham, Menno van Lookeren Campagne</div> <em>Nature</em> (2014-02) <a href="https://doi.org/f5r7wh">https://doi.org/f5r7wh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature13044">10.1038/nature13044</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24553140">24553140</a></div></div>
</div>
<div id="ref-ElanTkaD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Polygenic risk scores: from research tools to clinical instruments</strong> <div class="csl-block">Cathryn M Lewis, Evangelos Vassos</div> <em>Genome Medicine</em> (2020-05-18) <a href="https://doi.org/ghdnpx">https://doi.org/ghdnpx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13073-020-00742-5">10.1186/s13073-020-00742-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32423490">32423490</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7236300">PMC7236300</a></div></div>
</div>
<div id="ref-BqjxxhyF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Demystifying non-coding GWAS variants: an overview of computational tools and methods</strong> <div class="csl-block">Marijn Schipper, Danielle Posthuma</div> <em>Human Molecular Genetics</em> (2022-08-16) <a href="https://doi.org/grqdp7">https://doi.org/grqdp7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/hmg/ddac198">10.1093/hmg/ddac198</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35972862">35972862</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9585674">PMC9585674</a></div></div>
</div>
<div id="ref-9Pr9idng" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>The GTEx Consortium atlas of genetic regulatory effects across human tissues</strong><div class="csl-block">, François Aguet, Shankara Anand, Kristin G Ardlie, Stacey Gabriel, Gad A Getz, Aaron Graubert, Kane Hadley, Robert E Handsaker, Katherine H Huang, … Simona Volpi</div> <em>Science</em> (2020-09-11) <a href="https://doi.org/ghbnhr">https://doi.org/ghbnhr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.aaz1776">10.1126/science.aaz1776</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32913098">32913098</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7737656">PMC7737656</a></div></div>
</div>
<div id="ref-IrGVpNji" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>Molecular mechanisms of epistasis within and between genes</strong> <div class="csl-block">Ben Lehner</div> <em>Trends in Genetics</em> (2011-08) <a href="https://doi.org/fk3d5b">https://doi.org/fk3d5b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.tig.2011.05.007">10.1016/j.tig.2011.05.007</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21684621">21684621</a></div></div>
</div>
<div id="ref-QIEm5ZWK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>Strengthening Causal Inference for Complex Disease Using Molecular Quantitative Trait Loci</strong> <div class="csl-block">Sonja Neumeyer, Gibran Hemani, Eleftheria Zeggini</div> <em>Trends in Molecular Medicine</em> (2020-02) <a href="https://doi.org/ggmq94">https://doi.org/ggmq94</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.molmed.2019.10.004">10.1016/j.molmed.2019.10.004</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31718940">31718940</a></div></div>
</div>
<div id="ref-1FHKg3JK5" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>eQTL studies: from bulk tissues to single cells</strong> <div class="csl-block">Jingfei Zhang, Hongyu Zhao</div> <em>Journal of Genetics and Genomics</em> (2023-12) <a href="https://doi.org/g7n6d9">https://doi.org/g7n6d9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.jgg.2023.05.003">10.1016/j.jgg.2023.05.003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37207929">37207929</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10656365">PMC10656365</a></div></div>
</div>
<div id="ref-14ZMJkeR2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>The omnigenic model and polygenic prediction of complex traits</strong> <div class="csl-block">Iain Mathieson</div> <em>The American Journal of Human Genetics</em> (2021-09) <a href="https://doi.org/gmv9s5">https://doi.org/gmv9s5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ajhg.2021.07.003">10.1016/j.ajhg.2021.07.003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34331855">34331855</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8456163">PMC8456163</a></div></div>
</div>
<div id="ref-p13D2SfG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>Splicing QTL analysis focusing on coding sequences reveals mechanisms for disease susceptibility loci</strong> <div class="csl-block">Kensuke Yamaguchi, Kazuyoshi Ishigaki, Akari Suzuki, Yumi Tsuchida, Haruka Tsuchiya, Shuji Sumitomo, Yasuo Nagafuchi, Fuyuki Miya, Tatsuhiko Tsunoda, Hirofumi Shoda, … Yuta Kochi</div> <em>Nature Communications</em> (2022-08-24) <a href="https://doi.org/gqr76q">https://doi.org/gqr76q</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-022-32358-1">10.1038/s41467-022-32358-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36002455">36002455</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9402578">PMC9402578</a></div></div>
</div>
<div id="ref-WqkQN8X" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>Annotation-free quantification of RNA splicing using LeafCutter</strong> <div class="csl-block">Yang I Li, David A Knowles, Jack Humphrey, Alvaro N Barbeira, Scott P Dickinson, Hae Kyung Im, Jonathan K Pritchard</div> <em>Nature Genetics</em> (2017-12-11) <a href="https://doi.org/gcwk77">https://doi.org/gcwk77</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-017-0004-9">10.1038/s41588-017-0004-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29229983">29229983</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5742080">PMC5742080</a></div></div>
</div>
<div id="ref-19XiXgYmd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>MAGMA: Generalized Gene-Set Analysis of GWAS Data</strong> <div class="csl-block">Christiaan A de Leeuw, Joris M Mooij, Tom Heskes, Danielle Posthuma</div> <em>PLOS Computational Biology</em> (2015-04-17) <a href="https://doi.org/gf92gp">https://doi.org/gf92gp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1004219">10.1371/journal.pcbi.1004219</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25885710">25885710</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4401657">PMC4401657</a></div></div>
</div>
<div id="ref-6CLIWmu2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>FLAGS: A Flexible and Adaptive Association Test for Gene Sets Using Summary Statistics</strong> <div class="csl-block">Jianfei Huang, Kai Wang, Peng Wei, Xiangtao Liu, Xiaoming Liu, Kai Tan, Eric Boerwinkle, James B Potash, Shizhong Han</div> <em>Genetics</em> (2016-01-14) <a href="https://doi.org/f8db5f">https://doi.org/f8db5f</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1534/genetics.115.185009">10.1534/genetics.115.185009</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26773050">26773050</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4788129">PMC4788129</a></div></div>
</div>
<div id="ref-bjkAXJ07" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>A transcriptome-wide association study implicates specific pre- and post-synaptic abnormalities in schizophrenia</strong> <div class="csl-block">Lynsey S Hall, Christopher W Medway, Oliver Pain, Antonio F Pardiñas, Elliott G Rees, Valentina Escott-Price, Andrew Pocklington, Nicholas J Bray, Peter A Holmans, James TR Walters, … Michael C O’Donovan</div> <em>Human Molecular Genetics</em> (2019-11-06) <a href="https://doi.org/gst8k8">https://doi.org/gst8k8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/hmg/ddz253">10.1093/hmg/ddz253</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31691811">31691811</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7416679">PMC7416679</a></div></div>
</div>
<div id="ref-ReOPt75u" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>From GWAS to Gene: Transcriptome-Wide Association Studies and Other Methods to Functionally Understand GWAS Discoveries</strong> <div class="csl-block">Binglan Li, Marylyn D Ritchie</div> <em>Frontiers in Genetics</em> (2021-09-30) <a href="https://doi.org/g7n6kb">https://doi.org/g7n6kb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fgene.2021.713230">10.3389/fgene.2021.713230</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34659337">34659337</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8515949">PMC8515949</a></div></div>
</div>
<div id="ref-l6ogswV3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>Opportunities and challenges for transcriptome-wide association studies</strong> <div class="csl-block">Michael Wainberg, Nasa Sinnott-Armstrong, Nicholas Mancuso, Alvaro N Barbeira, David A Knowles, David Golan, Raili Ermel, Arno Ruusalepp, Thomas Quertermous, Ke Hao, … Anshul Kundaje</div> <em>Nature Genetics</em> (2019-03-29) <a href="https://doi.org/gf3hmr">https://doi.org/gf3hmr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-019-0385-z">10.1038/s41588-019-0385-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30926968">30926968</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6777347">PMC6777347</a></div></div>
</div>
<div id="ref-RmhTH35O" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>Transcriptome-wide association studies: recent advances in methods, applications and available databases</strong> <div class="csl-block">Jialin Mai, Mingming Lu, Qianwen Gao, Jingyao Zeng, Jingfa Xiao</div> <em>Communications Biology</em> (2023-09-01) <a href="https://doi.org/gs9f8r">https://doi.org/gs9f8r</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s42003-023-05279-y">10.1038/s42003-023-05279-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37658226">37658226</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10474133">PMC10474133</a></div></div>
</div>
<div id="ref-Z8bvDdVq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>A gene-based association method for mapping traits using reference transcriptome data</strong> <div class="csl-block">Eric R Gamazon, Heather E Wheeler, Kaanan P Shah, Sahar V Mozaffari, Keston Aquino-Michaels, Robert J Carroll, Anne E Eyler, Joshua C Denny, Dan L Nicolae, … Hae Kyung Im</div> <em>Nature Genetics</em> (2015-08-10) <a href="https://doi.org/f7p9zv">https://doi.org/f7p9zv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.3367">10.1038/ng.3367</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26258848">26258848</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4552594">PMC4552594</a></div></div>
</div>
<div id="ref-1D63fEEPb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>Integrative approaches for large-scale transcriptome-wide association studies</strong> <div class="csl-block">Alexander Gusev, Arthur Ko, Huwenbo Shi, Gaurav Bhatia, Wonil Chung, Brenda WJH Penninx, Rick Jansen, Eco JC de Geus, Dorret I Boomsma, Fred A Wright, … Bogdan Pasaniuc</div> <em>Nature Genetics</em> (2016-02-08) <a href="https://doi.org/f3vf4p">https://doi.org/f3vf4p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.3506">10.1038/ng.3506</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26854917">26854917</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4767558">PMC4767558</a></div></div>
</div>
<div id="ref-19Klmizj8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>TIGAR: An Improved Bayesian Tool for Transcriptomic Data Imputation Enhances Gene Mapping of Complex Traits</strong> <div class="csl-block">Sini Nagpal, Xiaoran Meng, Michael P Epstein, Lam C Tsoi, Matthew Patrick, Greg Gibson, Philip L De Jager, David A Bennett, Aliza P Wingo, Thomas S Wingo, Jingjing Yang</div> <em>The American Journal of Human Genetics</em> (2019-08) <a href="https://doi.org/g7n6dv">https://doi.org/g7n6dv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ajhg.2019.05.018">10.1016/j.ajhg.2019.05.018</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31230719">31230719</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6698804">PMC6698804</a></div></div>
</div>
<div id="ref-vLyTudUB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>Exploring the phenotypic consequences of tissue specific gene expression variation inferred from GWAS summary statistics</strong> <div class="csl-block">Alvaro N Barbeira, Scott P Dickinson, Rodrigo Bonazzola, Jiamao Zheng, Heather E Wheeler, Jason M Torres, Eric S Torstenson, Kaanan P Shah, Tzintzuni Garcia, Todd L Edwards, … Hae Kyung Im</div> <em>Nature Communications</em> (2018-05-08) <a href="https://doi.org/gdjvp5">https://doi.org/gdjvp5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-018-03621-1">10.1038/s41467-018-03621-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29739930">29739930</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5940825">PMC5940825</a></div></div>
</div>
<div id="ref-1FFzCXo1s" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>Integrating predicted transcriptome from multiple tissues improves association detection</strong> <div class="csl-block">Alvaro N Barbeira, Milton Pividori, Jiamao Zheng, Heather E Wheeler, Dan L Nicolae, Hae Kyung Im</div> <em>PLOS Genetics</em> (2019-01-22) <a href="https://doi.org/ghs8vx">https://doi.org/ghs8vx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pgen.1007889">10.1371/journal.pgen.1007889</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30668570">30668570</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6358100">PMC6358100</a></div></div>
</div>
<div id="ref-93R9hBin" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>A statistical framework for cross-tissue transcriptome-wide association analysis</strong><div class="csl-block">, Yiming Hu, Mo Li, Qiongshi Lu, Haoyi Weng, Jiawei Wang, Seyedeh M Zekavat, Zhaolong Yu, Boyang Li, Jianlei Gu, … Hongyu Zhao</div> <em>Nature Genetics</em> (2019-02-25) <a href="https://doi.org/gjjt5f">https://doi.org/gjjt5f</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-019-0345-7">10.1038/s41588-019-0345-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30804563">30804563</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6788740">PMC6788740</a></div></div>
</div>
<div id="ref-1Dsj6bEd3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>PWAS: proteome-wide association study—linking genes and phenotypes by functional variation in proteins</strong> <div class="csl-block">Nadav Brandes, Nathan Linial, Michal Linial</div> <em>Genome Biology</em> (2020-07-14) <a href="https://doi.org/gnvtr9">https://doi.org/gnvtr9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-020-02089-x">10.1186/s13059-020-02089-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32665031">32665031</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7386203">PMC7386203</a></div></div>
</div>
<div id="ref-CthLehGE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>PWAS Hub: Exploring Gene-Based Associations of Common Complex Diseases</strong> <div class="csl-block">Guy Kelman, Roei Zucker, Nadav Brandes, Michal Linial</div> <em>Cold Spring Harbor Laboratory</em> (2024-01-22) <a href="https://doi.org/g7n6hx">https://doi.org/g7n6hx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2024.01.20.23300645">10.1101/2024.01.20.23300645</a></div></div>
</div>
<div id="ref-TPqvAJda" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>Epigenome-wide association studies: current knowledge, strategies and recommendations</strong> <div class="csl-block">Maria Pia Campagna, Alexandre Xavier, Jeannette Lechner-Scott, Vicky Maltby, Rodney J Scott, Helmut Butzkueven, Vilija G Jokubaitis, Rodney A Lea</div> <em>Clinical Epigenetics</em> (2021-12) <a href="https://doi.org/gn3kds">https://doi.org/gn3kds</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13148-021-01200-8">10.1186/s13148-021-01200-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34863305">34863305</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8645110">PMC8645110</a></div></div>
</div>
<div id="ref-1DNxNkLsT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>DNA methylome-wide association study of genetic risk for depression implicates antigen processing and immune responses</strong> <div class="csl-block">Xueyi Shen, Doretta Caramaschi, Mark J Adams, Rosie M Walker, Josine L Min, Alex Kwong, Gibran Hemani, Miruna C Barbu, Heather C Whalley, … Andrew M McIntosh</div> <em>Genome Medicine</em> (2022-03-31) <a href="https://doi.org/g7n6jn">https://doi.org/g7n6jn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13073-022-01039-5">10.1186/s13073-022-01039-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35354486">35354486</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8969265">PMC8969265</a></div></div>
</div>
<div id="ref-J7SRKSJl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>Histone Acetylome-wide Association Study of Autism Spectrum Disorder</strong> <div class="csl-block">Wenjie Sun, Jeremie Poschmann, Ricardo Cruz-Herrera del Rosario, Neelroop N Parikshak, Hajira Shreen Hajan, Vibhor Kumar, Ramalakshmi Ramasamy, TGrant Belgard, Bavani Elanggovan, Chloe Chung Yi Wong, … Shyam Prabhakar</div> <em>Cell</em> (2016-11) <a href="https://doi.org/f9dpzf">https://doi.org/f9dpzf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2016.10.031">10.1016/j.cell.2016.10.031</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27863250">27863250</a></div></div>
</div>
<div id="ref-FeaSnwhX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>Histone Variants: Guardians of Genome Integrity</strong> <div class="csl-block">Juliette Ferrand, Beatrice Rondinelli, Sophie E Polo</div> <em>Cells</em> (2020-11-05) <a href="https://doi.org/g6vwfw">https://doi.org/g6vwfw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3390/cells9112424">10.3390/cells9112424</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33167489">33167489</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7694513">PMC7694513</a></div></div>
</div>
<div id="ref-KfWCLspO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><strong>Epigenetics: The DNA Methylation Profile of Tissue-Dependent and Differentially Methylated Regions in Cells</strong> <div class="csl-block">J Ohgane, S Yagi, K Shiota</div> <em>Placenta</em> (2008-03) <a href="https://doi.org/dfph4p">https://doi.org/dfph4p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.placenta.2007.09.011">10.1016/j.placenta.2007.09.011</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18031808">18031808</a></div></div>
</div>
<div id="ref-vpIDZCSa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><strong>An Expanded View of Complex Traits: From Polygenic to Omnigenic</strong> <div class="csl-block">Evan A Boyle, Yang I Li, Jonathan K Pritchard</div> <em>Cell</em> (2017-06) <a href="https://doi.org/gcpgdz">https://doi.org/gcpgdz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2017.05.038">10.1016/j.cell.2017.05.038</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28622505">28622505</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5536862">PMC5536862</a></div></div>
</div>
<div id="ref-15ldVppuv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>Polygenic inheritance, GWAS, polygenic risk scores, and the search for functional variants</strong> <div class="csl-block">Daniel JM Crouch, Walter F Bodmer</div> <em>Proceedings of the National Academy of Sciences</em> (2020-08-04) <a href="https://doi.org/gqh7nq">https://doi.org/gqh7nq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.2005634117">10.1073/pnas.2005634117</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32753378">32753378</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7431089">PMC7431089</a></div></div>
</div>
<div id="ref-Qxk70FFV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline"><strong>XV.—The Correlation between Relatives on the Supposition of Mendelian Inheritance.</strong> <div class="csl-block">RA Fisher</div> <em>Transactions of the Royal Society of Edinburgh</em> (1919) <a href="https://doi.org/gf2bfm">https://doi.org/gf2bfm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1017/s0080456800012163">10.1017/s0080456800012163</a></div></div>
</div>
<div id="ref-RASeYPIy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline"><strong>Finding the missing heritability of complex diseases</strong> <div class="csl-block">Teri A Manolio, Francis S Collins, Nancy J Cox, David B Goldstein, Lucia A Hindorff, David J Hunter, Mark I McCarthy, Erin M Ramos, Lon R Cardon, Aravinda Chakravarti, … Peter M Visscher</div> <em>Nature</em> (2009-10) <a href="https://doi.org/dt3">https://doi.org/dt3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature08494">10.1038/nature08494</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19812666">19812666</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2831613">PMC2831613</a></div></div>
</div>
<div id="ref-1AOy1zxAv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline"><strong>Solving the missing heritability problem</strong> <div class="csl-block">Alexander I Young</div> <em>PLOS Genetics</em> (2019-06-24) <a href="https://doi.org/ggnpnc">https://doi.org/ggnpnc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pgen.1008222">10.1371/journal.pgen.1008222</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31233496">31233496</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6611648">PMC6611648</a></div></div>
</div>
<div id="ref-mwTa2RUK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline"><strong>Common polygenic variation contributes to risk of schizophrenia and bipolar disorder</strong> <em>Nature</em> (2009-07-01) <a href="https://doi.org/cb5f2j">https://doi.org/cb5f2j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature08185">10.1038/nature08185</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19571811">19571811</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3912837">PMC3912837</a></div></div>
</div>
<div id="ref-lVJVFaaZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline"><strong>Toward Identification of Functional Sequences and Variants in Noncoding DNA</strong> <div class="csl-block">Remo Monti, Uwe Ohler</div> <em>Annual Review of Biomedical Data Science</em> (2023-08-10) <a href="https://doi.org/g7p7v4">https://doi.org/g7p7v4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1146/annurev-biodatasci-122120-110102">10.1146/annurev-biodatasci-122120-110102</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37262323">37262323</a></div></div>
</div>
<div id="ref-XvQe1H3A" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline"><strong>Contrasting genetic architectures of schizophrenia and other complex diseases using fast variance-components analysis</strong><div class="csl-block">, Po-Ru Loh, Gaurav Bhatia, Alexander Gusev, Hilary K Finucane, Brendan K Bulik-Sullivan, Samuela J Pollack, Teresa R de Candia, Sang Hong Lee, Naomi R Wray, … Alkes L Price</div> <em>Nature Genetics</em> (2015-11-02) <a href="https://doi.org/f3t9dq">https://doi.org/f3t9dq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.3431">10.1038/ng.3431</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26523775">26523775</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4666835">PMC4666835</a></div></div>
</div>
<div id="ref-LXvTZzEA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline"><strong>Trans Effects on Gene Expression Can Drive Omnigenic Inheritance</strong> <div class="csl-block">Xuanyao Liu, Yang I Li, Jonathan K Pritchard</div> <em>Cell</em> (2019-05) <a href="https://doi.org/gfz8bj">https://doi.org/gfz8bj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2019.04.014">10.1016/j.cell.2019.04.014</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31051098">31051098</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6553491">PMC6553491</a></div></div>
</div>
<div id="ref-82fs3SZo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline"><strong>Common Disease Is More Complex Than Implied by the Core Gene Omnigenic Model</strong> <div class="csl-block">Naomi R Wray, Cisca Wijmenga, Patrick F Sullivan, Jian Yang, Peter M Visscher</div> <em>Cell</em> (2018-06) <a href="https://doi.org/gdqcd4">https://doi.org/gdqcd4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2018.05.051">10.1016/j.cell.2018.05.051</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29906445">29906445</a></div></div>
</div>
<div id="ref-javtAG7r" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline"><strong>The Omnigenic Model: Response from the Authors</strong> <div class="csl-block">Journal of Psychiatry and Brain Science</div> (2017-10) <a href="https://doi.org/g7tw5d">https://doi.org/g7tw5d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.20900/jpbs.20170014s8">10.20900/jpbs.20170014s8</a></div></div>
</div>
<div id="ref-1BVFHVkmf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline"><strong>A population genetic interpretation of GWAS findings for human quantitative traits</strong> <div class="csl-block">Yuval B Simons, Kevin Bullaughey, Richard R Hudson, Guy Sella</div> <em>PLOS Biology</em> (2018-03-16) <a href="https://doi.org/gc64tx">https://doi.org/gc64tx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pbio.2002985">10.1371/journal.pbio.2002985</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29547617">29547617</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5871013">PMC5871013</a></div></div>
</div>
<div id="ref-fV6THoBG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline"><strong>The role of polygenic risk score gene-set analysis in the context of the omnigenic model of schizophrenia</strong><div class="csl-block">, Alexandros Rammos, Lara ANeira Gonzalez, Daniel R Weinberger, Kevin J Mitchell, Kristin K Nicodemus</div> <em>Neuropsychopharmacology</em> (2019-05-11) <a href="https://doi.org/g7p7t3">https://doi.org/g7p7t3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41386-019-0410-z">10.1038/s41386-019-0410-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31078131">31078131</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6785707">PMC6785707</a></div></div>
</div>
<div id="ref-EgquVUSg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline"><strong>Gene expression predictions and networks in natural populations supports the omnigenic theory</strong> <div class="csl-block">Aurélien Chateigner, Marie-Claude Lesage-Descauses, Odile Rogier, Véronique Jorge, Jean-Charles Leplé, Véronique Brunaud, Christine Paysant-Le Roux, Ludivine Soubigou-Taconnat, Marie-Laure Martin-Magniette, Leopoldo Sanchez, Vincent Segura</div> <em>BMC Genomics</em> (2020-06-22) <a href="https://doi.org/ghqvzz">https://doi.org/ghqvzz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12864-020-06809-2">10.1186/s12864-020-06809-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32571208">32571208</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7310122">PMC7310122</a></div></div>
</div>
<div id="ref-13VkGrWQR" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline"><strong>Non-Coding RNAs Extended Omnigenic Module of Cancers</strong> <div class="csl-block">Jie Li, Bingbo Wang, Xiujuan Ma</div> <em>Entropy</em> (2024-07-27) <a href="https://doi.org/g7p7wq">https://doi.org/g7p7wq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3390/e26080640">10.3390/e26080640</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/39202109">39202109</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11353529">PMC11353529</a></div></div>
</div>
<div id="ref-1MuVFAgM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline"><strong>Gene Interactions in Human Disease Studies—Evidence Is Mounting</strong> <div class="csl-block">Pankhuri Singhal, Shefali Setia Verma, Marylyn D Ritchie</div> <em>Annual Review of Biomedical Data Science</em> (2023-08-10) <a href="https://doi.org/g7p7vx">https://doi.org/g7p7vx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1146/annurev-biodatasci-102022-120818">10.1146/annurev-biodatasci-102022-120818</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37196359">37196359</a></div></div>
</div>
<div id="ref-dg9nKuy0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline"><strong>The great hairball gambit</strong> <div class="csl-block">Jonathan Flint, Trey Ideker</div> <em>PLOS Genetics</em> (2019-11-26) <a href="https://doi.org/g7p7wc">https://doi.org/g7p7wc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pgen.1008519">10.1371/journal.pgen.1008519</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31770365">31770365</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6910700">PMC6910700</a></div></div>
</div>
<div id="ref-Zm8nV5AC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline"><strong>High-throughput transcriptomics</strong> <div class="csl-block">Nunzio D’Agostino, Wenli Li, Dapeng Wang</div> <em>Scientific Reports</em> (2022-11-29) <a href="https://doi.org/g7p7vf">https://doi.org/g7p7vf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41598-022-23985-1">10.1038/s41598-022-23985-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36446824">36446824</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9708670">PMC9708670</a></div></div>
</div>
<div id="ref-SD886fuE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline"><strong>Gene module level analysis: identification to networks and dynamics</strong> <div class="csl-block">Xuewei Wang, Ertugrul Dalkic, Ming Wu, Christina Chan</div> <em>Current Opinion in Biotechnology</em> (2008-10) <a href="https://doi.org/fprfkv">https://doi.org/fprfkv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.copbio.2008.07.011">10.1016/j.copbio.2008.07.011</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18725293">18725293</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2615490">PMC2615490</a></div></div>
</div>
<div id="ref-QUI8V4RY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline"><strong>Emergence of co-expression in gene regulatory networks</strong> <div class="csl-block">Wencheng Yin, Luis Mendoza, Jimena Monzon-Sandoval, Araxi O Urrutia, Humberto Gutierrez</div> <em>PLOS ONE</em> (2021-04-01) <a href="https://doi.org/g7p7wf">https://doi.org/g7p7wf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pone.0247671">10.1371/journal.pone.0247671</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33793561">33793561</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8016302">PMC8016302</a></div></div>
</div>
<div id="ref-K4KeL4gw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline"><strong>Gene regulatory network reconstruction: harnessing the power of single-cell multi-omic data</strong> <div class="csl-block">Daniel Kim, Andy Tran, Hani Jieun Kim, Yingxin Lin, Jean Yee Hwa Yang, Pengyi Yang</div> <em>npj Systems Biology and Applications</em> (2023-10-19) <a href="https://doi.org/gt65pb">https://doi.org/gt65pb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41540-023-00312-6">10.1038/s41540-023-00312-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37857632">37857632</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10587078">PMC10587078</a></div></div>
</div>
<div id="ref-Ko8gMp8P" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline"><strong>The future is unsupervised</strong> <div class="csl-block">Casey S Greene</div> <em>Science Translational Medicine</em> (2016-07-06) <a href="https://doi.org/gg7qr3">https://doi.org/gg7qr3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/scitranslmed.aag3101">10.1126/scitranslmed.aag3101</a></div></div>
</div>
<div id="ref-1BVbSrr6M" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline"><strong>A comprehensive evaluation of module detection methods for gene expression data</strong> <div class="csl-block">Wouter Saelens, Robrecht Cannoodt, Yvan Saeys</div> <em>Nature Communications</em> (2018-03-15) <a href="https://doi.org/gc9x36">https://doi.org/gc9x36</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-018-03424-4">10.1038/s41467-018-03424-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29545622">29545622</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5854612">PMC5854612</a></div></div>
</div>
<div id="ref-1703M7bru" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline"><strong>Module representatives for refining gene co-expression modules</strong> <div class="csl-block">Nathan Mankovich, Helene Andrews-Polymenis, David Threadgill, Michael Kirby</div> <em>Physical Biology</em> (2023-05-04) <a href="https://doi.org/g7p7vk">https://doi.org/g7p7vk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1088/1478-3975/acce8d">10.1088/1478-3975/acce8d</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37075776">37075776</a></div></div>
</div>
<div id="ref-Ki2ij7zE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline"><strong>Pathway-level information extractor (PLIER) for gene expression data</strong> <div class="csl-block">Weiguang Mao, Elena Zaslavsky, Boris M Hartmann, Stuart C Sealfon, Maria Chikina</div> <em>Nature Methods</em> (2019-06-27) <a href="https://doi.org/gf75g6">https://doi.org/gf75g6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41592-019-0456-1">10.1038/s41592-019-0456-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31249421">31249421</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7262669">PMC7262669</a></div></div>
</div>
<div id="ref-X4fhSCkz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline"><strong>GenomicSuperSignature facilitates interpretation of RNA-seq experiments through robust, efficient comparison to public databases</strong> <div class="csl-block">Sehyun Oh, Ludwig Geistlinger, Marcel Ramos, Daniel Blankenberg, Marius van den Beek, Jaclyn N Taroni, Vincent J Carey, Casey S Greene, Levi Waldron, Sean Davis</div> <em>Nature Communications</em> (2022-06-27) <a href="https://doi.org/gqd7hm">https://doi.org/gqd7hm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-022-31411-3">10.1038/s41467-022-31411-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35760813">35760813</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9237024">PMC9237024</a></div></div>
</div>
<div id="ref-14rnBunuZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline"><strong>MultiPLIER: A Transfer Learning Framework for Transcriptomics Reveals Systemic Features of Rare Disease</strong> <div class="csl-block">Jaclyn N Taroni, Peter C Grayson, Qiwen Hu, Sean Eddy, Matthias Kretzler, Peter A Merkel, Casey S Greene</div> <em>Cell Systems</em> (2019-05) <a href="https://doi.org/gf75g5">https://doi.org/gf75g5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cels.2019.04.003">10.1016/j.cels.2019.04.003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31121115">31121115</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6538307">PMC6538307</a></div></div>
</div>
<div id="ref-1CDQw6VDO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline"><strong>MousiPLIER: A Mouse Pathway-Level Information Extractor Model</strong> <div class="csl-block">Shuo Zhang, Benjamin J Heil, Weiguang Mao, Maria Chikina, Casey S Greene, Elizabeth A Heller</div> <em>eneuro</em> (2024-05-24) <a href="https://doi.org/g7p7wj">https://doi.org/g7p7wj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1523/eneuro.0313-23.2024">10.1523/eneuro.0313-23.2024</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/38789274">38789274</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11154669">PMC11154669</a></div></div>
</div>
<div id="ref-1HNaN8NTb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline"><strong>Deep learning sequence-based ab initio prediction of variant effects on expression and disease risk</strong> <div class="csl-block">Jian Zhou, Chandra L Theesfeld, Kevin Yao, Kathleen M Chen, Aaron K Wong, Olga G Troyanskaya</div> <em>Nature Genetics</em> (2018-07-16) <a href="https://doi.org/gdvmqw">https://doi.org/gdvmqw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-018-0160-6">10.1038/s41588-018-0160-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30013180">30013180</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6094955">PMC6094955</a></div></div>
</div>
<div id="ref-CdrNvVBn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline"><strong>Predicting mRNA Abundance Directly from Genomic Sequence Using Deep Convolutional Neural Networks</strong> <div class="csl-block">Vikram Agarwal, Jay Shendure</div> <em>Cell Reports</em> (2020-05) <a href="https://doi.org/ggw7fr">https://doi.org/ggw7fr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.celrep.2020.107663">10.1016/j.celrep.2020.107663</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32433972">32433972</a></div></div>
</div>
<div id="ref-Gy0Zkruj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline"><strong>Atlas of primary cell-type-specific sequence models of gene expression and variant effects</strong> <div class="csl-block">Ksenia Sokolova, Chandra L Theesfeld, Aaron K Wong, Zijun Zhang, Kara Dolinski, Olga G Troyanskaya</div> <em>Cell Reports Methods</em> (2023-09) <a href="https://doi.org/gt6n4t">https://doi.org/gt6n4t</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.crmeth.2023.100580">10.1016/j.crmeth.2023.100580</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37703883">37703883</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10545936">PMC10545936</a></div></div>
</div>
<div id="ref-M8q5fpdY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline"><strong>Genomic data imputation with variational auto-encoders</strong> <div class="csl-block">Yeping Lina Qiu, Hong Zheng, Olivier Gevaert</div> <em>GigaScience</em> (2020-08-01) <a href="https://doi.org/gjbph2">https://doi.org/gjbph2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/gigascience/giaa082">10.1093/gigascience/giaa082</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32761097">32761097</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7407276">PMC7407276</a></div></div>
</div>
<div id="ref-nGVsOd2S" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">71. </div><div class="csl-right-inline"><strong>pmVAE: Learning Interpretable Single-Cell Representations with Pathway Modules</strong> <div class="csl-block">Gilles Gut, Stefan G Stark, Gunnar Rätsch, Natalie R Davidson</div> <em>Cold Spring Harbor Laboratory</em> (2021-01-30) <a href="https://doi.org/g7p7vq">https://doi.org/g7p7vq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2021.01.28.428664">10.1101/2021.01.28.428664</a></div></div>
</div>
<div id="ref-JkqcQgM7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">72. </div><div class="csl-right-inline"><strong>Biologically informed deep learning to query gene programs in single-cell atlases</strong> <div class="csl-block">Mohammad Lotfollahi, Sergei Rybakov, Karin Hrovatin, Soroor Hediyeh-zadeh, Carlos Talavera-López, Alexander V Misharin, Fabian J Theis</div> <em>Nature Cell Biology</em> (2023-02-02) <a href="https://doi.org/gtb96q">https://doi.org/gtb96q</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41556-022-01072-x">10.1038/s41556-022-01072-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36732632">36732632</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9928587">PMC9928587</a></div></div>
</div>
<div id="ref-YgLIM2bZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">73. </div><div class="csl-right-inline"><strong>An efficient, not-only-linear correlation coefficient based on clustering</strong> <div class="csl-block">Milton Pividori, Marylyn D Ritchie, Diego H Milone, Casey S Greene</div> <em>Cell Systems</em> (2024-09) <a href="https://doi.org/nhs4">https://doi.org/nhs4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cels.2024.08.005">10.1016/j.cels.2024.08.005</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/39243756">39243756</a></div></div>
</div>
<div id="ref-NM3rHx1i" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline"><strong>Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms</strong> <div class="csl-block">Milton Pividori, Sumei Lu, Binglan Li, Chun Su, Matthew E Johnson, Wei-Qi Wei, Qiping Feng, Bahram Namjou, Krzysztof Kiryluk, Iftikhar J Kullo, … Casey S Greene</div> <em>Nature Communications</em> (2023-09-09) <a href="https://doi.org/gspsxr">https://doi.org/gspsxr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-023-41057-4">10.1038/s41467-023-41057-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37689782">37689782</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10492839">PMC10492839</a></div></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
